---
title: "Borgmeyer_2021_Proteomics_&_Lipidomics"
author: "Stephanie Huang"
date: "2025_10_24"
output: html_document
---

## Re-analysis/Re-plotting of Borgmeyer et al., 2021

*in Cell Reports - "Multiomics of synaptic junctions reveals altered lipid metabolism and signaling following environmental enrichment"*

Analysis originally done in R_WD \> Proteomics \> EE_TMT \> v3_branched on 2025_10_24

ADD More details!

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# library(tidyverse) # ggplot, dplyr, etc
library(dplyr)
library(ggplot2)
library(stringr)
library(readr)
library(readxl) # for read_excel()
library(ggrepel) #  for geom_text_repel()
library(gt) # gt() tables
library(UniProt.ws) # for mapping | note conflicts with dplyr select
```

### Proteomics - Lipidomics - Table S7 - LFQ DEG MSstats - EE vs SH Hippocampus Synaptic Junctions

```{r}
hip_pro_file <- read_excel("input/Borgmeyer_2021_Table_S7_ProteinsLipids_EE_fig4a4b.xlsx",
                         sheet = "Proteins")
colnames(hip_pro_file)
str(hip_pro_file)
head(hip_pro_file)
```

```{r}

hippro_df <- hip_pro_file %>%
  rename("Protein" = "Protein/LipidName") %>%
  rename("Protein Name" = "ProteinName/Lipid Species") %>%
  mutate(EntryName = str_to_title(EntryName)) %>%
  arrange(pvalue)
  
```

<https://www.sthda.com/english/wiki/wiki.php?id_contents=7991> (text labels)

```{r fig.height=7, fig.width=10}
volcano_pv <- ggplot(hippro_df, 
                     aes(x =log(FoldChange), y = -log10(pvalue), label = EntryName)) +
  geom_point(aes(color = case_when(pvalue < 0.05 & log(FoldChange) > 0 ~ "Upregulated (p<0.05)",
                                   pvalue < 0.05 & log(FoldChange) < 0 ~ "Downregulated (p<0.05)",
                                   pvalue < 0.1 & log(FoldChange) > 0 ~ "Upregulated (p<0.1)",
                                   pvalue < 0.1 & log(FoldChange) < 0 ~ "Downregulated (p<0.1)",
                                   TRUE ~ "Not Significant")),
            alpha = 1, size = 3, shape = 1) +
  # facet_wrap(~ Label) +
  # xlim(-2,2) +
  scale_color_manual(values = c("firebrick3", 
                                "royalblue4", 
                                 "grey80"),
                     breaks = c("Upregulated (p<0.05)", 
                                "Downregulated (p<0.05)", 
                                "Not Significant"),
                     labels = c("Upregulated Proteins (p<0.05)", 
                                "Downregulated Proteins (p<0.05)", 
                                "Not Significant")) +
  
  geom_text_repel() +
  labs(title = "Borgmeyer 2021 SuppTable 7 - Significant Proteins (p<0.05) (24 Oct 2025)",
       subtitle = "Presented like Borgmeyer 2021 Fig4a or PhD Fig10A - log(FoldChange) by -log10(pvalue)",
       x = "Log(Fold Change)",
       y = "-Log10(p-value)",
       color = "Protein Regulation") +
  # geom_hline(yintercept = -log2(0.05), linetype = "dashed", linewidth = 0.2,  color = "grey30") +
  # annotate("text", x = .86, y = -log2(0.05), label = "Adj. p-value cutoff (0.05) \n",
  #          fontface = 'italic', size = 4, color = "grey10") +
  # geom_hline(yintercept = -log2(0.1), linetype = "dashed", linewidth = 0.2,  color = "grey50") +
  # annotate("text", x = .86, y = -log2(0.1), label = "Adj. p-value cutoff (0.1) \n",
  #          fontface = 'italic', size = 4, color = "grey30") +
  theme_bw() + 
  theme(legend.position = "bottom")



volcano_pv
```

```{r fig.height=7, fig.width=10}
volcano_adjpv <- ggplot(hippro_df, 
                     aes(x =log2FC, y = -log2(adj.pvalue), label = EntryName)) +
 geom_point(aes(color = case_when(adj.pvalue < 0.05 & log2FC > 0 ~ "Upregulated (adj. p<0.05)",
                                   adj.pvalue < 0.05 & log2FC < 0 ~ "Downregulated (adj. p<0.05)",
                                   adj.pvalue < 0.1 & log2FC > 0 ~ "Upregulated (adj. p<0.1)",
                                   adj.pvalue < 0.1 & log2FC < 0 ~ "Downregulated (adj. p<0.1)",
                                   TRUE ~ "Not Significant")),
            alpha = 1, size = 3, shape = 1) +
  # facet_wrap(~ Label) +
  # xlim(-1,1) +
  scale_color_manual(values = c("firebrick3", 
                                "royalblue4", 
                                "lightpink1", 
                                "lightblue3" , 
                                 "grey80"),
                     breaks = c("Upregulated (adj. p<0.05)", 
                                "Downregulated (adj. p<0.05)", 
                                "Upregulated (adj. p<0.1)", 
                                "Downregulated (adj. p<0.1)", 
                                "Not Significant"),
                     labels = c("Upregulated Proteins (up to adj. p<0.05)", 
                                "Downregulated Proteins (up to adj. p<0.05)", 
                                "Upregulated Proteins (up to adj. p<0.1)", 
                                "Downregulated Proteins (up to adj. p<0.1)" , 
                                "Not Significant")) +
  
  geom_text_repel() +
  labs(title = "Borgmeyer 2021 SuppTable 7 - Significant Proteins (p<0.05) (24 Oct 2025)",
       subtitle = "Presented like MSstats and my data log2FC & -log2(adj.pvalue)",
       x = "log2FC",
       y = "-Log2 (adjusted p-value)",
       color = "Protein Regulation") +
  # geom_hline(yintercept = -log2(0.05), linetype = "dashed", linewidth = 0.2,  color = "grey30") +
  # annotate("text", x = .86, y = -log2(0.05), label = "Adj. p-value cutoff (0.05) \n",
  #          fontface = 'italic', size = 4, color = "grey10") +
  # geom_hline(yintercept = -log2(0.1), linetype = "dashed", linewidth = 0.2,  color = "grey50") +
  # annotate("text", x = .86, y = -log2(0.1), label = "Adj. p-value cutoff (0.1) \n",
  #          fontface = 'italic', size = 4, color = "grey30") +
  theme_bw() + 
  theme(legend.position = "bottom")



volcano_adjpv
```

-   <https://gt.rstudio.com/reference/gt.html>

-   <https://gt.rstudio.com/reference/fmt_number.html>

```{r}
hippro_df |>
  gt(
    groupname_col = "Regulation"
    ) |>
  fmt_number(decimals = 3) |>
  # data_color(columns = "Regulation") +
  tab_options(table.font.size=12) 
  # tab_style(
  #   style = cell_fill(color = "lightblue"),
  #   locations = cells_body(rows = log2FC < 0)
  # )
  
```

### Lipidomics - Table S7 - LFQ DEG MSstats - EE vs SH Hippocampus Synaptic Junctions

To add

------------------------------------------------------------------------

## Map 2 uniprot + making lit review table.

Pull uniprot data to map gene ids

See if mice ids match rats \| use predownloaded rda from Uniprot.Ws (otherwise takes forever - see uniprot script).

```{r}
load(file = "input/mouse_ref_proteome_2025_12_13_54864x295.rda")
load(file = "input/rat_ref_proteome_2025_12_13_51872x295.rda")
```

```{r}
# cols I want from uniprot
# my_cols <- c('accession', 'id', 'gene_names', 'gene_primary', 'protein_name',
#              	'xref_proteomes', '	reviewed', 'annotation_score', 'protein_existence',
#              'length', 'organism_name', 'keyword', 'keywordid', 'protein_families',
#              'cc_function', 'cc_developmental_stage', 'cc_tissue_specificity',
#              'xref_geneid')

#colnames(mouse_ref_proteome_2025_12_13)

my_cols <- c("Entry", "Entry.Name", "Protein.names", "Gene.Names..primary.",
             "Gene.Names", "Gene.Names..synonym.",  "Annotation", "Reviewed",
             "Protein.existence", "Protein.families", "Length", "Organism",
             "Keywords", "Keyword.ID", "Function..CC.", "Miscellaneous..CC.", "Pathway",
             "GeneID", "Gene.Ontology..GO.", "Gene.Ontology..cellular.component.",
             "Gene.Ontology..molecular.function.", "Gene.Ontology..biological.process.",
             "Gene.Ontology.IDs", "KEGG", "Reactome", "STRING",
             "Entry.version", "Date.of.creation", "Date.of.last.modification",
             "Date.of.last.sequence.modification"
             )

```

```{r}
mouse_ref <- mouse_ref_proteome_2025_12_13 %>% 
  dplyr::select(all_of(my_cols)) %>%
  mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence)
```

```{r}
# https://www.uniprot.org/help/protein_existence - see rankings 
unique(rat_ref_proteome_2025_12_13$Protein.existence) 


rat_ref <- rat_ref_proteome_2025_12_13 %>% 
  dplyr::select(all_of(my_cols)) %>%
  mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence) #%>%
  #mutate(occupied = rowSums(!is.na(rat_ref))) # some blanks not recognised
  #mutate(occupied = rowSums(!= "")) # recognises blanks better but still some na
```

```{r}
#mapped additional uniprot information by provided uniprot accession
# all 178 mapped 1 to 1
hip_mouse_mapped <- left_join(hippro_df, mouse_ref,
                              by = join_by(Protein == Entry))

# if using borgmeyers genenames then only 72/178 map
# slight difference in new gene entries! 
length(intersect(hip_mouse_mapped$EntryName, hip_mouse_mapped$Gene.Names..primary.))
```

-   <https://stackoverflow.com/questions/44759180/filter-by-multiple-patterns-with-filter-and-str-detect>

```{r}
# # make vector of gene names from Borgmeyer
# genes_hip <- hippro_df$EntryName
# 
# # then filter by matching to any of in the uniprot gene.names column (multiple entries)
# rat_genes_hip <- rat_ref %>%
#   filter(str_detect(Gene.Names,
#                     pattern = paste(genes_hip, collapse = '|'))) 
# 
# length(unique(rat_genes_hip$Gene.Names..primary.)) # 115 primary unique gene entries 
# # BUT NOTE - some are falsely picked up with str_detect e.g. syt1 will get sty11 too.
# # so need to filter & check further!
# 
# mean(rat_genes_hip$Annotation) # 2.69
```

```{r}
# colnames(hip_mouse_mapped)

hip_mouse_mapped_2 <- hip_mouse_mapped %>%
  dplyr::select(Protein:Gene.Names) %>% #1:14
  rename(entry.name.mouse = Entry.Name, # new = old
         protein.names.mouse = Protein.names,
         gene.names.primary.mouse = Gene.Names..primary.,
         gene.names.mouse =  Gene.Names)

# length(unique(hip_mouse_mapped_2$gene.names.primary.mouse)) #178/178!
```

```{r}
# mapped by primary genenames 
rat_hip_map_1 <- left_join(hip_mouse_mapped_2, rat_ref,
                           by = join_by( # 606 rows
                             gene.names.primary.mouse == Gene.Names..primary.)) %>%
  # trim multi entries based off of best match e.g. swiss prot > highest anno > evidence
  group_by(gene.names.primary.mouse) %>% # 178 from Borgmeyer
  filter(if(any(Reviewed == "reviewed", na.rm = TRUE)) {
    Reviewed == "reviewed"
  } else {TRUE}) %>% # 256 rows
  slice_max(Annotation, n = 1) %>% # 219
  slice_max(existence_ranking, n = 1) # 204

```

```{r}
# see duplicates | ~ 13 entries with 2 or more mappings
remaining_multi_1 <- rat_hip_map_1 %>%
   count(EntryName) %>%
   arrange(desc(n))
```

```{r}
# # unsuccessfully mapped. | 5 of 178 unmapped
rat_hip_map_1_na <- rat_hip_map_1 %>%
  filter(is.na(Entry))
```

### Now try to map the remaining 5 entries 

```{r}
rat_ref_split <- rat_ref %>%
  mutate(split_synonyms = str_split(Gene.Names..synonym., " ")) %>% #splitby space now ,
  tidyr::unnest(split_synonyms)

```

```{r}
rat_hip_map_2 <- rat_hip_map_1_na %>%
  dplyr::select(Protein:gene.names.mouse) %>%
  left_join(rat_ref_split, # this recoveres 2 of the 5 missing
            by = join_by(gene.names.primary.mouse == split_synonyms)) 

# also Hbb-b1 but not in reference proteome hence unmapped. 
# maybe use non reference proteome too??? idk # none of the ciritical ones were missed
# so not a big deal
```

------------------------------------------------------------------------

```{r}
# final combined verison.
rat_hip_map_3 <- bind_rows(list(by_primary_gene = rat_hip_map_1,
                                by_synonym = rat_hip_map_2),
                           .id = "map_method_df") %>%
  arrange(desc(Regulation), gene.names.primary.mouse)
```

------------------------------------------------------------------------

### Lit review table

```{r}
# colnames(com_df_w_ids) # check col names 


# for lit review table | Note - duplicates have not been removed - to do manually
lit_re <- rat_hip_map_3 %>%
  mutate(
    Year = 2021,
    Author = 'Borgmeyer',
    Animal = 'Mice',
    Condition = "Fraction - Synaptic Junction",
    Region = 'Hippocampus',
    Whole = 'No',
    Exp_type = 'Protein',
    Method = 'PROTEOMICS_LCMS_LFQ_OS',
    Used_name = `Protein Name`,
    Mod = " ",
    Mod_type = " ",
    Gene_name = `gene.names.primary.mouse`, # this was used to map the primary rat gene names
    Comparison = 2,
    In_EE = Regulation,
    Fold_change = log2FC,
    Arrows = case_when( # dplyr case when, instead of ifelse
      log2FC > 2.5 ~ '+++',
      log2FC > 1 ~ '++',
      log2FC > .05 ~ '+',
      log2FC > -.05 ~ '=',
      log2FC > -1 ~ '-',
      log2FC > -2.5 ~ '--',
      log2FC <= -2.5 ~ '---',
      ),
    Significant = ifelse(adj.pvalue <= 0.05, 'yes', 'no'),
    pvalue = adj.pvalue,
    N_tested = ' ',
    Stats_adj = 'adjusted',
    Note = ' ',
    Full_name = Protein.names,
    Uniprot_rat_accession = Entry,
    Uniprot_entry_name = Entry.Name,
    Uniprot_link = ifelse(!is.na(Entry),
                          paste0('https://www.uniprot.org/uniprotkb/', Entry,'/entry'),
                          ' '),
    Wikipedia_link = ifelse(!is.na(Entry),
                            paste0('https://en.wikipedia.org/wiki/',
                                   `gene.names.primary.mouse`),
                            ' '),
    Curation_method = case_when(
      Reviewed == "reviewed" & map_method_df == "by_primary_gene" ~ "auto - reviewed (primary gene name)",
      Reviewed == "reviewed" & map_method_df == "by_synonym" ~ "auto - reviewed (synonym)",
      Reviewed == "unreviewed" & map_method_df == "by_primary_gene" ~ "auto - ref proteome (primary gene name)",
      Reviewed == "unreviewed" & map_method_df == "by_synonym" ~ "auto - ref proteome (synonym)",
      is.na(Entry.Name) ~ "manual - na" 
    ),
    Class_group = ' ',
    Other_link = ' ',
    Other_comments = 'misc shows combined provided info from their df',
    misc = paste0(Protein, '_', EntryName,  `Protein Name`)
  )
```

```{r}
lit_re %>%
  count(Curation_method) %>%
  arrange(desc(n))
```

```{r}
# #write out | na values = blank
# write_excel_csv(lit_re,
#                 "output/2025_12_15_Borgmeyer_2021_Hip_SJ_proteomics_lit_re_rats_v1.csv",
#                 na = "")
```
