---
title: "Lichti_2014_proteomics_data"
output: html_document
---

# Lichti et al., 2014 (Green Group) LC-MS Proteomics Re-analysis (Main effect of housing - EE vs IC rats NAc)

*from 'Environmental enrichment alters protein expression as well as the proteomic response to cocaine in rat nucleus accumbens' in [frontiers in behavioral neuroscience] <https://doi.org/10.3389/fnbeh.2014.00246>*

*Also see Zhang et al., 2016 from Green Group <https://doi.org/10.1016/j.neuroscience.2016.09.051>*

-   [*UNCLEAR 1:*]{.underline} *Not quite sure if their p-values are adjusted or not? The paper mentions using BH as adjustment method? But they only report p-value in the paper & in their column headers are \`EC p value\` - so not sure?*
    -   *They use \>1.5 fold & \<0.05 cut off, which makes me think they did not use adjusted p-values*
-   [*UNCLEAR 2:*]{.underline} *Also, not quite sure how they did their GSEA as they did not do it separately for their 3 fractions (only present one result).*
    -   *I have done it separately for each fraction here.*

[***Contents***]{.underline}

-   *Mapped to more unirprot data*
-   *Basic plots (scatter, lollipop)*
-   *gt table of significant proteins (rat only)*
-   *lit review table (not shown in knit)*
-   *For loop KEGG gsea (clusterprofiler)*
-   *Check of table 1 from paper*
-   *Nested for loop GO gsea*
    -   *Works on SBS Jennifer server but not laptop (insufficient memory? -\> look into?)*

### Load Packages

```{r message=FALSE, warning=FALSE}
library(readxl) # read_excel()
library(tidyverse) # dplyr, stringr, ggplot2, readr, tidyr
library(gt) # for gt tables
library(UniProt.ws) # 1st test -  uniprot api 
library(org.Rn.eg.db) # Rat genome annotation package - incl. AnnotationDbi
library(clusterProfiler) #gsea
library(enrichplot) # for dotplot()
library(pathview) # KEGG pathview 
library(colorspace) # loliplot colors

# library(pathview) # commented out atm | (but for visual kegg mapping)
# library(msigdbr) # commented out atm | MSigDB gene sets 
# # note some packages called directly instead e.g. cowplot:: & ggrepel::
```

### Load initial Lichti df table

```{r}
df_1 <- read_excel("input/Lichti_2014_datasheet_1.XLSX",
                   sheet = "quant-all",
                   trim_ws = TRUE) # does not trim white space behind accessions?

```

```{r paged.print=TRUE}
colnames(df_1)
str(df_1)
length(unique(df_1$Accession)) # 1915
head(df_1, 10)
tail(df_1, 10)
```

### Get extra uniprot in via uniprot.ws

Getting info like protein names, etc

```{r}
ratUp <- UniProt.ws(10116) # rat
```

```{r eval=FALSE, include=FALSE}
keytypes(ratUp) # key types that can be retrieved

columns(ratUp) # columns that can be retrieved
```

```{r}
my_ids <- df_1$Accession
```

```{r}
# https://www.uniprot.org/help/return_fields
# https://www.uniprot.org/help/general_annotation
# takes the rat & mouse uniprot accession IDs in Lichti's df & maps further uniprot info onto them
up_rat_mouse <- mapUniProt(
  from = "UniProtKB_AC-ID",
  to = "UniProtKB",
  columns = c("accession", "id", "reviewed", "protein_name", "gene_names",
              "gene_primary", "organism_name", "length", "keywordid", "keyword", "xref_geneid"),
  query = my_ids
  #query = list(taxId = 10116, ids = my_ids) # isn't selective to only rats?
)
```

```{r}
# add website links
# follows a certain stucture (doesn't account all but has most)
up_rat_mouse_entry <- up_rat_mouse %>%
  mutate(uniprot_link = paste0("https://www.uniprot.org/uniprotkb/", Entry, "/entry")) %>% # add uniprot entry link! 
  mutate(wiki_link = paste0("https://en.wikipedia.org/wiki/", str_to_upper(Gene.Names..primary.))) # add wikipedia (needs 2 be uppercase)
  #most of the wiki ones work e.g. in if 4 or more characters
```

```{r}
# write out basic df
# write_excel_csv(up_rat_mouse_entry, "output/2025_12_08_lichti_2014_test_map_v1.csv")
```

### Bind uniprot info & df + add other info & clean up data

```{r}
df_2 <- dplyr::left_join(df_1, up_rat_mouse_entry,
                         by = join_by(Accession == From)) %>%
  mutate(Direction = ifelse(`EC log2 fold` < 0, "Down", "Up"),
         gene_name = str_to_title(`Gene.Names..primary.`),
         .after = `Gene Symbol`) %>%
  mutate(Significant = ifelse(`EC p value` <= 0.05, 'yes', 'no'), .after = `EC p value`)  %>%
  filter(str_detect(Name, "RAT")) # 2835 to 1506
```

```{r}
df_3 <- df_2 %>%
  dplyr::select(Accession:Significant, fraction:wiki_link) %>% 
  mutate(across(where(is.numeric), ~ round(., 3))) %>% #if numeric col, round to 3 dp
  group_by(fraction, Direction) %>%  
  arrange(`EC p value`, `EC log2 fold`) 

```

### Plots

#### ggplot rat only non sig dot plot

```{r fig.height=5, fig.width=11.5}
# non significant plot - looks messy but you get the idea
ns_plot <- df_3 %>%
  ggplot(aes(x = gene_name, y = `EC log2 fold`, color = Direction, alpha = Significant)) +
  geom_point(shape = 1) +
  geom_hline(yintercept = 0, color = 'grey20') +
  scale_color_manual(breaks = c("Up", "Down"),
                     values = c("#dc322f", "#268bd2")) + 
  scale_alpha_manual(values = c(0.2,1)) +
  labs(title = 'Lichti et al., 2014 - EC log2fc x gene name - NAc',
       subtitle = 'Rats only (1506/2835), non-sig incl., split by fraction, LC-MS proteomics') +
  facet_wrap(~fraction, space = 'free_x', scale = 'free_x') + 
  theme_bw() 

ns_plot
```

#### ggplot lollipop chart (sig only)

```{r fig.height=5, fig.width=11.5}
# assign colors based on direction. | works by nvm ordering is weird
#label_colors <- ifelse(df_3$`EC log2 fold` > 0 , "#dc322f", "#268bd2")

# lollipop - ggplot ver - a big ugly
sig_plot <- df_3 %>%
  filter(`EC p value` <= 0.05) %>% # 139
  ggplot(aes(x = reorder(gene_name, `EC log2 fold`), # orders by log fold
                         y = `EC log2 fold`,
             color = Direction)) +
  geom_segment(aes(x = gene_name,
                   yend = `EC log2 fold`, y = 0)) +
  geom_point(aes(alpha = `EC p value`),
             shape = 19, size = 4) +
  scale_alpha(range = c(1,.1), guide = 'none') + # rev alpha so darker = more significant
  # geom_text(aes(label = gene_name), color = "grey10", size = 2.5,
  #           position = position_stack()) +
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  labs(title = 'Lichti et al., 2014 - EC Log2FC x Gene Name - NAc',
       subtitle = 'Rats only (139/1506/2835), sig only, by fraction, LC-MS proteomics',
       y = 'EC Log2FC',
       x = 'Gene Name') +
  #coord_flip() + #fliptheplot
  facet_wrap(~fraction, space = 'free_x', scale = 'free_x') +
  #theme_void() +
  theme_bw() +
  theme(
    #axis.title = element_blank(),
    #axis.text.x = element_blank() 
    axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 9, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20", size = 10), # rota 90 & hjust=1 (top/right)
        ) 

sig_plot
```

#### ggplot volcano

```{r fig.height=18, fig.width=12}
volcano_adjpv <- ggplot(df_3, 
                     aes(x =`EC log2 fold`,
                         y = -log2(`EC p value`), label = gene_name)) +
 geom_point(aes(color = case_when(
   `EC p value` < 0.05 & `EC log2 fold` > 0 ~ "Upregulated (p<0.05)",
   `EC p value` < 0.05 & `EC log2 fold` < 0 ~ "Downregulated (p<0.05)",
   `EC p value` < 0.1 & `EC log2 fold` > 0 ~ "Upregulated (p<0.1)",
   `EC p value` < 0.1 & `EC log2 fold` < 0 ~ "Downregulated (p<0.1)",
   TRUE ~ "Not Significant")),
   alpha = 1, size = 2.5, shape = 1) +
  xlim(-7.5,7.5) +
  geom_vline(xintercept = -1.5, linewidth =0.2, linetype = "dashed", color = "grey50") +
  geom_vline(xintercept = 1.5, linewidth =0.2, linetype = "dashed", color = "grey50") +
  geom_hline(yintercept = -log2(0.05), linetype = "dashed", linewidth = 0.2,  color = "grey30") +
  scale_color_manual(values = c("firebrick3", 
                                "royalblue4", 
                                "lightpink1", 
                                "lightblue3" , 
                                 "grey80"),
                     breaks = c("Upregulated (p<0.05)", 
                                "Downregulated (p<0.05)", 
                                "Upregulated (p<0.1)", 
                                "Downregulated (p<0.1)", 
                                "Not Significant"),
                     labels = c("Upregulated Proteins (up to p<0.05)", 
                                "Downregulated Proteins (up to p<0.05)", 
                                "Upregulated Proteins (up to p<0.1)", 
                                "Downregulated Proteins (up to p<0.1)" , 
                                "Not Significant")) +
  
  ggrepel::geom_text_repel() +
  facet_wrap(~ fraction, ncol = 1, scales = "free_y") +
  labs(
    title = "Lichti et al., 2014 LC-MS proteomics NAc rats - main effect of housing (EE vs IC) (2025_12_26)",
       subtitle = "Split by fraction | Volcano plots made from their provided df, not 100% sure on their p-value if adj or not?",
       x = "log2FC",
       y = "-Log2 (p-value)",
       color = "Protein Regulation") +
  geom_hline(yintercept = -log2(0.05), linetype = "dashed", linewidth = 0.2,  color = "grey30") +
  # annotate("text", x = .86, y = -log2(0.05), label = "Adj. p-value cutoff (0.05) \n",
  #          fontface = 'italic', size = 4, color = "grey10") +
  # geom_hline(yintercept = -log2(0.1), linetype = "dashed", linewidth = 0.2,  color = "grey50") +
  # annotate("text", x = .86, y = -log2(0.1), label = "Adj. p-value cutoff (0.1) \n",
  #          fontface = 'italic', size = 4, color = "grey30") +
  theme_bw() + 
  theme(legend.position = "bottom",
        strip.text = element_text(size = 12, face = "bold", color = "grey10")
        )


volcano_adjpv
```

### gt Table

gt table for sig proteins only

```{r}
df_3 |>
  dplyr::select(fraction, gene_name, `Protein.names`, Direction,
                `EC log2 fold`, `EC p value`, Entry) |> #cols for table
  dplyr::filter(`EC p value` <= 0.05) |>
  gt(
    groupname_col = dplyr::group_vars(df_3)
  ) |>
  tab_header(
    title = "Lichti et al., 2014 - EC vs. IC - NAc",
    subtitle = "Cleaned lc-ms proteomics dataframe, signficant rats only (139/1506/2835
    ), organised by fraction + direction & arranged by pvalue + EC log2FC"
  ) |>
   opt_stylize(style = 1, color = 'gray', add_row_striping = TRUE)
```

------------------------------------------------------------------------

# Bioinformatics

## old nooby 1 by 1 method (commented out)

### Ranked List

already partially prepped from above

```{r}
# # ranked combined list
# geneList_df <- df_3 %>%
#   mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
#   group_by(Entry) %>%
#   summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
#   arrange(-ranking)
# 
# # convert 2 a named vector to supply to fgsea()
# geneList <- tibble::deframe(geneList_df) # 955
# 
# plot(geneList)
```

```{r}
# # ranked cytosolic list
# geneList_df_c <- df_3 %>%
#   filter(fraction == 'cytosolic') %>%
#   mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
#   group_by(Entry) %>%
#   summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
#   arrange(-ranking)
# 
# # convert 2 a named vector to supply to fgsea()
# geneList_c <- tibble::deframe(geneList_df_c) # 689
# 
# plot(geneList_c)
```

```{r}
# # ranked membrane list
# geneList_df_m <- df_3 %>%
#   filter(fraction == 'membrane') %>%
#   mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
#   group_by(Entry) %>%
#   summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
#   arrange(-ranking)
# 
# # convert 2 a named vector to supply to fgsea()
# geneList_m <- tibble::deframe(geneList_df_m) # 340
# 
# plot(geneList_m)
```

```{r}
# # ranked nuclear list
# geneList_df_n <- df_3 %>%
#   filter(fraction == 'nuclear') %>%
#   mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
#   group_by(Entry) %>%
#   summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
#   arrange(-ranking)
# 
# # convert 2 a named vector to supply to fgsea()
# geneList_n <- tibble::deframe(geneList_df_n) # 477
# 
# plot(geneList_n)
```

### KEGG

#### Nuclear

```{r}
# # nuclear
# kegg_res_n <- gseKEGG(geneList = geneList_n,
#                     organism = 'rno',
#                     keyType = 'uniprot',
#                     by = 'fgsea',
#                     pvalueCutoff = 0.05,
#                     pAdjustMethod = 'BH', #BejaminiHoc Correciton
#                     )
# 
# gse_kegg_tib_n <- as_tibble(kegg_res_n) # 21 sig
```

```{r}
# enrichplot::dotplot(kegg_res_n,
#                     split = ".sign",  # split by + vs - enrirchment score
#                     showCategory = 30,
#                     title = "Nuclear - NAc - Lichti et al., 2014 (EE vs IC) - gseKEGG_r - adj.p<0.05") +
#   facet_grid(.~.sign) # split plot by seign)
```

#### Cytosolic

```{r}
# # cytosolic
# kegg_res_c <- gseKEGG(geneList = geneList_c,
#                     organism = 'rno',
#                     keyType = 'uniprot',
#                     by = 'fgsea',
#                     pvalueCutoff = 0.05,
#                     pAdjustMethod = 'BH', #BejaminiHoc Correciton
#                     )
# 
# gse_kegg_tib_c <- as_tibble(kegg_res_c) # 21 sig
```

```{r}
# enrichplot::dotplot(kegg_res_c,
#                     split = ".sign",  # split by + vs - enrirchment score
#                     showCategory = 30,
#                     title = "Cytosolic - NAc - Lichti et al., 2014 (EE vs IC) - gseKEGG_r - adj.p<0.05") +
#   facet_grid(.~.sign) # split plot by seign)
```

#### Membrane

```{r}
# # membrane
# kegg_res_m <- gseKEGG(geneList = geneList_m,
#                     organism = 'rno',
#                     keyType = 'uniprot',
#                     by = 'fgsea',
#                     pvalueCutoff = 0.05,
#                     pAdjustMethod = 'BH', #BejaminiHoc Correciton
#                     )
# 
# gse_kegg_tib_m <- as_tibble(kegg_res_m) # 3 sig
```

```{r}
# enrichplot::dotplot(kegg_res_m,
#                     split = ".sign",  # split by + vs - enrirchment score
#                     showCategory = 30,
#                     title = "Membrane - NAc - Lichti et al., 2014 (EE vs IC) - gseKEGG_r - adj.p<0.05") +
#   facet_grid(.~.sign) # split plot by seign)
```

```{r}
# # oxidative phosphorylation - rno00190
# rno00190 <- pathview(gene.data = geneList_n,
#                      gene.idtype = "SYMBOL",
#                      pathway.id = 'rno00190', # oxidative phosphorylation - Rat
#                      species = 'rno',
#                      limit = list(gene=max(abs(geneList)), cpd=1))

```

------------------------------------------------------------------------

## Trying for loops for gsea (KEGG)

-   Based off of <https://www.epirhandbook.com/en/new_pages/iteration.html#looping-plots> & my previous clusterprofiler analyses

### Setup containers & prepare info for sequences

```{r}
# make vector of fractions (cytosolic, membrane, & nuclear) - in prep of sequences
fraction_names <- unique(df_3$fraction)

# make container lists to store results
# gsea_results_list <- vector(mode = "list", length = 10)
gsea_results_list <- vector(mode = "list") # actually make empty
```

### Operation

```{r}
for (fr in fraction_names){ # SEQUENCE - for each item (fr) in fraction_names
  
  # OPERATIONS
  # Part 1
  # make a ranked gene list for this fraciton
  geneList_working_df <- df_3 %>%
      # filter by item | fraction is column in df_3, fr is i (current item in seq)
    filter(fraction == fr) %>% 
      # compute ranking metric
    mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
      # for any duplicate entries, # average & remove any nas
    group_by(Entry) %>% 
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
      # lastly, arrange from pos to neg 
    arrange(-ranking)
  
  # store in list (not necessary...)
  gsea_results_list$geneList_df[[fr]] <- geneList_working_df
  
  # Part 1.2
  # next, deframe into a named vector needed to supply to fgsea()
  geneList_vec_working <- tibble::deframe(geneList_working_df)
  
  # store in list 
  gsea_results_list$geneList_vec[[fr]] <- geneList_vec_working
  
  # Part 1.3 | plots fine but does not save?!?!?!
  # Plot ranked gene list
  gL_plot <- plot(geneList_vec_working, main = paste(fr, '- Ranked gene list'))
  print(gL_plot)

  gsea_results_list$geneList_plots[[fr]] <- gL_plot
  
 
  # Part 2
  # run KEGG gsea with clusterProfiler
  kegg_res_working <- clusterProfiler::gseKEGG(geneList = geneList_vec_working,
                              organism = 'rno',
                              keyType = 'uniprot',
                              by = 'fgsea',
                              pvalueCutoff = 0.05,
                              pAdjustMethod = 'BH', #BejaminiHoc Correciton
                              )
  
  # save kegg result original to list vec
  gsea_results_list$kegg_results[[fr]] <- kegg_res_working
  
  # not needed here so compute & save directly? or can do anyway? idk?
  gsea_results_list$kegg_results_tib[[fr]] <- tibble::as_tibble(kegg_res_working) 
  
  
  # Part 3
  # Make dotplots | this one plots & saves properly!
  # 3.1 - regular dot blot
  dp <- enrichplot::dotplot(kegg_res_working,
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              '{fr} - NAc - Lichti, 2014 (EE vs IC) - gseKEGG_r'))) 

  
  print(dp)   # print dot plots 
  gsea_results_list$dotplots[[fr]] <- dp # save to list 
  
  # 3.2 - dot plot split by sign
  dp_sign <- enrichplot::dotplot(kegg_res_working,
                            split = '.sign',
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              '{fr} - NAc - Lichti, 2014 (EE vs IC) - gseKEGG_r split by sign'))) +
    facet_grid(.~.sign)
  
  print(dp_sign) # print dot plots 
  gsea_results_list$dotplots_sign[[fr]] <- dp_sign # save to list

    
}
```

```{r}
# horrific name but at least clear...
#Lichti_2014_gsea_results_2025_12_12_v1 <- gsea_results_list 

# save(Lichti_2014_gsea_results_2025_12_12_v1, 
#      file = "output/2025_12_12_Lichti_2014_gsea_KEGG_results_rats_NAc_v1.rda")
```

```{r}
# Warning means that some genes have tied rankings
# note the membrane one is especially high 89.71%  - this corresponds to the weird # pattern in geneList ranking plot -  

# Warning messages:
# 1: In preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam,  :
#   There are ties in the preranked stats (33.82% of the list).
# The order of those tied genes will be arbitrary, which may produce unexpected results.
# 2: In preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam,  :
#   There are ties in the preranked stats (30.61% of the list).
# The order of those tied genes will be arbitrary, which may produce unexpected results.
# 3: In fgseaMultilevel(pathways = pathways, stats = stats, minSize = minSize,  :
#   For some of the pathways the P-values were likely overestimated. For such pathways log2err is set to NA.
# 4: In fgseaMultilevel(pathways = pathways, stats = stats, minSize = minSize,  :
#   For some pathways, in reality P-values are less than 1e-10. You can set the `eps` argument to zero for better estimation.
# 5: In preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam,  :
#   There are ties in the preranked stats (89.71% of the list).
# The order of those tied genes will be arbitrary, which may produce unexpected results.
```

```{r}
check_1 <- gsea_results_list$geneList_df$cytosolic

length(unique(gsea_results_list$geneList_df$cytosolic$Entry)) # 689 unique proteins
length(unique(gsea_results_list$geneList_df$cytosolic$ranking)) # 456 unique ranks
```

```{r fig.height=20, fig.width=12}
# cow plot for combined + plotlist = to plot a list of plots 
cowplot::plot_grid(plotlist = c(gsea_results_list$dotplots,
                                gsea_results_list$dotplots_sign),
                   ncol = 2)
# print(gsea_results_list$dotplots)
```

```{r}
# weird... highlights importance of plotting data!!!
plot(df_3$`EC log2 fold`)
```

[2025_12_12 - Note to self maybe add some testing & progress of loop info - ref to epi cook book. Also try nested for loops with GO analysis e.g. for each of the 3 fractions (cytosolic, nuclear, membrane) run each of the 3 GO (CC, BP, MF)]{.smallcaps}

------------------------------------------------------------------------

# Literature Review Table

### Make lit review table (similar to McNair one)

```{r}
colnames(df_3) # check col names 

# lit review table 
lit_re <- df_3 %>%
  mutate(
    Space = " ", # empty space column
    Year = 2014,
    Author = 'Lichti',
    Animal = 'Rat',
    Condition = paste0("Fraction - ", fraction),
    Region = 'Nucleus_Accumbens',
    Whole = 'No',
    Exp_type = 'Protein',
    Method = 'PROTEOMICS_LCMS_LFQ_OS',
    Used_name = `Gene Symbol`,
    Mod = " ",
    Mod_type = " ",
    Gene_name = gene_name, # lazy way to avoid moving rows (don't need headers anyway)
    Comparison = 1,
    In_EE = Direction, # note was coded as >0 Up, <0 Down. Maybe change one to same?? 
    Fold_change = `EC log2 fold`,
    #Arrows = ifelse(Direction == 'Up', '+', '-'),
    Arrows = case_when( # dplyr case when, instead of ifelse
      `EC log2 fold` > 2.5 ~ '+++',
      `EC log2 fold` > 1 ~ '++',
      `EC log2 fold` > .05 ~ '+',
      `EC log2 fold` > -.05 ~ '=',
      `EC log2 fold` > -1 ~ '-',
      `EC log2 fold` > -2.5 ~ '--',
      `EC log2 fold` <= -2.5 ~ '---',
      ),
    #Significant = ifelse(`EC p value` <= 0.05, 'yes', 'no'),
    signficant = Significant,
    pvalue = `EC p value`,
    N_tested = " ",
    Stats_adj = 'adjusted',
    Note = ' ',
    Full_name = Protein.names,
    Uniprot_rat_accession = Entry,
    Uniprot_entry_name = Entry.Name,
    Uniprot_link = uniprot_link,
    Wikipedia_link = wiki_link,
    Curation_method = ifelse(!is.na(Entry), 'auto - uniprot_acc_query', 'manual'),
    Class_group = ' ',
    Other_link = ' ',
    Other_comments = '*misc col shows the col names in their excel df',
    misc = paste0(Accession, ";", `Gene Symbol`, ";", Name)
  )
```

#### Export

```{r}
#write out | na values = blank 
# write_excel_csv(lit_re,
#                 "output/2025_12_12_Lichti_2014_rats_NAc_data_lit_review_v1.csv",
#                 na = "")
```

------------------------------------------------------------------------

## Check of table 1 from paper (selected sig proteins)

2025_12_24 - while most of the reanalysis matches the results presented in the paper. some did not quite. I suspected this was due to using the mouse database (which i excluded from my analysis). The code below is checking that.

```{r}
# list of EE vs IC sig proteins shown in table 1 of the paper | 15 genes/proteins
acc_2_check <- c('P61751', 'Q64611',  'P24268', 'Q641Y8', 'Q63622', 'Q6PDL0',
                 'Q8BGY2', 'O54921', 'P28741', 'P21396', 'P36506', 'P10637',
                 'Q80ZA5', 'P07895', 'Q60930')

diff_check <- df_1 %>% 
  filter(Accession %in% acc_2_check)
```

```{r paged.print=TRUE}
head(diff_check, 13)
tail(diff_check, 13)
```

------------------------------------------------------------------------

# Bioinformatics resumed

*2025_12_25/26 onwards*

## Running GO term GSEA with nested for loops

*Currently only works on SBS Jennifer (insufficient memory on laptop?)*

-   <https://www.epirhandbook.com/en/new_pages/iteration.html#looping-plots>
-   <https://yulab-smu.top/biomedical-knowledge-mining-book/021-go.html#go-gene-set-enrichment-analysis>
-   See earlier bit of code (above for KEGG) (basic for loop) & (just subset as `[[fr]][[go]]` for nested)

[*Previous Errors*]{.underline}

-   ~~*Error in serialize(data, node\$con) :… … error writing to connection*~~
    -   *Fixed (kinda?) - Fine on SBS Jennifer server, issues is just on laptop*
-   ~~*Error in \`\$\<-.data.frame\`(\`\*tmp\*\`, ".sign", value = "activated") : replacement has 1 row, data has 0*~~
    -   *Fixed with if else for dotplots to only run if not empty (error was due to some GO results being empty due to no significant findings)*

```{r}
# not needed - works fine with uniprot entries
# # make trimmed entrez_id (rm everything after semi colon)
# df_4 <- df_3 %>%
#   ungroup() %>%
#   mutate(entrez_id = str_replace(
#     GeneID, pattern = ";.*", replacement = ""))
```

### Setup containers & prepare info for sequences

```{r}
# make vector of fractions (cytolsolic, membrane, & nuclear)
fraction_names <- unique(df_3$fraction)

# GO terms
GO_names <- c('BP', 'CC', 'MF', 'ALL')

# make container list to store results | empty for lists
gsea_GO_results_lists <- vector(mode = "list") 
```

### Operation

1.  **For loop** (for each `fr` fraction), compute a ranked genelist...

2.  Then **nested for loop** (for each `go` GO term),

    1.  **if**, `fr` `go` result is not empty (\> 0) then, make dot plot

    2.  **else**, print message no result.

```{r fig.height=11, fig.width=10}
# works on SBS jennifer
for (fr in fraction_names){ # SEQUENCE - for each item (fr) in fraction_names
  
  # OPERATIONS
  # PART 1
  # make a ranked gene list for this fraction 
  geneList_working_df_go <- df_3 %>% # rats only mapped df from Lichti proteomics
    # filter by item | fraction is column in df_3, fr is i (current item in seq)

    filter(fraction == fr) %>%
    
    # compute ranking metric 
    mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
    # for any duplicate entries, # average & remove any nas
    # group_by(Gene.Names..primary.) %>%
    group_by(Entry) %>%
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
    # lastly, arrange from largest positive to most negative
    arrange(-ranking)
  
  # Part 1.2
  # next, deframe into a named vector needed to supply to fgsea()
  geneList_vec_working_go <- tibble::deframe(geneList_working_df_go)
  
geneList_vec_working <- geneList_vec_working_go[!is.na(names(geneList_vec_working_go))]
  
  # store in list 
  gsea_GO_results_lists$geneList_vec[[fr]] <- geneList_vec_working_go
  
  # Part 1.3 | plots fine but does not save?
  # Plot ranked gene List
  gL_plot_go <- plot(geneList_vec_working_go, main = paste0(fr, '- Ranked gene list'))
  
  print(gL_plot_go) 
  
  gsea_GO_results_lists$geneList_plots[[fr]] <- gL_plot_go
  
  
  # PART 2 
  # nested for loop
  # so now for each fraction, go through all 3 go terms (BP, CC, MF)
  for (go in GO_names){ 
    
    # operation
    # run gseGO for each go term 
    GO_res_working <- clusterProfiler::gseGO(geneList = geneList_vec_working_go,
                                             OrgDb = org.Rn.eg.db,
                                             keyType = "UNIPROT",
                                             ont = go, # 
                                             minGSSize = 10,
                                             maxGSSize = 500,
                                             pvalueCutoff = 0.05,
                                             pAdjustMethod = "BH"
                                             )
    
  # save GO result original to list vec
  gsea_GO_results_lists$GO_results[[fr]][[go]] <- GO_res_working
    
    
  # extra (not really needed but save anyway
  gsea_GO_results_lists$GO_results_tib[[fr]][[go]] <- tibble::as_tibble(GO_res_working)
    

  # PART 3 - if else for dot plot
  if (nrow(GO_res_working@result) > 0 ) {
    
    #Make dot plots | this one saves properly
    # Part 3.1 - regular dot plots
    dp_go <- enrichplot::dotplot(GO_res_working,
                              showCategory = 30,
                              font.size = 11,
                              title = paste0(stringr::str_glue(
                                '{fr} - {go} - NAc - Lichti 2014 EE vs IC - gseGO'
                              )))
    print(dp_go) # print dot plots
    gsea_GO_results_lists$dotplots[[fr]][[go]] <- dp_go # save to list
    
  } else{print(paste0("No GO gsea result for dot plot in ", fr, "-", go, " :("))}
  
  } 
  
}
```

```{r}
# # horrific name but at least clear...
# Lichti_2014_gsea_GO_results_2025_12_26_v1 <- gsea_GO_results_lists
# 
# save(Lichti_2014_gsea_GO_results_2025_12_26_v1,
#      file = "output/2025_12_26_Lichti_2014_gsea_GO_results_rats_NAc_v1.rda")
```

```{r paged.print=FALSE}
# cow plot for combined + plotlist = to plot a list of plots 
# cowplot::plot_grid(plotlist = c(gsea_GO_results_lists$dotplots),
#                    ncol = 2)

# cow plot not working 
# Warning message:
# In as_grob.default(plot) : Cannot convert object of class list into a grob.


# # this works 
# print(gsea_GO_results_lists$dotplots)
```

```{r}
sessionInfo()
```

------------------------------------------------------------------------

## Part 2 -  read in & comp

2026_02_16 - read in results & process for

```{r}
# GO 
load("output/2025_12_26_Lichti_2014_gsea_GO_results_rats_NAc_v1.rda")

# KEGG 
load("output/2025_12_12_Lichti_2014_gsea_results_rats_NAc_v1.rda")

# lit re tab
lit_re <- read_csv("output/2025_12_12_Lichti_2014_rats_NAc_data_lit_review_v1.csv")
```

```{r}
# MAP KEGG into single 
KEGG_res_lichti_sep <- map_dfr(
  Lichti_2014_gsea_results_2025_12_12_v1$kegg_results_tib,
  as_tibble, .id = "fraction")

```

```{r}
#lichti gene lists
lichti_genelist_sep <- map_dfr(Lichti_2014_gsea_results_2025_12_12_v1$geneList_vec,
                           ~enframe(.x),
                           .id = "fraction")
```

### Combined KEGG 

```{r}
lichti_genelist_com <- lit_re %>%
  mutate(ranking = sign(`EC log2 fold`)*(-log10(`EC p value`))) %>%
   group_by(Entry) %>%
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
    arrange(-ranking)

# working gL for gsea
lichti_genelist_com_wr <- deframe(lichti_genelist_com)
```

```{r}
plot(lichti_genelist_com$ranking)
plot(lichti_genelist_sep$value)
```

```{r}
Lichti_kegg_res_com <- clusterProfiler::gseKEGG(geneList = lichti_genelist_com_wr,
                              organism = 'rno',
                              keyType = 'uniprot',
                              by = 'fgsea',
                              pvalueCutoff = 0.05,
                              pAdjustMethod = 'BH', #BejaminiHoc Correciton
                              )
```

```{r}
Lichti_kegg_res_com_tib <- as_tibble(Lichti_kegg_res_com)
```

```{r}
# save(Lichti_kegg_res_com, file = "output/2026_02_16_Lichti_2014_combined_gL_KEGG_gsea_res.rda")
```

```{r}
# lichti_loli_c <- Lichti_kegg_res_com_tib %>%
lichti_loli_f <- KEGG_res_lichti_sep %>%  
  
  
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     guide = 'none') + 
  scale_size_continuous(range = c(3,9)) + # map size oflolly (set size)
  
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  labs(title = 'EE vs IC - KEGG GSEA (Lichti et al., 2014 - NAc rats)',
       # subtitle = 'Combined fractions into one ranked gene list', # combined
       subtitle = 'Fractions kept separate', # left as separate
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  facet_wrap(~fraction, space = 'free_x', scale = 'free_x') +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 9, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 9, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none",
    title = element_text(size = 10)
        ) 

lichti_loli_c
lichti_loli_f
```

```{r}
# side by side - shows that fraction = sig but not when combined
cowplot::plot_grid(lichti_loli_f, lichti_loli_c, ncol =1,
                   rel_widths = c(1, 0.15), rel_heights = c(1, 0.15))
```

#### Pathview - using a separate gene list with nuclear is fraction (where the significant differences are)

-   rno00190 = oxidative phosphorylation
-   rno00020 - TCA cycle
-   rno00010 - Glycolysis/Gluconeogenesis
-   rno05022 - pathways of neurodegeneration - multiple diseases
-   rno03010 - ribosome
-   rno05012 - parkinsons

```{r}
# entrez_gl <- bitr(lichti_genelist_sep$name, 
#      fromType="UNIPROT", toType="ENTREZID", OrgDb="org.Rn.eg.db")
# 
# entrez_gl_map <- left_join(lichti_genelist_sep, entrez_gl, 
#                            by =join_by(name == UNIPROT))

entrez_gl_map_n <- entrez_gl_map %>%
  filter(fraction == "nuclear") %>%
  select(ENTREZID, value) %>%
  deframe()
```

```{r}
pathview(gene.data=entrez_gl_map_n,
                # pathway.id="rno00190", # OP
                # pathway.id="rno00020", # TCA
                # pathway.id="rno00010", # glycolysis
                # pathway.id="rno05022", # path of neurodege
                # pathway.id="rno05022", # path of neurodege
                # pathway.id="rno03010", # ribosome
                pathway.id="rno05012", # parkinsons
                species = 'rno',
         out.suffix = "Lichti_nucl_parkinsons",
         kegg.native = T,
         same.layer = T, #slower & on top (out of bounds?)
         res = 600, #400 dpi (def 300 dpi for png) # not sure if changes anything..
         low = list(gene = "#3E76E8", cpd = "#A054A3"), # purple unused 
         mid = list(gene= "grey85", cpd = "grey85"), 
         high = list(gene = "#DA3558", cpd ="#F3AD6AFF"), # org unused
         new.signature = FALSE)

# my_colors <- c("#E22A47", "#DA3558", "#D33F6A", "#317DEC", "#3E76E8", "#4A6FE3" )
```
