---
title: "2025_11_21_Park_2015_transcriptomics_EExAging"
author: "Stephanie Huang"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# SAGEseq Transcriptomics - Aging x EE

*From (Park et al., 2015) Integrative Transcriptome Profiling of Cognitive Aging and Its Preservation through Ser/Thr Protein Phosphatase Regulation [PLOSone] <doi:10.1371/journal.pone.0130891>*

-   (2025_12_16) - Mapping method: maps the gene name provided in their dataframe (mouse) then directly \*

    -   Tried to map Ref_Seq from their id column but is a mess. Their ID column is ref_seq, mirbase (e.g. MI0005496)

-   (2025_12_17) NOTE to self - very confusing data - so be careful with export lit review tables! Look into later.

## Load Packages

```{r}
library(readxl)
library(readr)
library(dplyr)
library(stringr)
library(tidyr)
library(ggplot2)
library(ggrepel) #  for geom_text_repel()
library(gt)
```

## Load data

```{r}
# S11 Table. Differentially expressed genes in young EE over young SH.
s11_EEyng <-read_excel("input/Park_2015_SAGEseq_mRNA_EExAging_s11_yngEE.xlsx",
                      skip = 4) # skip weird merge headers (take row 5 headers)

# this one does not had the row 5 header, skip 2 (weird title) to get r3 headers
s13_EEold <-read_excel("input/Park_2015_SAGEseq_mRNA_EExAging_s13_oldEE.xlsx",
                      skip = 2) 

# then trim 1st 2 empty rows
s13_EEold <- s13_EEold[-1:-2,] # 1st id is NM_001081078	Lct  
```

```{r}
park_raw_comb <- bind_rows(list(EE_yng = s11_EEyng,
                                EE_old = s13_EEold), .id = "transcriptomics_id")
```

### Uniprot df

```{r}
# load in uniprot df
load(file = "input/mouse_ref_proteome_2025_12_13_54864x295.rda")
load(file = "input/rat_ref_proteome_2025_12_13_51872x295.rda")
load(file = "input/rat_other_proteome_2025_12_15_39538x295.rda")
```

```{r}
# cols I want from uniprot

#colnames(mouse_ref_proteome_2025_12_13)

my_cols <- c("Entry", "Entry.Name", "Protein.names", "Gene.Names..primary.",
             "Gene.Names", "Gene.Names..synonym.",  "Annotation", "Reviewed",
             "Protein.existence", "Protein.families", "Length", "Organism",
             "Keywords", "Keyword.ID", "Function..CC.", "Miscellaneous..CC.", "Pathway",
             "GeneID", "Ensembl", "RefSeq", 
             "Gene.Ontology..GO.", "Gene.Ontology..cellular.component.",
             "Gene.Ontology..molecular.function.", "Gene.Ontology..biological.process.",
             "Gene.Ontology.IDs", "KEGG", "Reactome", "STRING",
             "Entry.version", "Date.of.creation", "Date.of.last.modification",
             "Date.of.last.sequence.modification"
             )
```

```{r}
# https://www.uniprot.org/help/protein_existence - see rankings 
unique(rat_ref_proteome_2025_12_13$Protein.existence) 

# MOUSE REF PROTEOME
mouse_ref <- mouse_ref_proteome_2025_12_13 %>% 
  dplyr::select(all_of(my_cols)) %>%
  mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence) %>%
  mutate(across(where(is.character), ~ na_if(.x, ""))) 


# need to run in separate line as mouse ref is input (otheriwise object not found )
mouse_ref <- mouse_ref %>% 
  mutate(occupied = rowSums(!is.na(mouse_ref)))

# 2025_12_15 - NOTE DID NOT GET MOUSE OTHER PROTEOME FROM UNIPROT BECAUSE IT SAYS CONTAIMINATED!
```

```{r}
# RAT REF PROTEOME
rat_ref <- rat_ref_proteome_2025_12_13 %>% 
  dplyr::select(all_of(my_cols)) %>%
  mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence) %>%
  mutate(across(where(is.character), ~ na_if(.x, ""))) 

rat_ref <- rat_ref %>%
  mutate(occupied = rowSums(!is.na(rat_ref)))
```

## Mapping

### Mouse to Mouse map

#### Map 1 - by Gene Name (primary)

```{r}
# by gene name
mRNA_mouse_map_1 <- left_join(park_raw_comb, mouse_ref,
                           by = join_by(`Gene name` == Gene.Names..primary.))

sum(is.na(mRNA_mouse_map_1$Entry)) # 227 empty
length(unique(mRNA_mouse_map_1$Entry)) # 1862 unique
```

```{r}
mouse_ref_ref_seq <- mouse_ref %>%
  mutate(ref_seq_split = str_split(RefSeq, ";")) %>%
  tidyr::unnest(ref_seq_split) %>% #from 54864 to 114352
  mutate(ref_seq_trim = paste0(gsub("\\..*", "", ref_seq_split)))
```

#### Map 2 - by Ref Seq

Won't map, even after fixing ids, not sure why ⟶ just map by gene id to rat

```{r}
# #unique(park_raw_comb$Type)
#  
# # "_dup_?[0-9]+$" trims the _dup at the end
# park_raw_comb_2 <- park_raw_comb %>%
#   filter(Type %in% c("transcript", "rRNA", "tRNA")) %>% # removes miRNA
#   mutate(identifier_trimmed = paste(gsub("_dup_?[0-9]+$", "", Identifier)),
#        .after = Identifier)
```

```{r}
# mRNA_mouse_map_2 <- left_join(park_raw_comb_2, mouse_ref_ref_seq,
#                               by = join_by(identifier_trimmed == ref_seq_trim))
# 
# sum(is.na(mRNA_mouse_map_2$Entry)) # 1115
# length(unique(mRNA_mouse_map_2$Entry)) # 1
# 
# # tried again, still can't map even when filtering out miRNA (which has mirbase)
```

```{r}
# intersect(mouse_ref_ref_seq$ref_seq_trim, park_raw_comb_2$identifier_trimmed) #0?
```

### Mouse to Rat

Direct Gene (park df mouse) to Uniprot Rat Ref Proteome

#### Rat Map 1 - Mouse gene name to Rat primary gene name

```{r}
length(unique(park_raw_comb$`Gene name`)) # 872 unique genes
```

```{r}
# map by primary gene 
mRNA_rat_map_1 <-left_join(park_raw_comb, rat_ref,
                           by = join_by(`Gene name` == Gene.Names..primary.)) %>%
  filter(Type == "transcript") # removes miRNA, tRNA, rRNA

sum(is.na(mRNA_rat_map_1$Entry)) # 291 empty
length(unique(mRNA_rat_map_1$Entry)) # 1778 unique

mRNA_rat_map_1_mapped <- mRNA_rat_map_1 %>%
  filter(!is.na(Entry))

# unmapped by primary gene | 291 missing
mRNA_rat_map_1_na <- mRNA_rat_map_1 %>%
  filter(is.na(Entry)) %>%
  select(transcriptomics_id:`Mean SY`) # peel back

length(unique(mRNA_rat_map_1_na$`Gene name`)) #199 unique genes left
```

#### Rat Map 2 - Mouse gene name to rat gene synonym (for unmapped in 1st)

```{r}
# note gene names is all, gene names synonym is all except gene names primary
# expands from 51872 to 55829
rat_ref_split <- rat_ref %>%
  mutate(synonyms_split = str_split(Gene.Names..synonym., " "),
         .after = Gene.Names..synonym.) %>%
  tidyr::unnest(synonyms_split)
```

```{r}
# recovered (36 unique genes)
mRNA_rat_map_2 <-left_join(mRNA_rat_map_1_na, rat_ref_split,
                           by = join_by(`Gene name` == synonyms_split))

sum(is.na(mRNA_rat_map_2$Entry)) # 244/291 empty
length(unique(mRNA_rat_map_2$Entry)) # 43 unique


# 
mRNA_rat_map_2_mapped <- mRNA_rat_map_2 %>%
  filter(!is.na(Entry))

# unmapped by rat synonym & primary gene | 244 missing
mRNA_rat_map_2_na <- mRNA_rat_map_2 %>%
  filter(is.na(Entry)) %>%
  select(transcriptomics_id:`Mean SY`) # peel back

length(unique(mRNA_rat_map_2_na$`Gene name`)) #164 unique genes left

# tested rat other proteome - only recovered 1 extra (decided to ignore, not worth)
```

### Combine Dfs

```{r}
# checks to inform grouping below
length(unique(park_raw_comb$transcriptomics_id)) # 2
length(unique(park_raw_comb$`Gene name`)) # 872
length(unique(park_raw_comb$Identifier)) # 1104

# length(unique(mRNA_mouse_map_2$Identifier)) # 1104
# length(unique(mRNA_mouse_map_2$identifier_trimmed)) # 1049
```

```{r}
mRNA_rat_map_comb <- bind_rows(list(by_ref_primary_gene = mRNA_rat_map_1_mapped,
                                    by_ref_synonym = mRNA_rat_map_2),
                               .id = "map_method_df") %>% # 2584|2579|

  # group with `Mean EA` as Idenfitied isn't quite as unique as i thought
  # this way it retains all ariginal data! 
  group_by(map_method_df, transcriptomics_id, `Gene name`, Identifier, `Mean EA`) %>%

  filter(if(any(Reviewed == 'reviewed', na.rm = TRUE)) { #1773|1768|
    Reviewed == "reviewed"
  } else {TRUE}) %>%

  slice_max(Annotation, n = 1) %>% #1445|1450|
  slice_max(existence_ranking, n = 1) %>% #1337|1332|
  slice_max(occupied, n = 1) %>% # 1137|1132|

  mutate(id_priority = case_when( 
    !is.na(GeneID) & !is.na(Ensembl) & !is.na(KEGG) & is.na(RefSeq) ~ 7, # has 4
    !is.na(GeneID) & !is.na(Ensembl) & !is.na(KEGG) ~ 6, # has both geneID & ensembl!
    !is.na(GeneID) & !is.na(Ensembl) ~ 5,
    !is.na(GeneID) ~ 4, # has only gene ID
    !is.na(Ensembl) ~ 3, # has only Ensembl
    !is.na(KEGG) ~ 2, # has only KEGG
    TRUE ~ 0 # all others (if all above were FALSE, then TRUE - so it writes 0)
  )) %>%
  slice_max(id_priority, n = 1, with_ties = FALSE) %>% # 11
  dplyr::select(-id_priority) 

# #with ties = FALSE ends with 1101, which is the length of original - 5 miRNA
# length(unique(mRNA_rat_map_comb$identifier_trimmed)) #850

```

```{r}
sum(is.na(mRNA_rat_map_comb$Entry)) # still 247 empties 
# note some extra map mice but some others are just not in uniprot! (since RNA data)
```

```{r}
# export combinded file
# write_excel_csv(mRNA_rat_map_comb, 
#                 "output/Park_2015_mRNA_mouse2rat_map_EExAging_v1_w_unmapped.csv")
```

## Make lit review df

### Plot Log2

```{r}
plot(mRNA_rat_map_comb$`log2 Ratio`) # equivalent to log2 ratio?
plot(mRNA_rat_map_comb$`log2 Signal`) 
par(mfrow = c(1, 2)) 
#par(mfrow = c(2, 1)) 
```

```{r}
summary(mRNA_rat_map_comb)
```

### Lit Review 1

```{r}
# NOTE THIS VERSION HAS MULTI & NA
# e.g. BDNF was OBSERVED MULTIPLE TIMES (with id)
# for lit review table | Note - duplicates have not been removed - to do manually
lit_review_mRNA_1 <- mRNA_rat_map_comb %>%
  mutate(
    Space = " ",
    Year = 2015,
    Author = 'Park',
    Animal = 'Mice',
    Condition = paste0("Age - ", transcriptomics_id),
    Region = 'Hippocampus',
    Whole = 'Yes',
    Exp_type = 'mRNA',
    Method = 'TRANSCRIPTOMICS_SAGEseq',
    Used_name = `Gene name`,
    Mod = " ",
    Mod_type = " ",
    Gene_name = `Gene name`, # genename used to map mouse primary to rat gene
    Comparison = 2,
    In_EE = ifelse(`log2 Signal` > 0, "Up", "Down"), # computed! 
    Fold_change = `log2 Signal`, # prob the same as log2FC?
    Arrows = case_when( # dplyr case when, instead of ifelse
      `log2 Signal` > 2.5 ~ '+++',
      `log2 Signal` > 1 ~ '++',
      `log2 Signal` > .05 ~ '+',
      `log2 Signal` > -.05 ~ '=',
      `log2 Signal` > -1 ~ '-',
      `log2 Signal` > -2.5 ~ '--',
      `log2 Signal` <= -2.5 ~ '---',
      ),
    Significant = ifelse(`P-value` <= 0.05, 'yes', 'no'),
    pvalue = `P-value`,
    N_tested = 3, # 3 per group (3 x 4 - EE x Aging)
    Stats_adj = 'unadjusted',
    Note = '', 
    Full_name = Protein.names,
    Uniprot_rat_accession = Entry,
    Uniprot_entry_name = Entry.Name,
    Uniprot_link = ifelse(!is.na(Entry),
                          paste0('https://www.uniprot.org/uniprotkb/', Entry,'/entry'),
                          ' '),
    Wikipedia_link = ifelse(!is.na(Entry),
                            paste0('https://en.wikipedia.org/wiki/',
                                   `Gene name`), ' '),
    
    Curation_method = case_when(
      Reviewed == "reviewed" & map_method_df == "by_ref_primary_gene"
        ~ "auto - ref & reviewed (primary gene name)",
      
      Reviewed == "unreviewed" & map_method_df == "by_ref_primary_gene"
        ~ "auto - ref & unreviewed (primary gene name)",
      
      Reviewed == "reviewed" & map_method_df == "by_ref_synonym"
        ~ "auto - ref & reviewed (synonym)",
      
      Reviewed == "unreviewed" & map_method_df == "by_ref_synonym"
        ~ "auto - ref proteome (synonym)",
      
      is.na(Entry.Name)
        ~ "manual - na"),
    
    Class_group = ' ',
    Other_link = ' ',
    Other_comments = 'misc shows combined gene_identifier from their df',
    misc = paste0(`Gene name`, '_', Identifier)
  ) %>%
  arrange(transcriptomics_id, desc(In_EE), Gene_name)
```

```{r}
# #writes out
# write_excel_csv(lit_review_mRNA_1,
#                 "output/2025_12_17_Park_2015_mRNA_lit_review_EExAging_multi_nas_v1.csv",
#                 na = "")
```

### Lit Review 2

```{r}
# to inform grouping > broadest to narrowest (otherwise chops off!)
length(unique(lit_review_mRNA_1$transcriptomics_id)) # 2
length(unique(lit_review_mRNA_1$map_method_df)) # 2
length(unique(lit_review_mRNA_1$Reviewed)) # 3
length(unique(lit_review_mRNA_1$Protein.names)) #693
length(unique(lit_review_mRNA_1$Entry)) #697
length(unique(lit_review_mRNA_1$Entry.Name)) #697
length(unique(lit_review_mRNA_1$`Gene name`)) #860
```

```{r}
# Summarises by gene name (to collapse multi entries) so 1 val per gene per group.
# for example some genes were identified multiple times (diff transcript ID)
# so collapse for simplicity into a summarised value 

lit_review_mRNA_2 <- mRNA_rat_map_comb %>% # starts off 1101
  #filter(isPresent == TRUE) %>% # confused by this
  group_by(transcriptomics_id, map_method_df,  Reviewed, # see ordering above
           Protein.names, Entry, Entry.Name, `Gene name`) %>%
  summarise(across(where(is.numeric), ~ mean(.x, na.rm = TRUE))) %>% # ends 870 rows
  ungroup() %>%
  #filter(!is.na(Entry)) %>% # unmapped ones.
    mutate(
    Space = " ",
    Year = 2015,
    Author = 'Park',
    Animal = 'Mice',
    Condition = paste0("Age - ", transcriptomics_id),
    Region = 'Hippocampus',
    Whole = 'Yes',
    Exp_type = 'mRNA',
    Method = 'TRANSCRIPTOMICS_SAGEseq',
    Used_name = `Gene name`,
    Mod = " ",
    Mod_type = " ",
    Gene_name = `Gene name`, # genename used to map mouse primary to rat gene
    Comparison = 2,
    In_EE = ifelse(`log2 Signal` > 0, "Up", "Down"), # computed! 
    Fold_change = `log2 Signal`, # prob the same as log2FC?
    Arrows = case_when( # dplyr case when, instead of ifelse
      `log2 Signal` > 2.5 ~ '+++',
      `log2 Signal` > 1 ~ '++',
      `log2 Signal` > .05 ~ '+',
      `log2 Signal` > -.05 ~ '=',
      `log2 Signal` > -1 ~ '-',
      `log2 Signal` > -2.5 ~ '--',
      `log2 Signal` <= -2.5 ~ '---',
      ),
    Significant = ifelse(`P-value` <= 0.05, 'yes', 'no'),
    pvalue = `P-value`,
    N_tested = 3, # 3 per group (3 x 4 - EE x Aging)
    Stats_adj = 'unadjusted',
    Note = '', 
    Full_name = Protein.names,
    Uniprot_rat_accession = Entry,
    Uniprot_entry_name = Entry.Name,
    Uniprot_link = ifelse(!is.na(Entry),
                          paste0('https://www.uniprot.org/uniprotkb/', Entry,'/entry'),
                          ' '),
    Wikipedia_link = ifelse(!is.na(Entry),
                            paste0('https://en.wikipedia.org/wiki/',
                                   `Gene name`), ' '),
    
    Curation_method = case_when(
      Reviewed == "reviewed" & map_method_df == "by_ref_primary_gene"
        ~ "auto - ref & reviewed (primary gene name)",
      
      Reviewed == "unreviewed" & map_method_df == "by_ref_primary_gene"
        ~ "auto - ref & unreviewed (primary gene name)",
      
      Reviewed == "reviewed" & map_method_df == "by_ref_synonym"
        ~ "auto - ref & reviewed (synonym)",
      
      Reviewed == "unreviewed" & map_method_df == "by_ref_synonym"
        ~ "auto - ref proteome (synonym)",
      
      is.na(Entry.Name)
        ~ "manual - na"),
    
    Class_group = ' ',
    Other_link = ' ',
    Other_comments = ' ',
    misc = ' '
  ) %>%
  arrange(transcriptomics_id, desc(In_EE), Gene_name)
  
```

```{r}
# #writes out
# write_excel_csv(lit_review_mRNA_2,
#                 "output/2025_12_17_Park_2015_mRNA_lit_review_EExAging_summed_v2.csv",
#                 na = "")
```

```{r}
# 
lit_2_counts <- lit_review_mRNA_2 %>%
  count(Gene_name) %>%
  arrange(desc(n))
```

## Extras

#### Volcano Plot

```{r fig.height=8, fig.width=12}
volcano_pv <- ggplot(lit_review_mRNA_2, 
                     aes(x = `log2 Signal`, y = -log10(`P-value`),
                         label = Gene_name)) +
  geom_point(aes(color = case_when(
    `P-value` < 0.05 & `log2 Signal` > 0.05 ~ "Upregulated (p<0.05)",
    `P-value` < 0.05 & `log2 Signal` < -0.05 ~ "Downregulated (p<0.05)",
                                   TRUE ~ "Not Significant")),
            alpha = 1, size = 2, shape = 1) +
  xlim(-8,12) +
  geom_vline(xintercept = 0, linetype = "dashed",
             color = "grey40", linewidth = .5) +
  scale_color_manual(values = c("firebrick3", 
                                "royalblue4", 
                                 "grey80"),
                     breaks = c("Upregulated (p<0.05)", 
                                "Downregulated (p<0.05)", 
                                "Not Significant"),
                     labels = c("Upregulated Proteins (p<0.05)", 
                                "Downregulated Proteins (p<0.05)", 
                                "Not Significant")) +
  
  geom_text_repel() +
  facet_wrap(~transcriptomics_id, ncol = 1) +
  labs(
    title = "Park 2015 - Lit Review 2 df volcano plot mRNA Hip Aging x EE (2025_12_17)",
       x = "Log2 Signal",
       y = "-Log10(p-value)",
       color = "mRNA Regulation") +
  theme_bw() +
  theme(legend.position = "bottom")



volcano_pv
```

#### gt

```{r}
# gt 1 - by fold change direction & pvalue
lit_review_mRNA_2 |> 
  #dplyr::select(-(c(`Av. Ratio`, ))) |>
  dplyr::select(c(Condition, In_EE, Gene_name, Entry, Entry.Name,
                  Protein.names, Fold_change, pvalue,
                  `Mean EA`, `Mean EY`, `Mean SA`,  `Mean SY`)) |>
  dplyr::arrange(pvalue, Fold_change) |>
  gt(
    groupname_col = "Condition",
    rowname_col = "In_EE"
  ) |>
  fmt_number(decimals = 2) |>
    tab_header(
    title = "Park et al., 2015 - SAGEseq mRNA - Hippocampus (Mice) - Aging x EE",
    subtitle = "2025_12_17 - lit_review_mRNA_2 - mapped genes 2 UP rat + summed genes"
    ) |>
    tab_options(table.font.size=11.5,
                stub.font.weight = "bold",
                row_group.font.weight = "bold",
                row_group.font.size = 12.5,
                column_labels.font.size = 12.5,
                column_labels.font.weight = "bold")   
```

#### Lit Review 3 - (collapsed groups)

Some weird ones are sig (probably not good to do since data is unclear) - ignore - did not export?

```{r}
# lit_review_mRNA_3 <- mRNA_rat_map_comb %>% #1101
#   filter(isPresent == TRUE) %>% # 945
#   group_by(map_method_df, Reviewed, `Gene name`, Entry,
#            Entry.Name, Protein.names) %>%
#   summarise(across(where(is.numeric), mean)) %>% # 732
#   ungroup() %>%
#   filter(!is.na(Entry)) %>% #612
#     mutate(
#     Space = " ",
#     Year = 2015,
#     Author = 'Park',
#     Animal = 'Mice',
#     Condition = ' ',
#     Region = 'Hippocampus',
#     Whole = 'Yes',
#     Exp_type = 'mRNA',
#     Method = 'TRANSCRIPTOMICS_SAGEseq',
#     Used_name = `Gene name`,
#     Mod = " ",
#     Mod_type = " ",
#     Gene_name = `Gene name`, # genename used to map mouse primary to rat gene
#     Comparison = 2,
#     In_EE = ifelse(`log2 Signal` > 0, "Up", "Down"), # computed! 
#     Fold_change = `log2 Signal`, # prob the same as log2FC?
#     Arrows = case_when( # dplyr case when, instead of ifelse
#       `log2 Signal` > 2.5 ~ '+++',
#       `log2 Signal` > 1 ~ '++',
#       `log2 Signal` > .05 ~ '+',
#       `log2 Signal` > -.05 ~ '=',
#       `log2 Signal` > -1 ~ '-',
#       `log2 Signal` > -2.5 ~ '--',
#       `log2 Signal` <= -2.5 ~ '---',
#       ),
#     Significant = ifelse(`P-value` <= 0.05, 'yes', 'no'),
#     pvalue = `P-value`,
#     N_tested = 3, # 3 per group (3 x 4 - EE x Aging)
#     Stats_adj = 'unadjusted',
#     Note = '', 
#     Full_name = Protein.names,
#     Uniprot_rat_accession = Entry,
#     Uniprot_entry_name = Entry.Name,
#     Uniprot_link = ifelse(!is.na(Entry),
#                           paste0('https://www.uniprot.org/uniprotkb/', Entry,'/entry'),
#                           ' '),
#     Wikipedia_link = ifelse(!is.na(Entry),
#                             paste0('https://en.wikipedia.org/wiki/',
#                                    `Gene name`), ' '),
#     
#     Curation_method = case_when(
#       Reviewed == "reviewed" & map_method_df == "by_ref_primary_gene"
#         ~ "auto - ref & reviewed (primary gene name)",
#       
#       Reviewed == "unreviewed" & map_method_df == "by_ref_primary_gene"
#         ~ "auto - ref & unreviewed (primary gene name)",
#       
#       Reviewed == "reviewed" & map_method_df == "by_ref_synonym"
#         ~ "auto - ref & reviewed (synonym)",
#       
#       Reviewed == "unreviewed" & map_method_df == "by_ref_synonym"
#         ~ "auto - ref proteome (synonym)",
#       
#       is.na(Entry.Name)
#         ~ "manual - na"),
#     
#     Class_group = ' ',
#     Other_link = ' ',
#     Other_comments = ' ',
#     misc = ' '
#   ) %>%
#   arrange(desc(In_EE), Gene_name)
```

\~ \~ \~ \~

### Map remaining (just to see)

```{r}
# Tries to map the 244 empty cells from un mapped rat 2 
mRNA_mouse_map_3 <- left_join(mRNA_rat_map_2_na, mouse_ref, #274
                              by = join_by(`Gene name` == Gene.Names..primary.)) %>%
  group_by(transcriptomics_id, `Gene name`, Identifier, `Mean EA`) %>%

  filter(if(any(Reviewed == 'reviewed', na.rm = TRUE)) { #273
    Reviewed == "reviewed"
  } else {TRUE}) %>%
  slice_max(Annotation, n = 1) %>% #257
  slice_max(existence_ranking, n = 1) %>% #252
  slice_max(occupied, n = 1)  #244/
  
# most recovered are unreviered with low annotation scores anyway
```

```{r}

```

```{r}

```

```{r}
# # Originally all 4 means columns are Mean EA, Mean EY, Mean SA, Mean SY
# # so recognise common prefix & then take the remainder to the groups variable
# # and the the value & put into means column
# # end up with long form df with 2 cols - Group & Mean 
# park_df_long <- park_df %>%
#   pivot_longer(cols = starts_with("Mean "), 
#                names_to = "Group",
#                names_prefix = "Mean ",
#                values_to = "Mean")
```

```{r}
# density_plot <- ggplot(park_df_long,
#                        aes(x = Mean, fill = Group)) + 
#   geom_histogram(binwidth = 10, color = "grey10", alpha = .4) +
#   xlim(0,1000) +
#   facet_wrap(~Group, ncol = 1) + 
#   theme_bw()
# 
# density_plot
```

```{r}
# bp_mean_dist <- ggplot(park_df_long,
#                        aes(x = Group, y = Mean, fill = Group)) +
#   geom_boxplot(outliers = FALSE, alpha = 0.5) +
#   labs(title = "",
#        subtitle = "Yes! EE Young has the higher values (more transciptional activity?) 
#        \nGraph has outliers removed btw via geom_boxplot (outliers = FALSE)") +
#   theme_bw()
# 
# bp_mean_dist
```

------------------------------------------------------------------------

# Aging Peptides/Proteins/Phosphopeptides

Data from Supplementary Table 1

-   [NOTE - return to later for subsequent comparison of age. Data is a bit messy (not 100% clear on their processing). Results are not that exciting...]{.smallcaps}

-   [Some weird values (e.g. some 1000s, these huge numbers seem to be driving differences...) look in to..]{.smallcaps}

-   [Also, tried linear model only has 3 sig. Most are not significant as it seems the variance is huge! massive standard errors!]{.smallcaps}

    -   [Rab GDP dissociation inhibitor alpha (P15105) ⟶ ProteinGDIA -1.638e+02 4.479e+01 -3.657 0.000278 \*\*\*]{.smallcaps}

    -   Isoform of inositol 1,4,5-trisphosphate-gated calcium channel (P11881) [⟶ ProteinITPR1 -3.247e+02 4.479e+01 -7.249 1.36e-12 \*\*\*]{.smallcaps}

    -   Isoform of E3 ubiquitin-protein ligase UHRF2 (Q7TMI3) [⟶]{.smallcaps} [ProteinWRIP1 -1.539e+02 4.479e+01 -3.436 0.000633 \*\*\*]{.smallcaps}

LOOK INTO LATER!!!

```{r}
# # this one does not had the row 5 header, skip 2 (weird title) to get r3 headers
# s1_prot <-read_excel("input/Park_2015_iTRAQ_proteomics_aging_s1_all_pep.xlsx",
#                       skip = 2,
#                      range = cell_rows(3:1651),
#                      na = c("---", "###")) # get --- ### & input as na (+ prevents misasignment as character)
# 
# # then trim 1st 2 empty rows
# s1_prot <- s1_prot[-1:-2,] #1st entry is P62806 & last is Q6PFF0
```

```{r}
# # instpect raw
# # colnames(s1_prot)
# # unique(s1_prot$Protein)
# summary(s1_prot)
```

```{r}
# prot_aging_1 <- s1_prot %>% #1646
#   filter(Expect <= 0.05) %>% #1595
#   filter(!(str_detect(Protein, "ZCont0"))) %>% #1587 | rm those w/ ZCont0 contaminants
#   group_by(Group, `Accession Number`, Protein, `Protein description`) %>%
#   summarise(across(`Tag  114/117`:`Tag  116/117`, ~ mean(.x, na.rm = TRUE))) %>%
#   ungroup()


```

```{r}
# prot_count_1 <- prot_aging_1 %>%
#   count(Protein) %>%
#   arrange(desc(n)) 
# 
# # with 2 or more obs (so 1 in aging & in 1 young?)
# prot_count_2 <- prot_count_1 %>%
#   filter(n > 1)
```

```{r}
# prot_aging_2 <- prot_aging %>%
#   filter(Protein %in% prot_count_2$Protein) %>%
#   pivot_longer(cols = starts_with("Tag "),
#                names_to = "Tag",
#                values_to = "value") %>%
#   mutate(across(where(is.character), as.factor ))

```

```{r}
# prot_aging_2_plot <- ggplot(prot_aging_2, 
#                             aes(x = Protein, y = value, fill = Group)) +
#   geom_boxplot() +
#   theme_bw()

```

see <https://stats.oarc.ucla.edu/stat/data/intro_r/intro_r_interactive_flat.html#linear-regression>

-   `summary()`: regression table of coefficients, standard errors, test statistics, and p-values, as well as overall model fit statistics

```{r}
# m2 <- lm(formula = value ~ Protein + Group, data = prot_aging_2)
# 
# summary(m2) # summary produces regression table and model fit stats
# # coef(m1) # just model coefficients
# # confint(m1) # 95% confidence intervals
# 
# # first 5 observed values, predicted values and residuals
# # cbind() joins column vectors into a matrix
# #cbind(prot_aging_2$value, predict(m1), residuals(m1))
```
