---
title: "2025_11_04_Perez_2024_Proteomics_RNASeq"
author: "Stephanie Huang"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Perez 2024 - multiomics EE x aging (hippocampus mice)

Nice study! :)

*from (Perez et al., 2024) A multiomic atlas of the aging hippocampus reveals molecular changes in response to environmental enrichment [nature communications] <https://doi.org/10.1038/s41467-024-49608-z>*

# Proteomics

-   Their provided dfs have the uniprot accession from mice. So mapped those to the mouse_ref (uniprot df), to get gene names which were then mapped to the rat gene gene names (where uniprot infor was pulled) to get rat proteins & other details.

-   NOTE: some exported randomly have **Q6LED0 Histone H3.1 in the exported sheets? not sure why? look into?**

    -   Nothing to do with group & ungroup() - doesn't fix (tested 2025_12_27)

-   Code is a bit of a mess (maybe clean up later in a new version).

## Set up

```{r}
library(tidyverse)
library(readxl)
library(writexl)
library(ggrepel) #  for geom_text_repel()
library(gt)
```

### Read in proteomics sheets

```{r}
prot_df_EEall <- read_excel("input/Perez_2024_EE_age_proteomics_sds20_41467_2024_49608_MOESM23_ESM.xlsx",
                             sheet = "EEall")

prot_df_EEyng <- read_excel("input/Perez_2024_EE_age_proteomics_sds20_41467_2024_49608_MOESM23_ESM.xlsx",
                             sheet = "EEyng")

prot_df_EEold <- read_excel("input/Perez_2024_EE_age_proteomics_sds20_41467_2024_49608_MOESM23_ESM.xlsx",
                             sheet = "EEold")
```

#### Bind into a master file

```{r}
prot_df <- bind_rows(list(all = prot_df_EEall,
                          young = prot_df_EEyng,
                          old = prot_df_EEold),
                     .id = "proteomics_id")
```

### Uniprot dataframes

```{r}
# load in uniprot df
load(file = "input/mouse_ref_proteome_2025_12_13_54864x295.rda")
load(file = "input/rat_ref_proteome_2025_12_13_51872x295.rda")
load(file = "input/rat_other_proteome_2025_12_15_39538x295.rda")
```

```{r}
# cols I want from uniprot

#colnames(mouse_ref_proteome_2025_12_13)

my_cols <- c("Entry", "Entry.Name", "Protein.names", "Gene.Names..primary.",
             "Gene.Names", "Gene.Names..synonym.",  "Annotation", "Reviewed",
             "Protein.existence", "Protein.families", "Length", "Organism",
             "Keywords", "Keyword.ID", "Function..CC.", "Miscellaneous..CC.", "Pathway",
             "GeneID", "Ensembl",
             "Gene.Ontology..GO.", "Gene.Ontology..cellular.component.",
             "Gene.Ontology..molecular.function.", "Gene.Ontology..biological.process.",
             "Gene.Ontology.IDs", "KEGG", "Reactome", "STRING",
             "Entry.version", "Date.of.creation", "Date.of.last.modification",
             "Date.of.last.sequence.modification"
             )

```

#### Adding additional metrics to the UniProt dataframes

*The code below adds additional metrics to the uniprot dataframe including a calculation score for protein existence ranked from highest (evidence at the protein level) & and occupied metric that measures the number of NA values per row. These scoring metrics are useful to break ties when there are multiple instances of a mapping (see later ...).*

[Stepwise process includes...]{.smallcaps}

1.  [Selecting desired columns (based on the `my_cols` list above)]{.smallcaps}

2.  [Make a `Protein.existence` column with assigned rankings. For example, `Protein.existence == "Evidence at protein level" ~ 5` will allow it to try to keep the highest scoring observation when using `slice_max()` down the line (see <https://www.uniprot.org/help/protein_existence> for more info).]{.smallcaps}

3.  [Calculating NAs per row. this metric will allow keeping the row with the most info (is partially covered by the existing annotation column that is supplied by uniprot but still useful to break further ties at the end).]{.smallcaps}

    1.  [Note: some blank entries have `NA` but some are `""` (just blank) (which makes them automatically chr class btw), to convert them to all `NA` without disturbing other columns use `na_if()` from dplyr (<https://dplyr.tidyverse.org/reference/na_if.html>) & then mutate across where is.character (see below).]{.smallcaps}

    2.  [Afterwards then the next computed column occupied will correctly count the NA across the row using baseR rowSums]{.smallcaps}

```{r}
# https://www.uniprot.org/help/protein_existence - see rankings 
unique(rat_ref_proteome_2025_12_13$Protein.existence) 

# MOUSE REF PROTEOME
mouse_ref <- mouse_ref_proteome_2025_12_13 %>% 
  dplyr::select(all_of(my_cols)) %>%
  mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence) %>%
  mutate(across(where(is.character), ~ na_if(.x, ""))) 


# need to run in separate line as mouse ref is input (otheriwise object not found )
mouse_ref <- mouse_ref %>% 
  mutate(occupied = rowSums(!is.na(mouse_ref)))

# 2025_12_15 - NOTE DID NOT GET MOUSE OTHER PROTEOME FROM UNIPROT BECAUSE IT SAYS CONTAIMINATED!
```

```{r}
# RAT REF PROTEOME
rat_ref <- rat_ref_proteome_2025_12_13 %>% 
  dplyr::select(all_of(my_cols)) %>%
  mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence) %>%
  mutate(across(where(is.character), ~ na_if(.x, ""))) 

rat_ref <- rat_ref %>%
  mutate(occupied = rowSums(!is.na(rat_ref)))
```

```{r}
# RAT OTHER PROTEOME 
rat_other <- other_rat_proteome_2025_12_15 %>%
    dplyr::select(all_of(my_cols)) %>%
    mutate(existence_ranking = case_when( # ranking level of evidence
    Protein.existence == "Evidence at protein level" ~ 5,
    Protein.existence == "Evidence at transcript level" ~ 4,
    Protein.existence == "Inferred from homology" ~ 3, 
    Protein.existence == "Predicted" ~ 2, 
    Protein.existence == "Uncertain" ~ 1
    ), .after = Protein.existence) %>%
  mutate(across(where(is.character), ~ na_if(.x, ""))) 

rat_other <- rat_other %>%
  mutate(occupied = rowSums(!is.na(rat_other)))
```

## Mapping

### Mice: Map provided UniProt accession ➝ Gene names & other UP info

,Map uniprot accessions provided by perez to uniprot mice df to get primary gene names etc

Note: occasionally some uniprot ids are 2 in 1 row, sep by ; must unnest. So need to mutate a new column by string_split to recognize the ; when multiple entires are in one Perez's UNIPROT column. Then use tidyr unnest() to split & expand.

Side note: just unnest UNIPROT (ignore UNIPROT_DB since the other df has this info anyway.) otherwise Error in `tidyr::unnest()`: ! In row 279, can't recycle input of size 3 to size 2.

Join join provided df to uniprot mouse df (left_join keeps all of x (original df))

```{r}
#  String split & unnest to deal with 2 in 1 accessions in cells
prot_df_unnested <- prot_df %>%
  mutate(split_UNIPROT = str_split(UNIPROT, ";"),
         .after = UNIPROT) %>%
  tidyr::unnest(split_UNIPROT) # goes from 6750 to 6849 rows
```

```{r}
# Join Perez df to mouse_ref by uniprot accession. 
prot_df_mouse <- left_join(prot_df_unnested, mouse_ref,
                           by = join_by(UNIPROT == Entry))

sum(is.na(prot_df_mouse$Entry.Name)) # check unmapped | 291 of 6558 unmapped
#sum(!(is.na(prot_df_mouse$Entry.Name))) # 6558 
```

### Rat

#### Rat 1 - Map uniprot rat (primary genes) to mice primary genes

Now with the mapped mouse df. Use the prot_df_mouse, select cols of interest, rename those that will duplicate (to avoid .x at the end when joining, & then join.

Probably could have just mutated .x to .m, but oh well.

```{r}
#colnames(prot_df_mouse)

prot_df_map_rat_1 <- prot_df_mouse %>%
  dplyr::select(proteomics_id:Gene.Names, Annotation:Protein.existence,
                Protein.families:Length, GeneID:Ensembl) %>% # 24 cols
  rename(entry.name.m = Entry.Name, # new = old
         protein.names.m = Protein.names,
         gene.names.primary.m = Gene.Names..primary.,
         gene.names.m =  Gene.Names,
         annotation.m = Annotation,
         reviewed.m = Reviewed,
         protein.existence.m = Protein.existence,
         protein.families.m = Protein.families,
         length.m = Length,
         GeneID.m = GeneID,
         Ensembl.m = Ensembl) %>%
  # group_by(proteomics_id) %>%
  left_join(rat_ref, # many to many warning
            by = join_by(gene.names.primary.m == Gene.Names..primary.))

# gives many to many warning (fine?) but why 337629 (prev 24513) 
# unique entries are the same though 
```

#### Rat map 2 - trim multi entries (by group!!!)

Make sure to group by proteomics_id so it doesn't delete adjacent groups! (e.g. EEold).

Takes prot_df_map_rat_1 then filters out any NAs enties (so unable to map by primary gene name), then groups by proteomics_id (so EEyoung, EEold, EEall), then gene names & then UNIPROT id (provided by Perez) (this is to deal with any gene ids that get dealt to multiple unique uniprot accession entries). Afterwards, the filtering begins - swissprot reviewed \> best annotation score \> existence ranking \> occupied (least NAs).

```{r}
# informs grouping.
length(unique(prot_df_map_rat_1$proteomics_id)) # 3
length(unique(prot_df_map_rat_1$gene.names.primary.m)) # 2186
length(unique(prot_df_map_rat_1$split_UNIPROT)) # 8040
length(unique(prot_df_map_rat_1$Entry)) # 8040
```

```{r}
# take non empty entries & then group & filter any with multiples
# based on reviewed > annotation score > protein existence

prot_df_map_rat_2 <- prot_df_map_rat_1 %>% #24513 | 337629
  group_by(proteomics_id, gene.names.primary.m, split_UNIPROT) %>% # group by
    # 1 - remove NAs
  filter(!(is.na(Entry))) %>% #24117 | 337524
    # 2 - keep reviewed/swissprot entries
  # or maybe group by just proteomics_id, gene - since rat map is based on gene???
  filter(if(any(Reviewed == "reviewed", na.rm = TRUE)) {
     Reviewed == "reviewed" # if reviewed = keep,
   } else {TRUE}) %>% #10278|44907 - if none, keep unreviewed ones, & filter below
    # 3
  slice_max(Annotation, n = 1) %>% #8445|10773
    # 4 # keep entry with least NAs
  slice_max(existence_ranking, n = 1) %>% #7905|9942
    # 5 - keep entry with least NAs
  slice_max(occupied, n = 1) %>% # |7017 
  
  # still 216 with duplicate entries 
    # 6 - final 
  mutate(id_priority = case_when(
    !is.na(GeneID) & !is.na(Ensembl) & !is.na(KEGG) ~ 6, # has geneID & ensembl & KEGG!
    !is.na(GeneID) & !is.na(Ensembl) ~ 5, # has both geneID & ensembl!
    !is.na(GeneID) ~ 4, # has only gene ID
    !is.na(Ensembl) ~ 3, # has only Ensembl
    !is.na(KEGG) ~ 2, # has only KEGG
    TRUE ~ 0 # all others (if all above were FALSE, then TRUE - so it writes 0)
  )) %>%
  slice_max(id_priority, n = 1, with_ties = FALSE) %>% 
  dplyr::select(-id_priority) %>%
  ungroup() # good practice?

# 1st 6849 (geneID & ensembl - still has duplicates -> 153)
# 2nd 6924? (geneID & ensembl & kegg - still has duplicates -> 129)
# note varying combos of Gene.ID.m, Ensembl.m,  Reactome, STRING, 
# also tried go_id_count - did not resolve - seems to be isoforms? i think
```

```{r}
# # 216 with duplicate entries | after (GeneID & Ensembl trim) = 153 
# # 129 KEGG,GeneID& Ensembl| 0 eeny meeny miny moe then with_ties 
# 
# # for checking duplicate entries trimming (see case_when in the chunk above)
# prot_df_map_rat_2_dups <- prot_df_map_rat_2 %>%
#   count(gene.names.primary.m) %>%
#   filter(n > 1) 
# 
# prot_df_maps_rat_2_inspect <- prot_df_map_rat_2 %>%
#   filter(gene.names.primary.m %in% prot_df_map_rat_2_dups$gene.names.primary.m)
```

```{r}
# dropped a bit due to removing the NA values ?
length(unique(prot_df_map_rat_2$gene.names.primary.m)) # 2151
length(unique(prot_df_map_rat_2$split_UNIPROT)) # 2248
length(unique(prot_df_map_rat_2$Entry)) # 2151
```

#### Rat map 3 - map nas with gene synonyms

*Filter NA Entries (no rat accession), then select & peel back columns (to avoid duplicating in subsequent join), then filter out non NAs in the gene.names.primary.m col otherwise remaps & results in \>3mil cols*

```{r}
prot_df_map_rat_1_na <- prot_df_map_rat_1 %>%
  filter(is.na(Entry)) %>% #396|105 obs?
  dplyr::select(proteomics_id:Ensembl.m) %>% # peel back to 24 cols
  filter(!is.na(gene.names.primary.m)) # filter out non NAs otherwise maps >3 mil below
```

```{r}
# Prep rat_ref_split df 
rat_ref_split <- rat_ref %>% #51872 to 55829
  mutate(split_synonyms = str_split(Gene.Names..synonym., " ")) %>% #sep split by space
  tidyr::unnest(split_synonyms)
```

Recovered Sh3bgrl, Tceal3, Mtco2, Mtnd1, Mtnd5, Eif2s3x (gene synonyms that match mouse primary). No duplicates here so no need to rank.

```{r}
# note filtered NAs in gene.names.primary.m (otherwise maps 3 million as NA to NA)
# results in 18 of 105 recovered (or 6 proteins (since 3 groups))
prot_df_map_rat_3 <- left_join(prot_df_map_rat_1_na, rat_ref_split,
                               by = join_by(gene.names.primary.m == split_synonyms))

sum(is.na(prot_df_map_rat_3$Entry)) # still 87 NA
# many of those NAs see either too specific or too general or non reference prote.
# e.g.  Histone H3,  RIKEN cDNA 2900026A02 gene, etc unmapped
```

```{r}
prot_df_map_rat_3_filtered <- prot_df_map_rat_3 %>%
  filter(!is.na(Entry))
```

#### Rat map 4 - map remaining missing values to other rat proteome

```{r}
prot_df_map_rat_3_na <- prot_df_map_rat_3 %>%
  filter(is.na(Entry)) %>% #|87
  dplyr::select(proteomics_id:Ensembl.m) %>% # peel back to 24 cols
  filter(!is.na(gene.names.primary.m)) # filter out non NAs otherwise maps >3 mil below
```

```{r}
prot_df_map_rat_4 <- left_join(prot_df_map_rat_3_na, rat_other,
                               by = join_by(
                                 gene.names.primary.m == Gene.Names..primary.)) %>%
  # recovered Grcc10, Timm8a1, H3f3a, Phpt1, Vbp1 
  group_by(proteomics_id, gene.names.primary.m, split_UNIPROT) %>%
  filter(if(any(Reviewed == "reviewed", na.rm = TRUE)) {
     Reviewed == "reviewed"
   } else {TRUE}) %>%
  slice_max(Annotation, n = 1) %>%
  slice_max(existence_ranking, n = 1) %>%
  slice_max(occupied, n = 1) %>% # from 90 entries to 87 - only drops dup of Grcc10
  ungroup() # good practice?

sum(is.na(prot_df_map_rat_4$Entry)) # still 72 NA (or 24 missing?)

# Note: deleted rat_other_split with synonyms as it did not further recover any IDs. 
# hence combining this chunk
# note the join has 90 entries & follow up filter drops it to 3 (so only removes the 3)
```

### Combine rat maps

```{r}
prot_df_map_rat_combined <- bind_rows(list(
  by_ref_primary_gene = prot_df_map_rat_2,
  by_ref_synonym = prot_df_map_rat_3_filtered, 
  by_other_primary_gene = prot_df_map_rat_4),
  .id = "map_method_df") %>%
  arrange(proteomics_id, desc(dir), gene.names.primary.m) 
  
dim(prot_df_map_rat_combined) # 6849 x 58 | still some duplicates?
```

```{r}
# # export
# write_excel_csv(prot_df_map_rat_combined,
#                 "output/2025_12_16_Perez_2024_proteomics_map_rat_combined_v1.csv")

# # export | with ungroup() added ? not sure if it matters -> check
# write_excel_csv(prot_df_map_rat_combined,
#                 "output/2025_12_27_Perez_2024_proteomics_map_rat_combined_v1.csv")
```

### Proteomics Lit Review file

```{r}
plot(prot_df_map_rat_combined$logFC)
```

```{r}
# for lit review table | Note - duplicates have not been removed - to do manually
lit_review_proteomics <- prot_df_map_rat_combined %>%
  mutate(
    Year = 2024,
    Author = 'Perez',
    Animal = 'Mice',
    Condition = proteomics_id,
    Region = 'Hippocampus',
    Whole = 'Yes',
    Exp_type = 'Protein',
    Method = 'PROTEOMICS_LCMS_LFQ_DIA_OS',
    Used_name = SYMBOL,
    Mod = " ",
    Mod_type = " ",
    Gene_name = `gene.names.primary.m`, # genename used to map mouse 1° to rat gene
    Comparison = 2,
    In_EE = dir,
    Fold_change = logFC, # prob the same as log2FC?
    Arrows = case_when( # dplyr case when, instead of ifelse
      logFC > 2.5 ~ '+++',
      logFC > 1 ~ '++',
      logFC > .05 ~ '+',
      logFC > -.05 ~ '=',
      logFC > -1 ~ '-',
      logFC > -2.5 ~ '--',
      logFC <= -2.5 ~ '---',
      ),
    Significant = ifelse(adj.P.Val <= 0.05, 'yes', 'no'),
    pvalue = adj.P.Val,
    N_tested = ' ',
    Stats_adj = 'adjusted',
    Note = ' ',
    Full_name = Protein.names,
    Uniprot_rat_accession = Entry,
    Uniprot_entry_name = Entry.Name,
    Uniprot_link = ifelse(!is.na(Entry),
                          paste0('https://www.uniprot.org/uniprotkb/', Entry,'/entry'),
                          ' '),
    Wikipedia_link = ifelse(!is.na(Entry),
                            paste0('https://en.wikipedia.org/wiki/',
                                   `gene.names.primary.m`), ' '),
    
    Curation_method = case_when(
      Reviewed == "reviewed" & map_method_df == "by_ref_primary_gene"
        ~ "auto - ref & reviewed (primary gene name)",
      
      Reviewed == "unreviewed" & map_method_df == "by_ref_primary_gene"
        ~ "auto - ref & unreviewed (primary gene name)",
      
      Reviewed == "reviewed" & map_method_df == "by_ref_synonym"
        ~ "auto - ref & reviewed (synonym)",
      
      Reviewed == "unreviewed" & map_method_df == "by_ref_synonym"
        ~ "auto - ref proteome (synonym)",
      
      Reviewed == "reviewed" & map_method_df == "by_other_primary_gene"
        ~ "auto - other & reviewed (primary gene name)",
      
      is.na(Entry.Name)
        ~ "manual - na"),
    
    Class_group = ' ',
    Other_link = ' ',
    Other_comments = 'misc shows combined gene_uniprot_ensembl info from their df',
    misc = paste0(SYMBOL, '_', UNIPROT, '_',  GENEID)
  )
```

Write excel

<https://r4ds.hadley.nz/spreadsheets.html#sec-writing-to-excel>

```{r}
# results in 3 x [2283 x 89] df
split_prot_df_map_rat <- split(lit_review_proteomics,
                               lit_review_proteomics$proteomics_id)

# then add the combined one at the end 
split_prot_df_map_rat$combined <- lit_review_proteomics
```

```{r}
# #writes out by sheet (using writexl package)
# write_xlsx(split_prot_df_map_rat,
#            path = "output/2025_12_16_Perez_2024_proteomics_lit_review_all_v1.xlsx")

# #writes out by sheet (using writexl package) | with ungroup?
# write_xlsx(split_prot_df_map_rat,
#            path = "output/2025_12_27_Perez_2024_proteomics_lit_review_all_v1.xlsx")
```

Significant Only

```{r}
lit_review_proteomics_sig <- lit_review_proteomics %>%
  filter(adj.P.Val <= 0.05) #6936 to 2015

split_prot_df_map_rat_sig <- split(lit_review_proteomics_sig,
                                   lit_review_proteomics_sig$proteomics_id)

split_prot_df_map_rat_sig$combined <- lit_review_proteomics_sig
```

```{r}
# write_xlsx(split_prot_df_map_rat_sig,
#            path = "output/2025_12_16_Perez_2024_proteomics_lit_review_sig_v1.xlsx")
```

```{r}
prot_df_map_rat_sig <- prot_df_map_rat_combined  %>%
  filter(adj.P.Val <= 0.05) # 2097 entries! 
```

### Plots

#### Volcano plot

```{r fig.height=18, fig.width=12}
# color coding doesn't work!?!?
volcano_pv <- lit_review_proteomics_sig %>%
  ungroup() %>%
  ggplot(aes(x = logFC, y = -log2(adj.P.Val),
                         label = Gene_name)) +
  geom_point(aes(color = case_when(
    adj.P.Val < 0.05 & logFC > 0.05 ~ "Upregulated (adj.p<0.05)",
    adj.P.Val < 0.05 & logFC < -0.05 ~ "Downregulated (adj.p<0.05)",
                                   TRUE ~ "Not Significant")),
            alpha = 1, size = 1, shape = 1) +
  # xlim(-8,12) +
  geom_vline(xintercept = 0, linetype = "dashed",
             color = "grey40", linewidth = .5) +
  scale_color_manual(values = c("firebrick3", 
                                "royalblue4", 
                                 "grey80"),
                     breaks = c("Upregulated (adj.p<0.05)", 
                                "Downregulated (adj.p<0.05)", 
                                "Not Significant"),
                     labels = c("Upregulated Proteins (p<0.05)", 
                                "Downregulated Proteins (p<0.05)", 
                                "Not Significant")) +
  
  geom_text_repel(label.size = .1) +
  facet_wrap(~proteomics_id, ncol = 1,
             scales = "free_y") +
  labs(
    title = "Perez 2024 - Proteomics - Hippocampus (Mice) - Aging x EE (2025_12_16/17)",
    subtitle = "lit_review_proteomics_sig df - map up mouse (provided) 2 up mouse gene 2 up rat gene",
       x = "Log2 Signal",
       y = "-Log2(adj.p-value)",
       color = "mRNA Regulation") +
  theme_bw() +
  theme(legend.position = "bottom",
        strip.text = element_text(size = 12))



volcano_pv
```

### gt tables

ignore

```         
NA  Q6LED0  H31_RAT Histone H3.1
```

#### gt 1 - All (EE vs SH)

```{r}
# EE VS SH (ALL)
# gt 1 - by fold change direction & pvalue
lit_review_proteomics_sig |> 
  dplyr::ungroup() |>
  dplyr:: filter(Condition == "all") %>%
  #dplyr::select(-(c(`Av. Ratio`, ))) |>
  dplyr::select(c(Condition, In_EE, Gene_name, Entry, Entry.Name,
                  Protein.names, Fold_change, adj.P.Val)) |>
  dplyr::arrange(desc(In_EE), adj.P.Val, Fold_change) |>
  gt(
    groupname_col = "Condition",
    rowname_col = "In_EE"
  ) |>
  fmt_number(decimals = 3) |>
    tab_header(
    title = "All - Perez 2024 - Proteomics - Hippocampus (Mice) (adj.p <=0.05 only)",
    subtitle = "2025_12_17 - lit_review_proteomics_sig - (EE vs SH - by all)"
    ) |>
    tab_options(table.font.size=11.5,
                stub.font.weight = "bold",
                row_group.font.weight = "bold",
                row_group.font.size = 12.5,
                column_labels.font.size = 12.5,
                column_labels.font.weight = "bold")   
```

#### gt 2 - Young (EE vs SH)

```{r}
# EE VS SH (Young)
# gt 2 - by fold change direction & pvalue
lit_review_proteomics_sig |> 
  dplyr::ungroup() |>
  dplyr:: filter(Condition == "young") %>%
  #dplyr::select(-(c(`Av. Ratio`, ))) |>
  dplyr::select(c(Condition, In_EE, Gene_name, Entry, Entry.Name,
                  Protein.names, Fold_change, adj.P.Val)) |>
  dplyr::arrange(desc(In_EE), adj.P.Val, Fold_change) |>
  gt(
    groupname_col = "Condition",
    rowname_col = "In_EE"
  ) |>
  fmt_number(decimals = 3) |>
    tab_header(
    title = "Young - Perez 2024 - Proteomics - Hippocampus (Mice) (adj.p <=0.05 only)",
    subtitle = "2025_12_17 - lit_review_proteomics_sig - (EE vs SH - by young)"
    ) |>
    tab_options(table.font.size=11.5,
                stub.font.weight = "bold",
                row_group.font.weight = "bold",
                row_group.font.size = 12.5,
                column_labels.font.size = 12.5,
                column_labels.font.weight = "bold")   
```

#### gt 3 - Old (EE vs SH)

```{r}
# EE VS SH (Yold)
# gt 3 - by fold change direction & pvalue
lit_review_proteomics_sig |> 
  dplyr::ungroup() |>
  dplyr:: filter(Condition == "old") %>%
  #dplyr::select(-(c(`Av. Ratio`, ))) |>
  dplyr::select(c(Condition, In_EE, Gene_name, Entry, Entry.Name,
                  Protein.names, Fold_change, adj.P.Val)) |>
  dplyr::arrange(desc(In_EE), adj.P.Val, Fold_change) |>
  gt(
    groupname_col = "Condition",
    rowname_col = "In_EE"
  ) |>
  fmt_number(decimals = 3) |>
    tab_header(
    title = "Old - Perez 2024 - Proteomics - Hippocampus (Mice) (adj.p <=0.05 only)",
    subtitle = "2025_12_17 - lit_review_proteomics_sig - (EE vs SH - by old)"
    ) |>
    tab_options(table.font.size=11.5,
                stub.font.weight = "bold",
                row_group.font.weight = "bold",
                row_group.font.size = 12.5,
                column_labels.font.size = 12.5,
                column_labels.font.weight = "bold")   
```

### Extra

```{r}
# prot_df_map_rat_keywords <- prot_df_map_rat_4 %>%
#   filter(adj.P.Val <= 0.05) %>% # 2309
#   filter(!is.na(Entry)) %>% # 2278
#   mutate(hashtags = str_split(Keywords, ";")) %>%
#   tidyr::unnest(hashtags) # 22696
```

```{r}
# prot_rat_sig_count <- prot_df_map_rat_sig %>%
#   #group_by(proteomics_id, dir) %>%
#   group_by(proteomics_id) %>%
#   filter(hashtags != "Reference proteome") %>% 
#   count(hashtags, sort = TRUE) 
```

```{r}
# find_mouse_Q6LED0 <- mouse_ref %>%
#   filter(Entry == "Q6LED0")
# 
# find_rat_Q6LED0 <- rat_ref %>%
#   filter(Entry == "Q6LED0")
# 
# find_rat_other_Q6LED0 <- rat_other %>%
#   filter(Entry == "Q6LED0")
```

------------------------------------------------------------------------

### 2025_12_27 - read in of Zenodo data to check sample size etc

In their Zenodo readme doc, this data is described as... **2_proteomicsSWATH_phenodata_averagedReplicates.txt:** phenodata for the proteomics experiment, having averaged the technical replicates into single biological replicates. (so columns, except ID, equtes to 12 which probably means 12 mice total 6 per group?)

```{r}
norm_SWATH <- read_tsv("input/Perez_2024_2_proteomicsSWATH_normalized_log2_filtered_averagedReplicates.txt")

colnames(norm_SWATH)
str(norm_SWATH)
```

------------------------------------------------------------------------

# Transcriptomics

Just wrote out unique ensembls & searched in uniprot web (a lot faster & easier)

### Read in mRNA/transcirptomics DEG excel sheets (sup ds 18)

```{r}
# some NAs in the original df were directly coded as NA
# so need to specify na = "NA" otherwise some get classed as character & cant bind
rna_df_EEall <- read_excel("input/Perez_2024_EE_age_transcriptomics_sds18_41467_2024_49608_MOESM21_ESM.xlsx",
                             sheet = "EEall",
                           na = "NA")

rna_df_EEyng <- read_excel("input/Perez_2024_EE_age_transcriptomics_sds18_41467_2024_49608_MOESM21_ESM.xlsx",
                           sheet = "EEyng",
                           na = "NA")

rna_df_EEold <- read_excel("input/Perez_2024_EE_age_transcriptomics_sds18_41467_2024_49608_MOESM21_ESM.xlsx",
                           sheet = "EEold",
                           na = "NA")
```

```{r}
RNA_df <- bind_rows(list(all = rna_df_EEall,
                          young = rna_df_EEyng,
                          old = rna_df_EEold),
                     .id = "transcriptomics_id") %>%
  rename(ENSEMBL = GENEID)
```

------------------------------------------------------------------------

## test RNA direct gene symbol map

```{r}
RNA_df_map_rat_1 <- left_join(RNA_df, rat_ref,
                              by = join_by(SYMBOL == Gene.Names..primary.))
```

```{r}
RNA_df_map_rat_1_mapped <- RNA_df_map_rat_1 %>%
  filter(!is.na(Entry))
```

```{r}
RNA_df_map_rat_2 <- RNA_df_map_rat_1 %>%
  filter(is.na(Entry)) %>%
  select(transcriptomics_id:SYMBOL) %>%
  left_join(rat_ref_split, by = join_by(SYMBOL == split_synonyms))
```

```{r}
RNA_df_map_rat_combined <- bind_rows(list(
  by_rat_ref_primary_gene = RNA_df_map_rat_1_mapped,
  by_rat_ref_synonym = RNA_df_map_rat_2), 
  .id = "map_method_df") %>%
  group_by(map_method_df, transcriptomics_id, ENSEMBL) %>% 
  filter(if(any(Reviewed == "reviewed", na.rm = TRUE)) {
    Reviewed == "reviewed"
  } else{TRUE}) %>% 
  slice_max(Annotation, n = 1) %>% 
  slice_max(existence_ranking, n =1) %>% 
  slice_max(occupied, n = 1, with_ties = FALSE) %>%
  ungroup()
```

```{r}
# # export
# write_excel_csv(RNA_df_map_rat_combined,
#                 "output/2025_12_27_Perez_2024_RNA_map_rat_combined_v1.csv")
```

### RNA Literature Review file

```{r}
# to check range of 
plot(RNA_df_map_rat_combined$log2FoldChange)
summary(RNA_df_map_rat_combined) # see range
colnames(RNA_df_map_rat_combined)
```

```{r}
# for lit review table 
lit_review_transcriptomics <- RNA_df_map_rat_combined %>%
  mutate(
    Year = 2024,
    Author = 'Perez',
    Animal = 'Mice',
    Condition = transcriptomics_id,
    Region = 'Hippocampus',
    Whole = 'Yes',
    Exp_type = 'mRNA',
    Method = 'TRANSCRIPTOMICS_RNAseq',
    Used_name = SYMBOL,
    Mod = " ",
    Mod_type = " ",
    Gene_name = SYMBOL, # genename used to map mouse 1° to rat gene
    Comparison = 2,
    In_EE = dir,
    Fold_change = log2FoldChange, # prob the same as log2FC?
    Arrows = case_when( # dplyr case when, instead of ifelse
      log2FoldChange > 2.5 ~ '+++',
      log2FoldChange > 1 ~ '++',
      log2FoldChange> .05 ~ '+',
      log2FoldChange > -.05 ~ '=',
      log2FoldChange > -1 ~ '-',
      log2FoldChange > -2.5 ~ '--',
      log2FoldChange <= -2.5 ~ '---',
      ),
    Significant = ifelse(padj <= 0.05, 'yes', 'no'),
    pvalue = padj,
    N_tested = ' ',
    Stats_adj = 'adjusted',
    Note = ' ',
    Full_name = Protein.names,
    Uniprot_rat_accession = Entry,
    Uniprot_entry_name = Entry.Name,
    Uniprot_link = ifelse(!is.na(Entry),
                   paste0('https://www.uniprot.org/uniprotkb/', Entry,'/entry'),
                          ' '),
    Wikipedia_link = ifelse(!is.na(Entry),
                            paste0('https://en.wikipedia.org/wiki/',
                                   `Gene.Names..primary.`), ' '),
    
    Curation_method = case_when(
      Reviewed == "reviewed" & map_method_df == "by_rat_ref_primary_gene"
        ~ "auto - ref & reviewed (primary gene name)",
      
      Reviewed == "unreviewed" & map_method_df == "by_rat_ref_primary_gene"
        ~ "auto - ref & unreviewed (primary gene name)",
      
      Reviewed == "reviewed" & map_method_df == "by_rat_ref_synonym"
        ~ "auto - ref & reviewed (synonym)",
      
      Reviewed == "unreviewed" & map_method_df == "by_rat_ref_synonym"
        ~ "auto - ref proteome (synonym)",
      
      is.na(Entry.Name) ~ "manual - na"
      ),
    Class_group = ' ',
    Other_link = ' ',
    Other_comments = 'misc shows combined gene_ensembl info from their df',
    misc = paste0(SYMBOL, '_', ENSEMBL)
  ) %>%
  arrange(Condition, desc(In_EE), pvalue, Fold_change)
  
```

#### All for export

```{r}
# split by age (all, young, old) (in prep for sheet write out)
split_RNA_df_map_rat <- split(lit_review_transcriptomics,
                              lit_review_transcriptomics$transcriptomics_id)

# add combined one back onto it 
split_RNA_df_map_rat$combined <- lit_review_transcriptomics
```

```{r}
# # write transcriptomics lit review out excel 
# write_xlsx(split_RNA_df_map_rat,
#            path = "output/2025_12_27_Perez_2024_transcriptomics_lit_review_all_v1.xlsx")
```

#### Significant only for export

```{r}
# filter significant only 
# purrr meth (loaded with tidyverse?)
split_RNA_df_map_rat_sig  <- map(split_RNA_df_map_rat, ~filter(.x, padj <= 0.05))
```

```{r}
# #write out sig version
# write_xlsx(split_RNA_df_map_rat,
#            path = "output/2025_12_27_Perez_2024_transcriptomics_lit_review_sig_only_v1.xlsx")
```

### gt sig mRNA

-   <https://gt.albert-rapp.de/styling>
-   <https://www.danieldsjoberg.com/gt-and-gtsummary-presentation/#1>

```{r}
split_RNA_df_map_rat_sig$combined |> 
  dplyr::select(c(Condition, In_EE, Gene_name, Entry, `Entry.Name`,
                  `Protein.names`, Fold_change, pvalue, Reviewed )) |>
  gt(
    groupname_col = "Condition",
    rowname_col = "In_EE"
  ) |>
  fmt_number(decimals = 3) |>
    tab_header(
    title = "Perez 2024 - Transcriptomics - Hippocampus Mice - adjp <= 0.05 only",
    subtitle = "2025_12_27 - lit_review_transcriptomics/split_RNA_df_map_rat_sig - (EE vs SH split by Age (all (96), young (60), old (36)))"
    ) |>
    tab_options(heading.title.font.size = 16,
                table.font.size=11.5,
                stub.font.weight = "bold",
                row_group.font.weight = "bold",
                row_group.font.size = 12.5,
                column_labels.font.size = 12.5,
                column_labels.font.weight = "bold") #|>
  # opt_stylize(style = 6, color = 'pink') # cute but also ew looks like excel
  # opt_stylize(style = 1, color = 'gray', add_row_striping = TRUE)
```

### Plots

```{r}
plot(lit_review_transcriptomics$pvalue) # see distribution of pvalues
plot(lit_review_transcriptomics$Fold_change) # see distribution of fold change
```

#### Volcano plot

```{r fig.height=18, fig.width=12}
volcano_RNA <- lit_review_transcriptomics %>%
  filter(pvalue <= 0.2) %>% # filter otherwise 20000 dots per plot... 
  ggplot(aes(x = Fold_change, y = -log2(pvalue),
                         label = Gene_name)) +
  geom_point(aes(color = case_when(
    pvalue < 0.05 & Fold_change > 0.05 ~ "Upregulated (adj.p<0.05)",
    pvalue < 0.05 & Fold_change < -0.05 ~ "Downregulated (adj.p<0.05)",
    pvalue < 0.1 & Fold_change > 0.05 ~ "Upregulated (adj.p<0.1)",
    pvalue < 0.1 & Fold_change < -0.05 ~ "Downregulated (adj.p<0.1)",
                                   TRUE ~ "Not Significant")),
            alpha = 1, size = 1.5, shape = 1) +
  # xlim(-8,12) +
  geom_vline(xintercept = 0, linetype = "dashed",
             color = "grey40", linewidth = .5) +
  scale_color_manual(values = c("firebrick3", 
                                "royalblue4", 
                                "lightpink1", 
                                "lightblue3" ,
                                 "grey80"),
                     breaks = c("Upregulated (adj.p<0.05)", 
                                "Downregulated (adj.p<0.05)", 
                                "Upregulated (adj.p<0.1)",
                                "Downregulated (adj.p<0.1)",
                                "Not Significant"),
                     labels = c("Upregulated mRNA (adj.p<0.05)", 
                                "Downregulated mRNA (adj.p<0.05)", 
                                "Upregulated mRNA (adj.p<0.1)", 
                                "Downregulated mRNA (adj.p<0.1)", 
                                "Not Significant")) +
  
  geom_text_repel() +
  facet_wrap(~Condition, ncol = 1,
             scales = "free_y") +
  labs(
    title = "Perez 2024 - Transriptomics - Hippocampus (Mice) - EE x Aging  (2025_12_27)",
    subtitle = "",
       x = "Log2 Signal",
       y = "-Log2(adj.p-value)",
       color = "mRNA Regulation") +
  theme_bw() +
  theme(legend.position = "bottom",
        strip.text = element_text(size = 12))

volcano_RNA
```

------------------------------------------------------------------------

```{r}
# intersect of observed 
intersect(split_prot_df_map_rat$combined$Gene_name,
          split_RNA_df_map_rat$combined$Gene_name)

length(intersect(split_prot_df_map_rat$combined$Gene_name,
          split_RNA_df_map_rat$combined$Gene_name)) #2101 (just proteins from)


#intersect of significant observed 
intersect(split_prot_df_map_rat_sig$combined$Gene_name,
          split_RNA_df_map_rat_sig$combined$Gene_name)

length(intersect(split_prot_df_map_rat_sig$combined$Gene_name,
          split_RNA_df_map_rat_sig$combined$Gene_name)) #10


```

```{r}
# # mega df (combining botth lit reviews)
# mega_df <- bind_rows(list(lit_review_proteomics = lit_review_proteomics,
#                     lit_review_transcriptomics = lit_review_transcriptomics), 
#                     .id = "omics_assay")

# list of common genes/proteins (SYMBOL) observed in both
common_list <- intersect(split_prot_df_map_rat$combined$Gene_name,
                        split_RNA_df_map_rat$combined$Gene_name)

# make vector of conditions
# conditions <- unique(mega_df$Condition)
conditions <- unique(lit_review_proteomics$Condition)
```

TO DO: Maybe do a for loop to get the overlapping proteins/genes for each entry? look at their old code?
