---
title: "BioIn_2026_01_v3.4.2_ClusterProfiler"
author: "Stephanie Huang"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# ClusterProfiler BioIn vis - 1pspg v3.2.2 EE x Age t1

*for MSstatsTMT 1pspg v3.2.2 EE x Age t1 -(defaults) with B1 129N removed (TMT channel with too much loaded)*

This script is for running the GSEA - uses for loops.

This is a temporary script is for visualization of the results

Outputs

1.  2025_12_31

2.  

## Set up

### Packages

```{r}
# # Install if needed 
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("clusterProfiler")
# BiocManager::install("msigdbr")
# BiocManager::install("org.Rn.eg.db")
# BiocManager::install("enrichplot")
# BiocManager::install("pathview")
```

```{r}
# Load Packages 
library(tidyverse) # for dplyr, ggplot, stringr, readr etc 
library(clusterProfiler) #GSEA includes gsea, enrichplot etc.
library(org.Rn.eg.db) # Rat genome anno pack- incl. AnnotationDbi
library(msigdbr) # MSigDB gene sets
library(enrichplot) # for dotplot()
library(ggridges) # for ridgeplot()
library(pathview) # pathview() - Kegg pathway visualisation
```

### Import data

Load the mapped comparisons output generated from

```{r}
# mapped_comp_v3.2.2_EExAge_t1
load("input/protein_lists/2025_12_31_1pspg_v3.2.2_t1_EExAge_mapped_comp.rda")

df <- mapped_comp_v3.2.2_EExAge_t1
```

lil checks

```{r}
# lil checks
head(df) # peak
str(df) # structure | 7746 x 9
colnames(df) # column names

# lil checks (bit messy)
unique(df$issue) # if = na in issue column then good. if not exclude
check_1 <- df %>% filter(str_detect(log2FC, "Inf")) # 5 inf 
check_2 <- df %>% filter(str_detect(Protein, "Cont")) # 0 (since already removed)
unique(df$Label) 
length(unique(df$Label)) #8
```

### **Clean up df & rename**

```{r}
protein_comp <- df %>%
  filter(is.na(issue)) %>% # keeps rows with issue == na (so comparisons that are ok)
  filter(!str_detect(log2FC, "Inf")) # remove infinity/missing fold change values

# check 
head(protein_comp) 
colnames(protein_comp)

```

```{r}
# check 
unique(df$Label) 
# don't want "HxA_EE_YNG_vs_SH_OLD" "HxA_EE_OLD_vs_SH_YNG", so just do manually 

# VECTOR for labels (contrasts from MSstatsTMT analysis)
label_names <- c("H_EE_vs_SH", "A_YNG_vs_OLD",
                 "HxA_EE_YNG_vs_EE_OLD", "HxA_SH_YNG_vs_SH_OLD",
                 "HxA_EE_YNG_vs_SH_YNG", "HxA_EE_OLD_vs_SH_OLD")

# # CONTAINER - EMPTY VECTOR
# gsea_KEGG_results_list <- vector(mode = "list")
```

------------------------------------------------------------------------

## Testing conversion of bitr (extra)

From FASTA - uniprot uniprot acc

<https://yulab-smu.top/biomedical-knowledge-mining-book/utilities.html#sec-id-convert>

```{r}
protein_comp_uni_prot <- unique(protein_comp$Protein) # 8048 
protein_comp_uni_gene <- unique(protein_comp$gene) # 8014
```

```{r}
# Gene name 2 EntrezID | .0339 x 8104 = 272 ish
bitr_gn_2_entrez <- bitr(protein_comp_uni_gene,
                         fromType = "SYMBOL",
                         toType = "ENTREZID",
                         OrgDb = "org.Rn.eg.db") #3.39% failed to map... better
# 'select()' returned 1:1 mapping between keys and columns
#Warning message: In bitr(protein_comp_uni_gene, fromType = "SYMBOL", toType = "ENTREZID",:  3.39% of input gene IDs are fail to map...


# Seems gn 2 entrez is the superior option. 
length(unique(bitr_gn_2_entrez$SYMBOL)) # 7742
length(unique(bitr_gn_2_entrez$ENTREZID)) # 7742

# length(unique(bitr_up_2_entrez$UNIPROT)) # 7325
# length(unique(bitr_up_2_entrez$ENTREZID)) # 7392
```

```{r}
protein_comp_entrez <- left_join(protein_comp, bitr_gn_2_entrez,
                                 by = join_by(gene == SYMBOL)) %>%
  filter(!is.na(ENTREZID)) # remove non mapped - 64384 to 61952 
```

old stuff

------------------------------------------------------------------------

```{r}
# load gsea results from v3.4.2. 31 dec & 01 jan 
# GSEA KEGG - 2025_12_31 - gseKEGG - keytype Uniprot
load("output/2025_12_31_KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1.rda")
# # GSEA GO - 2025_12_31 / gseGO (keytype Uniprot) | takes forever....
# load("output/2025_12_31_GO_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1.rda")
# ~ ~ ~ ~ ~~
# GSEA Msigdb Hallmarks Collection  / gsea key type ncbi entrez - 2026_01_01
load("output/2026_01_01_gsea_msig_Hallmark_EExAge_cpv3.4.2_mssv3.2.2_t1.rda")
# GSEA Kegg REDO gsekegg / entrez - 2026_01_01
load("output/2026_01_01_KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1.rda")

```

### print dot plots 

```{r fig.height=13, fig.width=11}
print(KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1$dotplots) # kegg uniprot 


print(gsea_msig_Hmrk_EExAge_cp3.4.2_mss3.2.2_t1$dotplots) # msigdb hallmarks entrez
print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$dotplots) # kegg entrez
```

### Print dotplots sign

```{r fig.height=14, fig.width=12}
print(KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1$dotplots_sign) # kegg uniprot 


print(gsea_msig_Hmrk_EExAge_cp3.4.2_mss3.2.2_t1$dotplots_sign) # msigdb hallmarks entrez
print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$dotplots_sign) # kegg entrez
```

```{r}

print(KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1$dotplots$H_EE_vs_SH)
print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$dotplots$H_EE_vs_SH)
```

## Comparing 

```{r}
enrichplot::dotplot(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$kegg_results$H_EE_vs_SH,
                              showCategory = 50,
                              font.size = 11,
                              title = 'EE vs SH - entrez ver 20260101 ',
                              # subtitle = 
                              #   '1pspg MSstatsTMT (v3.2.2) 2025_12_31 - gseGO_r'
                    )
```

```{r}
enrichplot::dotplot(KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1$kegg_results$H_EE_vs_SH,
                              showCategory = 50,
                              font.size = 11,
                              title = 'EE vs SH - uniprot ver 20251231 ',
                              # subtitle = 
                              #   '1pspg MSstatsTMT (v3.2.2) 2025_12_31 - gseGO_r'
                    )
```

```{r}
enrichplot::dotplot(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$kegg_results$A_YNG_vs_OLD,
                              showCategory = 50,
                              font.size = 11,
                              title = 'Age - Yng vs old - entrez ver 20260101 ',
                              # subtitle = 
                              #   '1pspg MSstatsTMT (v3.2.2) 2025_12_31 - gseGO_r'
                    )
```

```{r}
enrichplot::dotplot(KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1$kegg_results$A_YNG_vs_OLD,
                              showCategory = 50,
                              font.size = 11,
                              title = 'Age - Yng vs old  - uniprot ver 20251231 ',
                              # subtitle = 
                              #   '1pspg MSstatsTMT (v3.2.2) 2025_12_31 - gseGO_r'
                    )
```

## Test of pathview 

### Main Effect of housing - EE vs SH

```{r}
# print EE vs SH sig KEGG gsea res tibble 
print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$kegg_results_tib$H_EE_vs_SH)
```

```{r}
# Produce the native KEGG plot (PNG)
dme <- pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno00190",
                species = 'rno')



knitr::include_graphics("rno00190.pathview.png")
```

```{r}
# # NVM can't map since uniprot IDs! 
# # Produce the native KEGG plot (PNG)
# dmet <- pathview(gene.data=KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1$geneList_vec$H_EE_vs_SH,
#                 pathway.id="rno00190",
#                 species = 'rno')
# 
# 
# 
# knitr::include_graphics("rno00190.pathview.png")
```

```{r}
#synaptic vesicle cycle
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno04721",
                species = 'rno')
```

```{r}
# Parkinson's
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno05012",
                species = 'rno')
```

```{r}
# Retrograde_endocannabinoid
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno04723",
                species = 'rno')

#Error in img[pidx[i, 3]:pidx[i, 4], sel.px, 1:3] : 
  # only 0's may be mixed with negative subscripts
```

```{r}
# Neuroactive_ligand
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno04082",
                species = 'rno')
```

```{r}
# Diabetic_cardiomyopathy
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno05415",
                species = 'rno')
```

```{r}
# proteosome 
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno03050",
                species = 'rno')
```

```{r}
# TCA_cycle
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno00020",
                species = 'rno')
```

```{r}
# Chemical_carcogenesis_ROS
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno05208",
                species = 'rno')

```

```{r}
# spliceosome
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$H_EE_vs_SH,
                pathway.id="rno03040",
                species = 'rno')
```

### Aging - Young vs Old 

```{r}
# print Yng vs Old sig KEGG gsea res tibble 
print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$kegg_results_tib$A_YNG_vs_OLD)
```

```{r}
# COVID19
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno05171",
                species = 'rno')
```

```{r}
# Ribosome
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno03010",
                species = 'rno')
```

```{r}
# Salmonella_infection 
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno05132",
                species = 'rno')
```

```{r}
# axon guidance  
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno04360",
                species = 'rno')
```

```{r}
# Proteasome  
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno03050",
                species = 'rno')
```

```{r}
# Diabetic Myocardiopathy
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno05415",
                species = 'rno')
```

```{r}
# FC_gam_Rm_phagocytosis rno04666
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno04666",
                species = 'rno')
```
