---
title: "BioIn_2025_12_v3.4.2_ClusterProfiler"
author: "Stephanie Huang"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# ClusterProfiler BioIn - 1pspg v3.2.2 EE x Age t1

*for MSstatsTMT 1pspg v3.2.2 EE x Age t1 - (defaults) with B1 129N removed (TMT channel with too much loaded)*

This script is for running the clusterprofiler GSEA for KEGG, GO & Msigdb (Hallmarks & CellTypes, (maybe add more?)) - uses for loops.

### Outputs

*Initial Tests (2025_12_31 - 2026_01_04)*

1.  **2025_12_31** - GO & KEGG (uniprot ID)

2.  **2026_01_01** - Msigdb Hallmarks (gene name 2 entrez) & KEGG redo (with gene name 2 entrez)

3.  2026_01_04 - Msigdb C8 cell types (gene name 2 entrez)

➝ [*Deleted old stuff (uniprot -\> now using gene name 2 entrez better mapping)*]{.underline}

*New version (used for PhD write up) (2026_01_12)*

1.  2026_01_12 - GO redo & KEGG redoredo, Msigdb Hallmarks & C8 celltypes redo

    1.  All use Entrez & feature all EExAge groups (8 instead 6) & for loops to run clusterprofiler GSEA (fgsea) all defaults.

    2.  [KEGG via gseKEGG]{.underline} ➝ KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3 ➝ 2026_01_12_KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3.rda

    3.  [**GO (MF, BP, & CC)** via gseGO]{.underline} ➝ GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2 ➝ 2026_01_12_GO_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t2.rda

    4.  [Msigdb H Hallmarks via GSEA]{.underline} ➝ gsea_msig_Hmrk_EExAge_mss3.2.2_t1_cp3.4.2_t2 ➝ 2026_01_12_gsea_msig_Hallmark_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda

    5.  [Msigdb C8 Celltypes_via_GSEA]{.underline} ➝ gsea_msig_C8_ct_EExAge_cp3.4.2_mss3.2.2_t1 ➝ 2026_01_04_gsea_msig_C8_ct_EExAge_cpv3.4.2_mssv3.2.2_t1.rda

## Set up

### Packages

```{r}
# # Install if needed 
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("clusterProfiler")
# BiocManager::install("msigdbr")
# BiocManager::install("org.Rn.eg.db")
# BiocManager::install("enrichplot")
# BiocManager::install("pathview")
#BiocManager::install("ReactomePA")
```

```{r}
# Load Packages 
library(tidyverse) # for dplyr, ggplot, stringr, readr etc 
library(clusterProfiler) #GSEA includes gsea, enrichplot etc.
library(org.Rn.eg.db) # Rat genome anno pack- incl. AnnotationDbi
library(msigdbr) # MSigDB gene sets
library(enrichplot) # for dotplot()
library(ggridges) # for ridgeplot()
library(pathview) # pathview() - Kegg pathway visualisation
# library(ReactomePA)
```

### Import data

Load the mapped comparisons output generated from

```{r}
# mapped_comp_v3.2.2_EExAge_t1
load("input/protein_lists/2025_12_31_1pspg_v3.2.2_t1_EExAge_mapped_comp.rda")

df <- mapped_comp_v3.2.2_EExAge_t1
```

lil checks

```{r}
# lil checks
head(df) # peak
str(df) # structure | 7746 x 9
colnames(df) # column names

# lil checks (bit messy)
unique(df$issue) # if = na in issue column then good. if not exclude
check_1 <- df %>% filter(str_detect(log2FC, "Inf")) # 5 inf 
check_2 <- df %>% filter(str_detect(Protein, "Cont")) # 0 (since already removed)
unique(df$Label) 
length(unique(df$Label)) #8
```

### **Clean up df & rename**

```{r}
protein_comp <- df %>%
  filter(is.na(issue)) %>% # keeps rows with issue == na (so comparisons that are ok)
  filter(!str_detect(log2FC, "Inf")) # remove infinity/missing fold change values

# check 
head(protein_comp) 
colnames(protein_comp)
```

## Testing conversion of bitr

From FASTA - uniprot uniprot acc

<https://yulab-smu.top/biomedical-knowledge-mining-book/utilities.html#sec-id-convert>

```{r}
protein_comp_uni_prot <- unique(protein_comp$Protein) # 8048 
protein_comp_uni_gene <- unique(protein_comp$gene) # 8014
```

```{r}
# using bitr()
# Uniprot 2 EntrezID | 0.0898 x 80
bitr_up_2_entrez <- bitr(protein_comp_uni_prot,
                         fromType = "UNIPROT", 
                         toType = "ENTREZID",
                         OrgDb = "org.Rn.eg.db") # 8.98% failed 2 map


# Gene name 2 EntrezID | .0339 x 8104 = 272 ish
bitr_gn_2_entrez <- bitr(protein_comp_uni_gene,
                         fromType = "SYMBOL",
                         toType = "ENTREZID",
                         OrgDb = "org.Rn.eg.db") #3.39% failed to map... better


# Seems gn 2 entrez is the superior option. 
length(unique(bitr_gn_2_entrez$SYMBOL)) # 7742
length(unique(bitr_gn_2_entrez$ENTREZID)) # 7742

length(unique(bitr_up_2_entrez$UNIPROT)) # 7325
length(unique(bitr_up_2_entrez$ENTREZID)) # 7392
```

```{r}
length(intersect(bitr_gn_2_entrez$SYMBOL, hallmarks$gene_symbol)) # 2226
length(intersect(bitr_gn_2_entrez$ENTREZID, hallmarks$ncbi_gene)) # 2237 !
# so use ncbi_gene (ENTREZID)

# intersect()
# 
setdiff(protein_comp$gene, bitr_gn_2_entrez$SYMBOL) # must be in order 
# most are "ENSRNOG00000066004" etc, then AABR07007121.1 & LOC108352348, ~ 200 
# less than a 100 are the lower case gene name looking ones like histone H3-3b or mitochon "Mt-nd4l"
# are hard to map, & also some rat specific ones like Akr1c9 so
# TLDR - not bad ! gene name is the way 2 go! 
```

```{r}
#  USE THIS METHOD! 
protein_comp_entrez <- left_join(protein_comp, bitr_gn_2_entrez,
                                 by = join_by(gene == SYMBOL)) %>%
  filter(!is.na(ENTREZID)) # remove non mapped - 64384 to 61952 
```

```{r}
# # check 
# unique(df$Label) 
# # don't want "HxA_EE_YNG_vs_SH_OLD" "HxA_EE_OLD_vs_SH_YNG", so just do manually 
# 
# # VECTOR for labels (contrasts from MSstatsTMT analysis)
# label_names <- c("H_EE_vs_SH", "A_YNG_vs_OLD",
#                  "HxA_EE_YNG_vs_EE_OLD", "HxA_SH_YNG_vs_SH_OLD",
#                  "HxA_EE_YNG_vs_SH_YNG", "HxA_EE_OLD_vs_SH_OLD")
```

```{r}
# VECTOR for labels (contrasts from MSstatsTMT analysis) 
label_names <- unique(df$Label) 
```

## KEGG ENTREZ ID all comps

<https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/>

```{r}
# # VECTOR for labels (contrasts from MSstatsTMT analysis) 
# label_names <- unique(df$Label) # already specified above

# # CONTAINER - EMPTY VECTOR
gsea_KEGG_results_list <- vector(mode = "list")
```

```{r}
for (lb in label_names) {
  
  # PART 1
  geneList_working_df <- protein_comp_entrez %>% # use ENTREZ!!!
    filter(Label == lb) %>% 
    mutate(ranking = sign(log2FC)*(-log10(pvalue))) %>%
    group_by(ENTREZID) %>% # use ENTREZID
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
    arrange(-ranking)
  
  # next deframe into a named vector needed to supply 2 fgsea()
  geneList_vec_working <- tibble::deframe(geneList_working_df)
  
  # store in list 
  gsea_KEGG_results_list$geneList_vec[[lb]] <- geneList_vec_working
  

  # PART 2 - KEGG GSEA
  # run KEGG gsea with clusterProfiler
  kegg_res_working <- clusterProfiler::gseKEGG(
                              geneList = geneList_vec_working, 
                              organism = 'rno',
                              keyType = 'ncbi-geneid', # ncbi ENTREZ 
                              by = 'fgsea',
                              pvalueCutoff = 0.05, 
                              pAdjustMethod = 'BH', #BejaminiHoc Correciton
                              minGSSize = 10, 
                              maxGSSize = 500,
                              eps = 1e-10
                              ) # ALL DEFAULTS
  
  # save kegg result original to list vec
  gsea_KEGG_results_list$kegg_results[[lb]] <- kegg_res_working
  
  # not needed here so compute & save directly? or can do anyway? idk?
  gsea_KEGG_results_list$kegg_results_tib[[lb]] <- tibble::as_tibble(kegg_res_working) 
  
  
  # PART 3 - PLOTS
  # Make dotplots | this one plots & saves properly!
  # 3.1 - regular dot blot
  dp <- enrichplot::dotplot(kegg_res_working,
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - 1pspg MSstatsTMT (v3.2.2) 2026_01_12 - gseKEGG_r'))) 

  
  print(dp)   # print dot plots 
  gsea_KEGG_results_list$dotplots[[lb]] <- dp # save to list 
  
  # 3.2 - dot plot split by sign
  dp_sign <- enrichplot::dotplot(kegg_res_working,
                            split = '.sign',
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - 1pspg MSstatsTMT (v3.2.2) 2026_01_12 - gseKEGG_r - by sign'))) +
    facet_grid(.~.sign)
  
  print(dp_sign) # print dot plots 
  gsea_KEGG_results_list$dotplots_sign[[lb]] <- dp_sign # save to list
  
}
```

```{r}
# 2026_01_12
# # horrific name but at least clear... | redo with ENTREZID ncbi instead of up
# KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3 <- gsea_KEGG_results_list
# 
# save(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3,
#      file = "output/2026_01_12_KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3.rda")
```

```{r}
# # VECTOR for labels (contrasts from MSstatsTMT analysis)
# label_names <- c("H_EE_vs_SH", "A_YNG_vs_OLD",
#                  "HxA_EE_YNG_vs_EE_OLD", "HxA_SH_YNG_vs_SH_OLD",
#                  "HxA_EE_YNG_vs_SH_YNG", "HxA_EE_OLD_vs_SH_OLD")

# OLD - 2026_01_01 - entrez one but not all comparisons (see above)
# # horrific name but at least clear... | redo with ENTREZID ncbi instead of up
# KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1 <- gsea_KEGG_results_list
# 
# # save(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1,
# #      file = "output/2026_01_01_KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1.rda")
```

```{r}
# OLD  - 2025_12_31 - used uniprot accession (deleted old code - see github)
# horrific name but at least clear...
# KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1 <- gsea_KEGG_results_list

# save(KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1,
#      file = "output/2025_12_31_KEGG_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1.rda")
```

------------------------------------------------------------------------

## GO GSEA via gseGO with nested for loop

based of for loop made for the Lichti 2014 reanalysis

### **Setup containers & prepare info for sequences**

```{r}
# # ALREADY DONE ABOVE 
# # VECTOR for labels (contrasts from MSstatsTMT analysis)

# GO_names <- c('BP', 'CC', 'MF', 'ALL') # my GO NAMES 
GO_names <- c('BP', 'CC', 'MF') # my GO NAMES 

# CONTAINER - EMPTY VECTOR
gsea_GO_results_lists <- vector(mode = "list")
```

### **Operation**

```{r}
for (lb in label_names){ 
  
 # PART 1
  geneList_working_df_go <- protein_comp_entrez %>% # use ENTREZ!!!
    filter(Label == lb) %>% 
    mutate(ranking = sign(log2FC)*(-log10(pvalue))) %>%
    group_by(ENTREZID) %>% # use ENTREZID
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
    arrange(-ranking)
  
 
  # Part 1.2
  # next, deframe into a named vector needed to supply to fgsea()
  geneList_vec_working_go <- tibble::deframe(geneList_working_df_go)
  
  geneList_vec_working_go <- geneList_vec_working_go[!is.na(names(geneList_vec_working_go))]
  
  # store in list 
  gsea_GO_results_lists$geneList_vec[[lb]] <- geneList_vec_working_go
  
  # Part 1.3 | plots fine but does not save?
  # Plot ranked gene List
  gL_plot_go <- plot(geneList_vec_working_go, main = paste0(lb, '- Ranked gene list'))
  
  print(gL_plot_go)
  
  gsea_GO_results_lists$geneList_plots[[lb]] <- gL_plot_go
  
  
  
  # PART 2 
  # nested for loop
  # so now for each label, go through all 3 go terms (BP, CC, MF)
  for (go in GO_names){ 
    
    # operation
    # run 
    GO_res_working <- clusterProfiler::gseGO(geneList = geneList_vec_working_go,
                                             OrgDb = org.Rn.eg.db, 
                                             keyType = "ENTREZID", # use entrez!
                                             ont = go,
                                             exponent = 1, # default settings for all 
                                              minGSSize = 10,
                                              maxGSSize = 500,
                                              eps = 1e-10,
                                              pvalueCutoff = 0.05,
                                              pAdjustMethod = "BH",
                                              by = "fgsea"
                                             )
    
  # save GO result original to list vec
  gsea_GO_results_lists$GO_results[[lb]][[go]] <- GO_res_working
    
    
  # extra (not really needed but save anyway
  gsea_GO_results_lists$GO_results_tib[[lb]][[go]] <- tibble::as_tibble(GO_res_working)
    

  # PART 3 - if else for dot plot to deal with any nas
  if (nrow(GO_res_working@result) > 0 ) {
    
    #Make dot plots | this one saves properly
    # Part 3.1 - regular dot plots 
    dp_go <- enrichplot::dotplot(GO_res_working,
                              showCategory = 30,
                              font.size = 11,
                              title = paste0(stringr::str_glue(
                                'Housing x Age - {lb} - {go} - 1pspg MSstatsTMT (v3.2.2) 2026_01_12 - gseGO_r'
                              )))
    print(dp_go) # print dot plots
    gsea_GO_results_lists$dotplots[[lb]][[go]] <- dp_go # save to list
    
    #Make sign dot plot 
    # Part 3.2 - regular dot plots 
    dp_go_sign <- enrichplot::dotplot(GO_res_working,
                              showCategory = 30,
                              font.size = 11,
                              split = '.sign',
                              title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - {go} - 1pspg MSstatsTMT (v3.2.2) 2026_01_12 - gseGO_r - by sign'
                              )))
    print(dp_go_sign) # print dot plots
    gsea_GO_results_lists$dotplots_sign[[lb]][[go]] <- dp_go_sign # save to list
    
  } else{print(paste0("No GO gsea result for dot plot in ", lb, "-", go, " :("))}
  
  } 
  
}
```

```{r}
# 2026_01_12
# # ENTREZID + all 8 groupings for EExAge  (2026_01_12)
# # horrific name but at least clear...
# GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2 <- gsea_GO_results_lists
# 
# save(GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2,
#      file = "output/2026_01_12_GO_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t2.rda")
```

```{r}
# OLD - (2025_12_31)
# # horrific name but at least clear...
# GO_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1 <- gsea_GO_results_lists
# 
# save(GO_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1,
#      file = "output/2025_12_31_GO_gsea_EExAge_cp_v3.4.2_msstats_v3.2.2_t1.rda")
```

*Temp Note - Finished running on Jen 31 Dec 2025*

------------------------------------------------------------------------

## Msigdbr Hallmarks

see the **BioIn_2025_09_v3.3.1_fgsea_BSS_tut.Rmd** script for more details

-   <https://yulab-smu.top/biomedical-knowledge-mining-book/027-universal-enrichment.html#msigdb-analysis>

```{r}
# # msigdbr helper functions
# msigdbr_species() # 20 species
# msigdbr_collections() # 33730584 bytes?
# unique(msigdbr_df$db_version) # check ver
# citation("msigdbr") # how to cite
```

```{r}
#using msigdbr with clusterprofiler
msigdbr_species() #get supported species 

# msigdbr rat gene sets 
msig_rat_df <- msigdbr(species = "Rattus norvegicus")

# head(m_df, 3) |> as.data.frame() # see
```

```{r}
# look around 
unique(msig_rat_df$gs_collection)
unique(msig_rat_df$gs_subcollection)
unique(msig_rat_df$gs_collection_name)

```

Just do a single for loop for now.

```{r}
hallmarks <- msig_rat_df %>% 
  filter(gs_collection == "H") %>%
  dplyr::select(gs_name, ncbi_gene)
```

```{r}
# # VECTOR for labels (contrasts from MSstatsTMT analysis)
# label_names  -> already made above 

# Container - Emty Vector
gsea_msig_hallmarks_results_list <- vector(mode = "list")
```

```{r}
for (lb in label_names) {
  # operation 
  # PART 1 - calc ranked gene list
  geneList_working_df <- protein_comp_entrez %>%   # calc ranked gene list
    filter(Label == lb) %>%
    mutate(ranking = sign(log2FC)*(-log10(pvalue))) %>%
    group_by(ENTREZID) %>%
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
    arrange(-ranking)
  
  geneList_vec_working <- tibble::deframe(geneList_working_df) # deframe in 2 vec
  
  gsea_msig_hallmarks_results_list$geneList_vec[[lb]] <- geneList_vec_working
  
  
  # PART 2 - run gsea
  msig_H_res_working <- clusterProfiler::GSEA(geneList = geneList_vec_working,
                                              TERM2GENE = hallmarks,
                                              exponent = 1,
                                              minGSSize = 10,
                                              maxGSSize = 500,
                                              eps = 1e-10,
                                              pvalueCutoff = 0.05,
                                              pAdjustMethod = "BH",
                                              by = "fgsea") # defaults 
  
  #save gsea results 
  gsea_msig_hallmarks_results_list$gsea_results[[lb]] <- msig_H_res_working
  gsea_msig_hallmarks_results_list$gsea_results_tib[[lb]] <- tibble::as_tibble(msig_H_res_working)
  

  
  # PART 3 - if else for dot plot to deal with any nas
  if (nrow(msig_H_res_working@result) > 0 ) {
    
  # PART 3 - PLOTS 
  # regular dot plots 
  dp <- enrichplot::dotplot(msig_H_res_working,
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - msigdbr Hallmark Collection - via cp_gsea dotplot - 2026_01_12 - 1pspg MSstatsTMT (v3.2.2)'))) 
  
  print(dp) 
  gsea_msig_hallmarks_results_list$dotplots[[lb]] <- dp
  
  # sign dot plots 
  # 3.2 - dot plot split by sign
  dp_sign <- enrichplot::dotplot(msig_H_res_working,
                            split = '.sign',
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - msigdbr Hallmark Collection - 2026_01_12 - via cp_gsea dotplot by sign  - 1pspg MSstatsTMT (v3.2.2)'))) +
    facet_grid(.~.sign)
  
  print(dp_sign) # print dot plots 
  gsea_msig_hallmarks_results_list$dotplots_sign[[lb]] <- dp_sign # save to list
  
  } else{print(paste0("No GO gsea result for dot plot in ", lb,  " :("))}
  
}
```

```{r}
# # horrific name but at least clear...
# gsea_msig_Hmrk_EExAge_mss3.2.2_t1_cp3.4.2_t2  <- gsea_msig_hallmarks_results_list
# 
# save(gsea_msig_Hmrk_EExAge_mss3.2.2_t1_cp3.4.2_t2,
#      file = "output/2026_01_12_gsea_msig_Hallmark_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda")
```

```{r}
# OLD - 2025_01_01 
# # horrific name but at least clear...
# gsea_msig_Hmrk_EExAge_cp3.4.2_mss3.2.2_t1  <- gsea_msig_hallmarks_results_list
# 
# # save(gsea_msig_Hmrk_EExAge_cp3.4.2_mss3.2.2_t1,
# #      file = "output/2026_01_01_gsea_msig_Hallmark_EExAge_cpv3.4.2_mssv3.2.2_t1.rda")
```

------------------------------------------------------------------------

## Misigdb Cell Type C8

```{r}
head(msig_rat_df)

# look around 
unique(msig_rat_df$gs_collection)
unique(msig_rat_df$gs_subcollection)
unique(msig_rat_df$gs_collection_name)
```

```{r}
C8_celltypes <- msig_rat_df %>% 
  filter(gs_collection == "C8") %>%
  dplyr::select(gs_name, ncbi_gene)
```

```{r}
# # VECTOR for labels (contrasts from MSstatsTMT analysis)
# label_names  -> already made above 

# Container - Emty Vector
gsea_msig_C8_ct_results_list <- vector(mode = "list")
```

```{r}
#GSEA msigdb cell types 
for (lb in label_names) {
  # operation 
  # PART 1 - calc ranked gene list
  geneList_working_df <- protein_comp_entrez %>%   # calc ranked gene list
    filter(Label == lb) %>%
    mutate(ranking = sign(log2FC)*(-log10(pvalue))) %>%
    group_by(ENTREZID) %>%
    summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
    arrange(-ranking)
  
  geneList_vec_working <- tibble::deframe(geneList_working_df) # deframe in 2 vec
  
  gsea_msig_C8_ct_results_list$geneList_vec[[lb]] <- geneList_vec_working
  
  
  # PART 2 - run gsea
  msig_C8_res_working <- clusterProfiler::GSEA(geneList = geneList_vec_working,
                                              TERM2GENE = C8_celltypes,
                                              exponent = 1,
                                              minGSSize = 10,
                                              maxGSSize = 500,
                                              eps = 1e-10,
                                              pvalueCutoff = 0.05,
                                              pAdjustMethod = "BH",
                                              by = "fgsea") # defaults 
  
  #save gsea results 
  gsea_msig_C8_ct_results_list$gsea_results[[lb]] <- msig_C8_res_working
  gsea_msig_C8_ct_results_list$gsea_results_tib[[lb]] <- tibble::as_tibble(msig_C8_res_working)
  

  
  # PART 3 - if else for dot plot to deal with any nas
  if (nrow(msig_C8_res_working@result) > 0 ) {
    
  # PART 3 - PLOTS 
  # regular dot plots 
  dp <- enrichplot::dotplot(msig_C8_res_working,
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - msigdbr C8 Cell types 2026_01_12 - via cp_gsea dotplot - 1pspg MSstatsTMT (v3.2.2)'))) 
  
  print(dp) 
  gsea_msig_C8_ct_results_list$dotplots[[lb]] <- dp
  
  # sign dot plots 
  # 3.2 - dot plot split by sign
  dp_sign <- enrichplot::dotplot(msig_C8_res_working,
                            split = '.sign',
                            showCategory = 30,
                            font.size = 11,
                            title = paste0(stringr::str_glue(
                              'Housing x Age - {lb} - msigdbr C8 Cell types 2026_01_12 - via cp_gsea dotplot by sign  - 1pspg MSstatsTMT (v3.2.2)'))) +
    facet_grid(.~.sign)
  
  print(dp_sign) # print dot plots 
  gsea_msig_C8_ct_results_list$dotplots_sign[[lb]] <- dp_sign # save to list
  
  } else{print(paste0("No gsea result for dot plot in ", lb,  " :("))}
  
}
```

```{r}
# # horrific name but at least clear...
# gsea_msig_C8_ct_EExAge_mss3.2.2_t1_cp3.4.2_t2  <- gsea_msig_C8_ct_results_list
# 
# save(gsea_msig_C8_ct_EExAge_mss3.2.2_t1_cp3.4.2_t2,
#      file = "output/2026_01_12_gsea_msig_C8_ct_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda")
```

```{r}
# OLD - 2026_01_12 (only 6 comps)
# # horrific name but at least clear...
# gsea_msig_C8_ct_EExAge_cp3.4.2_mss3.2.2_t1  <- gsea_msig_C8_ct_results_list
# 
# # save(gsea_msig_C8_ct_EExAge_cp3.4.2_mss3.2.2_t1,
# #      file = "output/2026_01_04_gsea_msig_C8_ct_EExAge_cpv3.4.2_mssv3.2.2_t1.rda")
```

------------------------------------------------------------------------

## ~~ReactomePA via ClusterProfiler (not working!)~~

Try ReactomeGSA instead (R client directly recommended by Reactome)

-   <https://yulab-smu.top/biomedical-knowledge-mining-book/024-reactome.html>
-   <https://guangchuangyu.github.io/software/ReactomePA/>

```{r}
# # test for housing EE vs SH only 
# EEvsSH_genelist <- protein_comp_entrez %>%
#   filter(Label == "H_EE_vs_SH") %>% 
#   mutate(ranking = sign(log2FC)*(-log10(pvalue))) %>%
#   group_by(ENTREZID) %>%
#   summarise(ranking = mean(ranking, na.rm = TRUE)) %>%
#   arrange(-ranking)
# 
# EEvsSH_genelist_vec <- tibble::deframe(EEvsSH_genelist)
```

```{r}
# write_excel_csv(EEvsSH_genelist,
#                 "output/2026_01_04_EEvsSH_ranked_entrez_genelist.csv")
```

```{r}
# # run ReactomePA gsea 
# Reactome_EEvsSH <- gsePathway(EEvsSH_genelist_vec, 
#                               organism = 'rat', 
#                               pvalueCutoff = 0.2, #0.2 default?
#                               pAdjustMethod = "BH", 
#                               verbose = FALSE)

```

```{r}
# Reactome_EEvsSH_tibble <- tibble::as_tibble(Reactome_EEvsSH)
```
