------------------------------------------------------------------------

---
title: "BioIn_2026_01_v3.4.2_ClusterProfiler"
author: "Stephanie Huang"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# ClusterProfiler BioIn vis - 1pspg v3.2.2 EE x Age t1

*for MSstatsTMT 1pspg v3.2.2 EE x Age t1 -(defaults) with B1 129N removed (TMT channel with too much loaded)*

This script is for running the GSEA - uses for loops.

This is a temporary script is for visualization of the results

Outputs

1.  2025_12_31

2.  

## Set up

### Packages

```{r}
# # Install if needed 
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("clusterProfiler")
# BiocManager::install("msigdbr")
# BiocManager::install("org.Rn.eg.db")
# BiocManager::install("enrichplot")
# BiocManager::install("pathview")

# draw cell for SwissBioPics 
devtools::install_github("svalvaro/drawCell")
```

```{r}
# Load Packages 
library(tidyverse) # for dplyr, ggplot, stringr, readr etc 
library(clusterProfiler) #GSEA includes gsea, enrichplot etc.
library(org.Rn.eg.db) # Rat genome anno pack- incl. AnnotationDbi
library(msigdbr) # MSigDB gene sets
library(enrichplot) # for dotplot()
library(ggridges) # for ridgeplot()
library(pathview) # pathview() - Kegg pathway visualisation

library(drawCell)
library(htmlwidgets)
library(webshot2)

library(paletteer) # colors
library(colorspace) # colors
```

### Import data

Load the mapped comparisons output generated from

```{r}
# mapped_comp_v3.2.2_EExAge_t1
load("input/protein_lists/2025_12_31_1pspg_v3.2.2_t1_EExAge_mapped_comp.rda")

df <- mapped_comp_v3.2.2_EExAge_t1
```

lil checks

```{r}
# lil checks
head(df) # peak
str(df) # structure | 7746 x 9
colnames(df) # column names

# lil checks (bit messy)
unique(df$issue) # if = na in issue column then good. if not exclude
check_1 <- df %>% filter(str_detect(log2FC, "Inf")) # 5 inf 
check_2 <- df %>% filter(str_detect(Protein, "Cont")) # 0 (since already removed)
unique(df$Label) 
length(unique(df$Label)) #8
```

### **Clean up df & rename**

```{r}
protein_comp <- df %>%
  filter(is.na(issue)) %>% # keeps rows with issue == na (so comparisons that are ok)
  filter(!str_detect(log2FC, "Inf")) # remove infinity/missing fold change values

# check 
head(protein_comp) 
colnames(protein_comp)

```

```{r}
# # check 
# unique(df$Label) 
# # don't want "HxA_EE_YNG_vs_SH_OLD" "HxA_EE_OLD_vs_SH_YNG", so just do manually 
# 
# # VECTOR for labels (contrasts from MSstatsTMT analysis)
# label_names <- c("H_EE_vs_SH", "A_YNG_vs_OLD",
#                  "HxA_EE_YNG_vs_EE_OLD", "HxA_SH_YNG_vs_SH_OLD",
#                  "HxA_EE_YNG_vs_SH_YNG", "HxA_EE_OLD_vs_SH_OLD")

label_names <- unique(protein_comp$Label)

# # CONTAINER - EMPTY VECTOR
# gsea_KEGG_results_list <- vector(mode = "list")
```

------------------------------------------------------------------------

## Testing conversion of bitr (extra)

From FASTA - uniprot uniprot acc

<https://yulab-smu.top/biomedical-knowledge-mining-book/utilities.html#sec-id-convert>

```{r}
protein_comp_uni_prot <- unique(protein_comp$Protein) # 8048 
protein_comp_uni_gene <- unique(protein_comp$gene) # 8014
```

```{r}
# Gene name 2 EntrezID | .0339 x 8104 = 272 ish
bitr_gn_2_entrez <- bitr(protein_comp_uni_gene,
                         fromType = "SYMBOL",
                         toType = "ENTREZID",
                         OrgDb = "org.Rn.eg.db") #3.39% failed to map... better
# 'select()' returned 1:1 mapping between keys and columns
#Warning message: In bitr(protein_comp_uni_gene, fromType = "SYMBOL", toType = "ENTREZID",:  3.39% of input gene IDs are fail to map...


# Seems gn 2 entrez is the superior option. 
length(unique(bitr_gn_2_entrez$SYMBOL)) # 7742
length(unique(bitr_gn_2_entrez$ENTREZID)) # 7742

# length(unique(bitr_up_2_entrez$UNIPROT)) # 7325
# length(unique(bitr_up_2_entrez$ENTREZID)) # 7392
```

```{r}
protein_comp_entrez <- left_join(protein_comp, bitr_gn_2_entrez,
                                 by = join_by(gene == SYMBOL)) %>%
  filter(!is.na(ENTREZID)) # remove non mapped - 64384 to 61952 
```

------------------------------------------------------------------------

```{r}
load("output/gsea_results/2026_01_12_KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3.rda") # KEGG

load("output/gsea_results/2026_01_12_GO_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t2.rda") # GO

# MSIGDB
load("output/gsea_results/2026_01_12_gsea_msig_Hallmark_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda") 

load("output/gsea_results/2026_01_12_gsea_msig_C8_ct_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda")
```

------------------------------------------------------------------------

## print KEGG sign dot plots

```{r fig.height=13, fig.width=11}
print(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$dotplots_sign) # kegg


# print(gsea_msig_Hmrk_EExAge_cp3.4.2_mss3.2.2_t1$dotplots) # msigdb hallmarks entrez
# print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$dotplots) # kegg entrez
```

## print GO sign dot plots

```{r}
print(GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2$dotplots_sign)
```

## print Msigdb Hallmarks sign dot plots

```{r}
print(gsea_msig_Hmrk_EExAge_mss3.2.2_t1_cp3.4.2_t2$dotplots_sign)
```

## print Msigdb C8 CellTypes sign dot plots

```{r}
print(gsea_msig_C8_ct_EExAge_cp3.4.2_mss3.2.2_t1$dotplots_sign)
```

```{r}

print(gsea_msig_Hmrk_EExAge_mss3.2.2_t1_cp3.4.2_t2$dotplots_sign$H_EE_vs_SH)
print(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$dotplots_sign$H_EE_vs_SH)

```

```{r}
print(GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2$dotplots_sign$H_EE_vs_SH)
```

```{r}
EEvsSH_kegg_tib <- KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[["kegg_results_tib"]][["H_EE_vs_SH"]]
```

```{r}
enrichplot::dotplot(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$kegg_results$H_EE_vs_SH,
                              showCategory = 50,
                              font.size = 11,
                              title = 'EE vs SH - entrez ve ',
                # subtitle = '1pspg MSstatsTMT (v3.2.2) 2025_12_31 - gseGO_r'
                    )
```

```{r}
# colnames(EEvsSH_kegg_tib)

custom_dopplot <- EEvsSH_kegg_tib %>%
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  # geom_text(aes(label = gene_name), color = "grey10", size = 2.5,
  #           position = position_stack()) +
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(2,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  
  labs(title = 'EE vs SH KEGG',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  # facet_wrap(~fraction, space = 'free_x', scale = 'free_x') +
  #theme_void() +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 9, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 9, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

custom_dopplot  
```

```{r}
YNGvsOLD_kegg_tib <- KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[["kegg_results_tib"]][["A_YNG_vs_OLD"]]
```

```{r}
custom_dopplot_2 <- YNGvsOLD_kegg_tib %>%
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  # geom_text(aes(label = gene_name), color = "grey10", size = 2.5,
  #           position = position_stack()) +
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(3,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  
  labs(title = 'YNG vs OLD KEGG',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  # facet_wrap(~fraction, space = 'free_x', scale = 'free_x') +
  #theme_void() +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 9, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 9, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

custom_dopplot_2 
```

```{r}
# FC_gam_Rm_phagocytosis rno04666
pathview(gene.data=KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$geneList_vec$A_YNG_vs_OLD,
                pathway.id="rno04666",
                species = 'rno')
```

### Pathview test

-   <https://pathview.r-forge.r-project.org/pathview.pdf>

```{r}
data(gse16873.d)
data(gse16873.d)
```

```{r}
# oxidative phosphorylation
# add defailts! png
pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
                pathway.id="rno00190",
                species = 'rno',
         out.suffix = "defaults",
         kegg.native = T,
         # same.layer = F #slower & on top (out of bounds?)
         )
```

```{r}
# neurotrophic signalling
pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
                pathway.id="rno04722",
                species = 'rno',
         out.suffix = "defaults",
         kegg.native = T,
         # same.layer = F #slower & on top (out of bounds?)
         )
```

```{r}
# graphviz!
pv.out <- pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
                pathway.id="rno00190",
                species = 'rno',
         out.suffix = "graphviz",
         kegg.native = F,
         # same.layer = F #slower & on top (out of bounds?)
         )
```

```{r}
# graphviz!
pv.out <- pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
                pathway.id="rno04722",
                species = 'rno',
         out.suffix = "graphviz",
         kegg.native = F,
         # same.layer = F #slower & on top (out of bounds?)
         )
```

```{r}
# graphviz! | neuroactive ligand signalling 
pv.out <- pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
                pathway.id="rno04082",
                species = 'rno',
         out.suffix = "graphviz",
         kegg.native = F,
         # same.layer = F #slower & on top (out of bounds?)
         )
```

------------------------------------------------------------------------

drawCell:drawCell

-   TaxonId - <https://www.ncbi.nlm.nih.gov/taxonomy/>

-   list_sl_colors (subcellular location) - <http://current.geneontology.org/ontology/external2go/uniprotkb_sl2go>

-   Code for neuron is 6072

-   export widget <https://stackoverflow.com/questions/40089207/how-to-arrange-html-widget-plots-in-a-grid-for-export-as-pdf>

-   <https://forum.posit.co/t/save-viewer-object-rendered-in-rstudio-as-image/32796/10>

-   <https://github.com/wch/webshot/issues/72>

```{r}

# pdf(file = "output/testing.123")

mycell <- drawCell::drawCell(organism_identifier = "6072",
                   list_sl_colors = list("SL0288" = "orange",
                                         "SL0191" = "#1F4861",
                                         "SL0259" = "forestgreen",
                                         "SL0260" = "#BD88A5"),
                   # width = 1500,
                   # height = 500
                   )

saveWidget(mycell, "temp.html")
webshot(url = "temp.html",
        file = "temp.pdf",
        # file = "temp.png",
        vwidth = 600,
        vheight = 300,
        delay = 10 # adding delay fixes problem (slow! otherwise blanks)
        )
```

```{r}
drawCell::drawCellOutput(mycell, width = "100%", height = "400px")
```
