------------------------------------------------------------------------

---
title: "BioIn_2026_01_v3.4.2_ClusterProfiler"
author: "Stephanie Huang"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# ClusterProfiler BioIn vis - 1pspg v3.2.2 EE x Age t1

*for MSstatsTMT 1pspg v3.2.2 EE x Age t1 -(defaults) with B1 129N removed (TMT channel with too much loaded)*

This script is for running the GSEA - uses for loops.

This is a temporary script is for visualization of the results

Outputs

1.  2025_12_31

2.  <https://yulab-smu.top/biomedical-knowledge-mining-book/02-Enrichment.html#gsea-algorithm>

3.  <https://hbctraining.github.io/Intro-to-R-flipped/lessons/05_introR-data-wrangling.html>

## Set up

### Packages

```{r}
# # Install if needed 
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("clusterProfiler")
# BiocManager::install("msigdbr")
# BiocManager::install("org.Rn.eg.db")
# BiocManager::install("enrichplot")
# BiocManager::install("pathview")

# #draw cell for SwissBioPics 
# devtools::install_github("svalvaro/drawCell")
```

```{r}
set.seed(123) # for reproducibility

# Load Packages 
library(tidyverse) # for dplyr, ggplot, stringr, readr etc 
library(purrr) # map_dfr()
library(readxl)
library(clusterProfiler) #GSEA includes gsea, enrichplot etc.
library(GOSemSim) # needed for simplify or now?
library(org.Rn.eg.db) # Rat genome anno pack- incl. AnnotationDbi
library(msigdbr) # MSigDB gene sets
library(enrichplot) # for dotplot()
library(ggridges) # for ridgeplot()
library(pathview) # pathview() - Kegg pathway visualisation

library(gganatogram) # gganatogram CO GG
library(drawCell) # drawing GO CC - SwissBioPics 
library(htmlwidgets) #
library(webshot2)

library(paletteer) # colors
library(colorspace) # colors
library(cowplot) # plot_grid

# Load required packages
library(STRINGdb)
library(igraph)
library(tidyverse)

library(writexl) # for excel tab export
```

### Import data

Load the mapped comparisons output generated from

```{r}
# mapped_comp_v3.2.2_EExAge_t1
load("input/protein_lists/2025_12_31_1pspg_v3.2.2_t1_EExAge_mapped_comp.rda")

df <- mapped_comp_v3.2.2_EExAge_t1
```

lil checks

```{r}
# lil checks
head(df) # peak
str(df) # structure | 7746 x 9
colnames(df) # column names

# lil checks (bit messy)
unique(df$issue) # if = na in issue column then good. if not exclude
check_1 <- df %>% filter(str_detect(log2FC, "Inf")) # 5 inf 
check_2 <- df %>% filter(str_detect(Protein, "Cont")) # 0 (since already removed)
unique(df$Label) 
length(unique(df$Label)) #8
```

### **Clean up df & rename**

```{r}
protein_comp <- df %>%
  filter(is.na(issue)) %>% # keeps rows with issue == na (so comparisons that are ok)
  filter(!str_detect(log2FC, "Inf")) # remove infinity/missing fold change values

# check 
head(protein_comp) 
colnames(protein_comp)

```

```{r}
# # check 
# unique(df$Label) 
# # don't want "HxA_EE_YNG_vs_SH_OLD" "HxA_EE_OLD_vs_SH_YNG", so just do manually 
# 
# # VECTOR for labels (contrasts from MSstatsTMT analysis)
# label_names <- c("H_EE_vs_SH", "A_YNG_vs_OLD",
#                  "HxA_EE_YNG_vs_EE_OLD", "HxA_SH_YNG_vs_SH_OLD",
#                  "HxA_EE_YNG_vs_SH_YNG", "HxA_EE_OLD_vs_SH_OLD")

label_names <- unique(protein_comp$Label)

# # CONTAINER - EMPTY VECTOR
# gsea_KEGG_results_list <- vector(mode = "list")
```

------------------------------------------------------------------------

## Testing conversion of bitr (extra)

From FASTA - uniprot uniprot acc

<https://yulab-smu.top/biomedical-knowledge-mining-book/utilities.html#sec-id-convert>

```{r}
protein_comp_uni_prot <- unique(protein_comp$Protein) # 8048 
protein_comp_uni_gene <- unique(protein_comp$gene) # 8014
```

```{r}
# Gene name 2 EntrezID | .0339 x 8104 = 272 ish
bitr_gn_2_entrez <- bitr(protein_comp_uni_gene,
                         fromType = "SYMBOL",
                         toType = "ENTREZID",
                         OrgDb = "org.Rn.eg.db") #3.39% failed to map... better
# 'select()' returned 1:1 mapping between keys and columns
#Warning message: In bitr(protein_comp_uni_gene, fromType = "SYMBOL", toType = "ENTREZID",:  3.39% of input gene IDs are fail to map...


# Seems gn 2 entrez is the superior option. 
length(unique(bitr_gn_2_entrez$SYMBOL)) # 7742
length(unique(bitr_gn_2_entrez$ENTREZID)) # 7742

# length(unique(bitr_up_2_entrez$UNIPROT)) # 7325
# length(unique(bitr_up_2_entrez$ENTREZID)) # 7392
```

```{r}
protein_comp_entrez <- left_join(protein_comp, bitr_gn_2_entrez,
                                 by = join_by(gene == SYMBOL)) %>%
  filter(!is.na(ENTREZID)) # remove non mapped - 64384 to 61952 
```

------------------------------------------------------------------------

```{r}
# #READ IN DATA
# KEGG
load("output/gsea_results/2026_01_12_KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3.rda") 

# GO
load("output/gsea_results/2026_01_12_GO_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t2.rda") 

# GO simplified (was prev made in this script)
load("output/gsea_results/2026_01_24_GO_gsea_EExAge_cp_v3.4.2_t2_simplified.rda")

# MSIGDB
load("output/gsea_results/2026_01_12_gsea_msig_Hallmark_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda")

# # C8
# load("output/gsea_results/2026_01_12_gsea_msig_C8_ct_EExAge_mssv3.2.2_t1_cpv3.4.2_t2.rda")
```

------------------------------------------------------------------------

## print KEGG sign dot plots

```{r fig.height=13, fig.width=11}
print(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$dotplots_sign) # kegg


# print(gsea_msig_Hmrk_EExAge_cp3.4.2_mss3.2.2_t1$dotplots) # msigdb hallmarks entrez
# print(KEGG_gsea_EExAge_cpv3.4.2t2_mssv3.2.2t1$dotplots) # kegg entrez
```

## print GO sign dot plots

```{r}
print(GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2$dotplots_sign)
```

## print Msigdb Hallmarks sign dot plots

```{r}
print(gsea_msig_Hmrk_EExAge_mss3.2.2_t1_cp3.4.2_t2$dotplots_sign)
```

## print Msigdb C8 CellTypes sign dot plots

```{r}
# print(gsea_msig_C8_ct_EExAge_cp3.4.2_mss3.2.2_t1$dotplots_sign)
```

# KEGG

## lolliplots (alt to dotplots)

```{r}
EEvsSH_kegg_tib <- KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[["kegg_results_tib"]][["H_EE_vs_SH"]]
```

```{r}
enrichplot::dotplot(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$kegg_results$H_EE_vs_SH,
                              showCategory = 50,
                              font.size = 11,
                              title = 'EE vs SH - entrez ve ',
                # subtitle = '1pspg MSstatsTMT (v3.2.2) 2025_12_31 - gseGO_r'
                    )
```

```{r}
# colnames(EEvsSH_kegg_tib)

custom_dopplot <- EEvsSH_kegg_tib %>%
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  # geom_text(aes(label = gene_name), color = "grey10", size = 2.5,
  #           position = position_stack()) +
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(2,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  labs(title = 'EE vs SH KEGG',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  # facet_wrap(~fraction, space = 'free_x', scale = 'free_x') +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 9, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 9, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

custom_dopplot  
```

```{r}
YNGvsOLD_kegg_tib <- KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[["kegg_results_tib"]][["A_YNG_vs_OLD"]]
```

```{r}
custom_dopplot_2 <- YNGvsOLD_kegg_tib %>%
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  # geom_text(aes(label = gene_name), color = "grey10", size = 2.5,
  #           position = position_stack()) +
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(3,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  labs(title = 'YNG vs OLD KEGG',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  # facet_wrap(~fraction, space = 'free_x', scale = 'free_x') +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 9, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 9, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

custom_dopplot_2 
```

```{r}
plot_grid(custom_dopplot, custom_dopplot_2, ncol = 1)
```

### Working with lists (side tangent

-   [https://stackoverflow.com/questions/32819539/proper-way-to-access-list-elements-in-r](https://stackoverflow.com/questions/32819539/proper-way-to-access-list-elements-in-r#){.uri}
-   [https://www.r-bloggers.com/2023/07/how-to-subset-list-objects-in-r](https://www.r-bloggers.com/2023/07/how-to-subset-list-objects-in-r/#){.uri}
-   <https://www.reddit.com/r/rstats/comments/sd3n8k/extract_data_from_a_list_within_a_list_that_is/>
-   <https://pratikunterwegs.github.io/tres-tidy-tutorial/working-with-lists-and-iteration.html>
-   SOLUTION ! - purrr:::map_dfr() <https://purrr.tidyverse.org/reference/map_dfr.html>

```{r}
# baseR - nope assumes list
# unlisted_df <- unlist(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$kegg_results_tib)

KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[1] # gets genelist vec (1st list set)

# gets kegg_results_tib [[3]] & then A_YNG_vs_OLD [[2]]
KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[[3]][[2]]
KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[["kegg_results_tib"]][["A_YNG_vs_OLD"]]


df_str <- KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3[["kegg_results_tib"]]

```

```{r}
# SOLUTION !!! lol it really is that easy,,, 
KEGG_tibs_df <- purrr::map_dfr(
  KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$kegg_results_tib,
  as.data.frame,
  .id = "Label") 
```

#### KEGG facet lolliplot

```{r}
# looks kinda horrifc buy honestly a good check! 
lolly_KEGG_facet <- KEGG_tibs_df  %>%
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(3,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  ylim(-4,4)+
  
  labs(title = 'All KEGG EExAge - 2026_01_21',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  facet_wrap(~Label,
             space = 'free_x', 
             ) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 10, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 10, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

lolly_KEGG_facet
```

------------------------------------------------------------------------

## Pathview test

-   <https://pathview.r-forge.r-project.org/pathview.pdf>

```{r}
# data(gse16873.d)
# data(gse16873.d)
```

```{r}
# # oxidative phosphorylation
# # add defailts! png
# pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
#                 pathway.id="rno00190",
#                 species = 'rno',
#          out.suffix = "defaults",
#          kegg.native = T,
#          # same.layer = F #slower & on top (out of bounds?)
#          )
```

```{r}
# # neurotrophic signalling
# pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
#                 pathway.id="rno04722",
#                 species = 'rno',
#          out.suffix = "defaults",
#          kegg.native = T,
#          # same.layer = F #slower & on top (out of bounds?)
#          )
```

```{r}
# # graphviz!
# pv.out <- pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
#                 pathway.id="rno00190",
#                 species = 'rno',
#          out.suffix = "graphviz",
#          kegg.native = F,
#          # same.layer = F #slower & on top (out of bounds?)
#          )
```

```{r}
# # graphviz!
# pv.out <- pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
#                 pathway.id="rno04722",
#                 species = 'rno',
#          out.suffix = "graphviz",
#          kegg.native = F,
#          # same.layer = F #slower & on top (out of bounds?)
#          )
```

```{r}
# # graphviz! | neuroactive ligand signalling 
# pv.out <- pathview(gene.data=KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec$H_EE_vs_SH,
#                 pathway.id="rno04082",
#                 species = 'rno',
#          out.suffix = "graphviz",
#          kegg.native = F,
#          # same.layer = F #slower & on top (out of bounds?)
#          )
```

------------------------------------------------------------------------

# GO

```{r}
# UNNESTED UNESTED - nested map_dfr() inside another
GO_tibs_df <- purrr::map_dfr(
  GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2$GO_results_tib,
  ~map_dfr(.x, as_tibble, .id = "GO"),
  .id = "Label") 
# YAY THIS WORKS! 
```

```{r}
# baseR subsetting checks
unique(GO_tibs_df$Description[GO_tibs_df$GO == "CC"]) # 226! 

length(unique(GO_tibs_df$Description[GO_tibs_df$GO == "CC"])) # CC - 227
length(unique(GO_tibs_df$Description[GO_tibs_df$GO == "BP"])) # BP - 368
length(unique(GO_tibs_df$Description[GO_tibs_df$GO == "MF"])) # MF - 146
```

```{r}
# looks kinda horrifc buy honestly a good check! 
lolly_GO_facet <- GO_tibs_df  %>%
  
  filter(GO == "CC") %>% # just comment out accordingly easiest
  
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(3,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  ylim(-4,4)+
  
  labs(title = 'GO BP EExAge facet - 2026_01_21',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  facet_wrap(~Label,
             space = 'free_x', 
             ) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 10, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 10, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

lolly_GO_facet
```

## GO simplify

### Top Go (maybe later)

-   <https://bioconductor.org/packages/release/bioc/vignettes/topGO/inst/doc/topGO_manual.html>
-   <https://davidzhao1015.github.io/blog/2024/go-enrichment/#5-comparing-results>

### **ClusterProfiler Simplify() (from GOSemSim orginally)**

-   <https://yulab-smu.top/biomedical-knowledge-mining-book/utilities.html#reduce-redundancy-of-enriched-go-terms>

```{r}

GO_bp_EE <- pairwise_termsim(GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2$GO_results$H_EE_vs_SH$BP)

simGO_bp_EE <- clusterProfiler::simplify(GO_bp_EE,
                          cutoff = 0.7, # defaults for signatures 'gseaResult' 
                          by = "p.adjust",
                          select_fun = min,
                          measure = "Wang",
                          semData = NULL)
                          

p1 <-  enrichplot::emapplot(GO_bp_EE)

p2 <- emapplot(simGO_bp_EE) 

plot_list(p1, p2, ncol = 2, tag_levels = "A")

```

### Run GO simplify as loop

```{r}
# SET UP FOR LOOP
GOs <- c("BP", "CC", "MF") # GO terms, make sure order matches! 

label_names <- unique(protein_comp_entrez$Label)

GO_sim_2_append <- vector(mode = "list") # empty vec
```

```{r eval=FALSE, include=FALSE}
# # nested conditional for loop for GO simplification & network plot
# for (lb in label_names) {
#   for (go in GOs) {
#     
#     # 1- PREP
#     GO_big <- GO_gsea_EExAge_msstats_v3.2.2_t1_cp_v3.4.2_t2$GO_results[[lb]][[go]]
#     
#     bps <- enrichplot::pairwise_termsim(GO_big) # prep pairwise term sim
#     
#     GO_sim_2_append$pw_sim_GO_terms[[lb]][[go]] <- bps # save term sim  
#     
#     
#     # 2- RUN SIMPLIFY
#     bps_2 <- clusterProfiler::simplify(bps, 
#                           cutoff = 0.7, # defaults for signatures 'gseaResult' 
#                           by = "p.adjust",
#                           select_fun = min,
#                           measure = "Wang",
#                           semData = NULL)
#     
#     GO_sim_2_append$sim_GO_res[[lb]][[go]] <- bps_2 #save simplified GO res
#     GO_sim_2_append$sim_GO_res_tib[[lb]][[go]] <- tibble::as_tibble(bps_2) # tibble
#     
#     
#     # 3 - PLOTS 
#     p1 <- enrichplot::emapplot(bps) # bps - emapplot 1 (b4 simplification)
#     p2 <- enrichplot::emapplot(bps_2)  # bps_2 - emapplot 2 (after simplification)
#     
#     plot_list(p1, p2, ncol = 2, tag_levels = 2) # doesn't plot??? & doesn't save
#     
#     GO_sim_2_append$emapplot_1[[lb]][[go]] <- p1 # save plot 1 bps
#     GO_sim_2_append$emapplot_2_sim[[lb]][[go]] <- p2  # save plot 2 bps_2
#     
#   }
# }
```

```{r}
# #SAVING
# #GO_gsea_EExAge_mss_v3.2.2_t1_cp_v3.4.2_t2 made from this
# #the data below is only the simplied (in the future run everything together)
# GO_gsea_EExAge_cp_v3.4.2_t2_simplified <- GO_sim_2_append
# 
# save(GO_gsea_EExAge_cp_v3.4.2_t2_simplified,
#      file = "output/2026_01_24_GO_gsea_EExAge_cp_v3.4.2_t2_simplified.rda")
```

```{r}
# all in 1 DF for facet plotting!!! 
# UNNESTED UNNESTED FOR NESTED NESTED df! - nested map_dfr() inside another
GO_sim_res_tibs_df <- purrr::map_dfr(
  GO_gsea_EExAge_cp_v3.4.2_t2_simplified$sim_GO_res_tib,
  ~map_dfr(.x, as_tibble, .id = "GO"),
  .id = "Label")
```

```{r}
# baseR subsetting checks | prev CC 227, BP 368, MF 146
length(unique(GO_sim_res_tibs_df$Description[GO_sim_res_tibs_df$GO == "CC"])) # 130
length(unique(GO_sim_res_tibs_df$Description[GO_sim_res_tibs_df$GO == "BP"])) # 193
length(unique(GO_sim_res_tibs_df$Description[GO_sim_res_tibs_df$GO == "MF"])) # 73
```

```{r}
# looks kinda horrifc buy honestly a good check! 
lolly_GO_sim_facet <- GO_sim_res_tibs_df  %>%
  
  filter(GO == "MF") %>% # just comment out accordingly easiest
  
  ggplot(aes(x= reorder(Description, NES), 
             y = NES,
             fill = NES)) +
  geom_segment(aes(x = Description,
                   yend = NES, y = 0,
                   alpha = p.adjust)) +
  geom_point(aes(alpha = p.adjust, size = setSize),
             shape = 21) +
  scale_alpha(range = c(.9,.3), guide = 'none') + # rev alpha so darker = more sig
  geom_hline(yintercept = 0, color = 'grey20', linewidth = 0.1) +
  scale_color_manual(breaks = c("Up", "Down"),
                     # values = c("#dc322f", "#268bd2"),
                     guide = 'none') + 
  scale_size_continuous(range = c(3,9)) + # map size oflolly (set size)
  
  # scale_fill_continuous_diverging("Blue-Red 3") + # colorspace
  scale_fill_continuous_diverging("Blue-Red 2") +
  
  ylim(-4,4)+
  
  labs(title = 'GO simplified MF - EExAge facet - 2026_01_24',
       y = 'NES',
       x = 'KEGG PATH') +
  coord_flip() + #fliptheplot
  facet_wrap(~Label,
             space = 'free_x', 
             ) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1,
                               vjust = 0.5, # 90, hj1 t/R, vj.5 cent
                               size = 10, face = 'italic'), 
    axis.text.y = element_text(hjust = 1, color = "grey20",
                               size = 10, face = 'italic'), 
    axis.title.y = element_blank(),
    legend.position = "none"
        ) 

lolly_GO_sim_facet
```

```{r}
# quick checks
emapplot(GO_gsea_EExAge_cp_v3.4.2_t2_simplified$sim_GO_res$H_EE_vs_SH$BP)
emapplot(GO_gsea_EExAge_cp_v3.4.2_t2_simplified$sim_GO_res$H_EE_vs_SH$CC)
emapplot(GO_gsea_EExAge_cp_v3.4.2_t2_simplified$sim_GO_res$H_EE_vs_SH$MF)
```

------------------------------------------------------------------------

## GO CC mapping w gganatogram

<https://github.com/jespermaag/gganatogram/tree/master?tab=readme-ov-file#readme>

Cite Thul <https://www.science.org/doi/10.1126/science.aal3321>

```{r}
#gganatogram
cell_key #gganatogram cell structures key

# plot everthing
gganatogram(data=cell_key[['cell']], outline = T, fillOutline='steelblue', organism="cell", fill="colour")  +theme_void()   + coord_fixed()

```

```{r}
# eehh terms are kinda hard to automatically mapped maybe just do manual anno based on most affected! 
# need to have it as cell_key$cell as it is indexed cell_key[['cell']] above

cells_plot <- cell_key$cell %>%
  filter(organ %in% c("mitochondria", "actin_filaments",
                      "nuclear_membrane", "microtubule_organizing_center",
                      "plasma_membrane", "nuclearplasm", 
                      "nucleoli", "nuclear_bodies")) %>%
  gganatogram(outline = T, fillOutline='grey95', organism="cell", fill="colour")  +
  theme_void()   + 
  coord_fixed()

cells_plot

```

------------------------------------------------------------------------

## GO CC mapping w drawCell SwissBioPics

2026_01_23 - ehhh honestly would be easier to do manual or maybe looking into API queries myself. Did manage to get a plot but not much mapped so kinda anticlimactic...

drawCell:drawCell

-   TaxonId - <https://www.ncbi.nlm.nih.gov/taxonomy/>

-   list_sl_colors (subcellular location) - <http://current.geneontology.org/ontology/external2go/uniprotkb_sl2go>

    -   <https://www.uniprot.org/locations?query=>\*

-   Code for neuron is 6072

-   export widget <https://stackoverflow.com/questions/40089207/how-to-arrange-html-widget-plots-in-a-grid-for-export-as-pdf>

    -   <https://forum.posit.co/t/save-viewer-object-rendered-in-rstudio-as-image/32796/10>

    -   <https://github.com/wch/webshot/issues/72>

```{r}
# read in mapped subcellelular infor from uniprot
up_subcell_raw <- read_excel(
  "input/UP_anno/UP_subcellular_locations_all_2026_01_23.xlsx")
#   mutate(hashtags = str_split(Keywords, ";")) %>%
#   tidyr::unnest(hashtags) # 22696

# note alot here don't have GOID anyway
colnames(up_subcell_raw)
str(up_subcell_raw)
```

```{r}
up_subcell <-up_subcell_raw %>%
  # tidyr::separate_wider_delim(
  #   cols = `Gene Ontologies`,
  #   delim = ":", # split by : nooo has 2 : : so does not work :(
  #   names = c("GO_ID", "GO_Name"))
  
  tidyr::separate_wider_regex( # use this instead! 
    `Gene Ontologies`,
    patterns = c(go_id = "GO:\\d+", # matches GO:0002079:
                 ":",
                 go_term = ".*"), # matches any char at end of string
    cols_remove = FALSE) %>%
  mutate(subcell_ID = str_remove(`Subcellular location ID`, "-"),
         .after = `Subcellular location ID`) 
# omg turns out ID frm uniprot was "SL-0181" rather than "SL0181"
```

```{r}
GO_tibs_CC_EE <- GO_tibs_df %>%
  filter(Label == "H_EE_vs_SH" & GO == "CC") %>%
  left_join(up_subcell,
            by = join_by(ID == go_id)) %>% # heaps not mapped!
  filter(!is.na(subcell_ID)) %>%
  mutate(Direction = if_else(NES <0 , "Down", "Up")) 
```

```{r}
down_cols <- paletteer_dynamic("cartography::wine.pal", 15)
up_cols <- paletteer_dynamic("cartography::turquoise.pal", 15)
```

```{r}
# tidyerse AI
GO_tibs_CC_EE_cols <- GO_tibs_CC_EE %>%
  arrange(Direction, NES) %>% # sort by direc & then score
  group_by(Direction) %>% # group by dir
  mutate( # group wise calcs! so for each group xyz...
    idx = row_number(),
    colour = case_when(          
      Direction == "Up"   ~ down_cols[idx],
      Direction == "Down" ~ up_cols[idx]
    )
  ) %>%
  ungroup() #%>%
  # dplyr::select(-idx)
```

```{r}
# EE_color_CC_list <- GO_tibs_CC_EE_cols %>%
#   dplyr::select(subcell_ID, colour) %>%
#   tibble::deframe() # make named list 


# color_map_EE_CC <- GO_tibs_CC_EE_cols$val_4_dc
# neuron "6072" | "10116", # rat

# works but looks dissapointing...
my_neuron_EE <- drawCell::drawCell(organism_identifier = "6072", # neuron
                                   list_sl_colors = EE_color_CC_list,
                                   width = 1500,
                                   height = 500)

saveWidget(my_neuron_EE, "my_neuron_EE_v1.html")
webshot(url = "my_neuron_EE_v1.html",
        file = "my_neuron_EE_v1.pdf",
        # file = "temp.png",
        vwidth = 600,
        vheight = 300,
        delay = 10 # adding delay fixes problem (slow! otherwise blanks)
        )


my_cell_EE <- drawCell::drawCell(organism_identifier = "10116", # rat
                                   list_sl_colors = EE_color_CC_list,
                                   width = 1500,
                                   height = 500)


saveWidget(my_cell_EE, "my_cell_EE_v1.html")
webshot(url = "my_cell_EE_v1.html",
        file = "my_cell_EE_v1.pdf",
        # file = "temp.png",
        vwidth = 600,
        vheight = 300,
        delay = 10 # adding delay fixes problem (slow! otherwise blanks)
        )

```

```{r}
# mycell <- drawCell::drawCell(organism_identifier = "6072",
#                    list_sl_colors = list("SL0288" = "orange",
#                                          "SL0191" = "#1F4861",
#                                          "SL0259" = "forestgreen",
#                                          "SL0260" = "#BD88A5"),
#                    # width = 1500,
#                    # height = 500
#                    )
# 
# saveWidget(mycell, "temp.html")
# webshot(url = "temp.html",
#         file = "temp.pdf",
#         # file = "temp.png",
#         vwidth = 600,
#         vheight = 300,
#         delay = 10 # adding delay fixes problem (slow! otherwise blanks)
#         )

# drawCell::drawCellOutput(mycell, width = "100%", height = "400px")
```

```{r}
# paleteer
paletteer_d("nord::aurora")

RColorBrewer::display.brewer.all()

# Paletteer prints hexcode with color in console!!!! 
paletteer_d("RColorBrewer::RdYlGn")
paletteer_d("RColorBrewer::Spectral")
paletteer_d("RColorBrewer::PiYG")
paletteer_d("RColorBrewer::BrBG")
paletteer_d("RColorBrewer::Pastel2")
paletteer_d("RColorBrewer::Dark2")
paletteer_d("rcartocolor::Temps")
paletteer_d("rcartocolor::Geyser")
paletteer_d("rcartocolor::Sunset")
paletteer_d("rcartocolor::Emrld")
paletteer_d("rcartocolor::OrYel")
paletteer_dynamic("cartography::orange.pal", 15)
paletteer_dynamic("cartography::green.pal", 15)
paletteer_dynamic("cartography::blue.pal", 15)
paletteer_dynamic("cartography::red.pal", 15)
paletteer_dynamic("cartography::wine.pal", 15)
paletteer_dynamic("cartography::turquoise.pal", 15)

palettes_dynamic_names
```

------------------------------------------------------------------------

# Protein Interactions - w StringDB & igraph

tutorials

-   <https://www.youtube.com/watch?v=JEaNhUq_8rA>

-   <https://almeidasilvaf.github.io/NASB/chapters/04_analysis_of_PPI_networks.html>

-   <https://rpubs.com/HWH/913747>

-   <https://string-db.org/cgi/access?footer_active_subpage=archive>

Running test/tutorial

```{r}
# EXAMPLE DATA
data("diff_exp_example1")
head(diff_exp_example1)
```

```{r}
# SET UP
# human 9606, mice 10090, rat 10116
string_db <- STRINGdb$new(
  version = "12.0", # current since 2023
  species = 10116, # rat
  score_threshold = 400, # min score (0-1000, def 400)
  network_type = "full" # full, functional, physical etc
)
```

```{r}
# NICE DOCUMENTATION!!! YAY
# see all possible args of STRINGdb$nnew()
STRINGdb$help()

# see list of methods associated with STRINGdb class
STRINGdb$methods()

STRINGdb$help("get_graph") # get helpd documentation of each method
```

```{r}
# RUN 
example1 <- string_db$map(
  diff_exp_example1,
  "gene",
  removeUnmappedRows = TRUE
  ) 

head(example1)
```

```{r}
# PLOT NETWORK
top200 <- example1$STRING_id[1:200] 

string_db$plot_network(top200)
```

### 4.3 - Interacting w the STRING server

-   <https://almeidasilvaf.github.io/NASB/chapters/04_analysis_of_PPI_networks.html#interacting-with-the-string-server>

-   Payload mechism of STRING (adds colored border halo to nodes based on a specified variable) - here send data to STRING serves & use output as

```{r}
# Remove non-DE genes & add `color` column
example1_filtered <- string_db$add_diff_exp_color(
  example1[example1$pvalue < 0.05, ], # filter by sig
  logFcColStr = "logFC" # coloring by logFC
)

head(example1_filtered)
```

```{r}
# post payload info to STRING server
pid <- string_db$post_payload(
  example1_filtered$STRING_id, 
  colors = example1_filtered$color
)

pid
```

```{r}
# visualise network with borders ("halo")
string_db$plot_network(top200, # used the top 200 from before
                       payload_id = pid) # now plot sig ones on top of that! 
```

###  Find PPI Clusters

-   <https://almeidasilvaf.github.io/NASB/chapters/04_analysis_of_PPI_networks.html#find-ppi-clusters>

-   To detect clusters in the PPI network, use get_clusters() method, which in turn uses the clustering algorithim in cluster\_ in igraph functions - see `??igraph::cluster_`Â 

```{r}
# find clusters using only the 1st 600 genes
clusters <- string_db$get_clusters(example1$STRING_id[1:600])

clusters[1:4]
```

```{r}
# plot 1st 4 clusters
par(mfrow = c(2,2))
for(x in seq_len(4)) {
  string_db$plot_network(clusters[[x]])
}

par(mfrow = c(1,1)) # back to original number of rows and cols
```

more stuff like extracting info from proteins of interest & extracting homologous proteins (skipped for now)

## Now test with H_EE_vs_SH

```{r}
example2 <- string_db$map(
  protein_comp[Label == "H_EE_vs_SH"],
  "Protein",
  removeUnmappedRows = TRUE
  ) %>% 
  dplyr::arrange(example2$adj.pvalue) # arrange by smallest adj.p at top

head(example2)
```

```{r}
# PLOT NETWORK
top200_EEvsSH <- example2$STRING_id[1:200]

string_db$plot_network(top200_EEvsSH)
```

```{r}
# String Payload Mech
# Remove non-DE genes & add `color` column
example2_filtered <- string_db$add_diff_exp_color(
  example2[example2$adj.pvalue < 0.05, ], # filter sig adj.p < 0.05
  logFcColStr = "log2FC" # map log2FC to color
)

head(example2_filtered)
```

```{r}
# Post payload information to STRING server
pid2 <- string_db$post_payload(
  example2_filtered$STRING_id,
  colors = example2_filtered$color
 )

pid2

# plot payload infor (halo) with prev network
string_db$plot_network(top200_EEvsSH, payload_id = pid2)
```

### Running Overrepresentation analyses (with EE vs SH)

```{r}
# Using only genes in `example1` as background
bg <- unique(example2$STRING_id)

# set bg universe to observed genes only before running enrichment 
string_db$set_background(bg)


enrichment2 <- string_db$get_enrichment(top200_EEvsSH)
head(enrichment2)

# interesting enrichment (no bg from own), has way more sig & many synapse ones
# whereas enrichment2 (with bg of only observed), has fewer & no synapse
# so not that many synapse ones observed in actual proteomics sample
# LOOK INTO THIS
```

```{r}
# OR just get functional annotations (without running ORA)
annot <- string_db$get_annotations(top200_EEvsSH)

head(annot)

# SUPER USEFUL - can inform on bias in proteins observed in samples!
# e.g. see number of genes / ratio of set columns from result!


# although only gets top 2000
top_2000 <- example2$STRING_id[1:2000]

annot_2000 <- string_db$get_annotations(top_2000)

head(annot_2000) 
# NVM not that helpful idk, maybe look into later
```

```{r}
clusters2 <- string_db$get_clusters(example2$STRING_id[1:600])

clusters2[1:9]

par(mfrow = c(3,3)) 
for (x in seq_len(9)) {
  string_db$plot_network(clusters[[x]])
}

par(mfrow = c(1,1))
```

### detour to play around with web ver of string-db

note stringdb ranked analysis actually uses full list (e.g. entrez ranked genelist)

-   Permalink to EEvsSH analysis on StringDB - <https://version-12-0.string-db.org/cgi/globalenrichment?networkId=b5VSFSgnPE4Z>

```{r}
# get 250 most significant (smallest adjp for each label/comparison set)
protein_comp_entrez_top_250 <- protein_comp_entrez %>%
  ungroup() %>%
  group_by(Label) %>%
  slice_min(adj.pvalue, n = 250, with_ties = FALSE) # need with_ties = FALSE otherwise too many! 

# REMEMBER ENTREZ LIST IS SLIGHTLY SHORTED SINCE SOME WERE NOT MAPPABLE! 
```

```{r}
# # export df for web analysis
# write_excel_csv(protein_comp_entrez_top_250,
#          "output/2026_01_24_AgexEE_entrez_comps_top_250_sig_4_web_v1.csv")
```

#### Make & Export as tabs in excel!

-   Actually make a bunch of tab in excel with lists for each label (easier than way!)

-   btw look at differences with Uniprot org vs entrez (possible minor differences) but using the protein_comp_entrez df just to be consistent with other analyses!

```{r}
# top 250 entrez mapped 
split_protein_comp_entrez_top_250 <- split(protein_comp_entrez_top_250,
                                           protein_comp_entrez_top_250$Label)

# # write out excel! 
# write_xlsx(split_protein_comp_entrez_top_250, 
#            "output/2026_01_24_AgeXEE_split_protein_comp_entrez_top_250.xlsx")
```

-   <https://stats.oarc.ucla.edu/r/library/r-library-advanced-functions/>

```{r}
# TAKING DATA STORED INTHE KEGG .rda 
# convert the stored genelist vecs in mega lists in 2 lists for export

# only saves values not row name! 
# genelists_lists <- lapply(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec, as.data.frame)

# genelists_lists <- lapply(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec, function(x) {
#                           as.data.frame(row.names = names(x))
#       }) # NOPE this doesn't work

genelists_lists <- lapply(KEGG_gsea_EExAge_mssv3.2.2t1_cpv3.4.2t3$geneList_vec, function(x) {
  tibble::enframe(x)}) # easy way this works!!! yay! 


# write_xlsx(genelists_lists,
#            "output/2026_01_24_AgexEE_cpv3.4.2t3_entrez_ranked_genelists.xlsx")
```

```{r}
# adj p<0.05 sig only 
protein_comp_entrez_sig_adjp05 <- protein_comp_entrez %>%
  ungroup() %>%
  group_by(Label) %>%
  filter(adj.pvalue < 0.05)

split_protein_comp_entrez_sig_adjp05 <- split(protein_comp_entrez_sig_adjp05,
                                              protein_comp_entrez_sig_adjp05$Label)

#  # write out excel! 
# write_xlsx(split_protein_comp_entrez_sig_adjp05, 
#            "output/2026_01_24_AgeXEE_split_protein_comp_entrez_sig_adjp05.xlsx")
```

```{r}
# adj p<0.1 sig only 
protein_comp_entrez_sig_adjp1 <- protein_comp_entrez %>%
  ungroup() %>%
  group_by(Label) %>%
  filter(adj.pvalue < 0.1)

split_protein_comp_entrez_sig_adjp1 <- split(protein_comp_entrez_sig_adjp1,
                                              protein_comp_entrez_sig_adjp1$Label)

#  # write out excel!
# write_xlsx(split_protein_comp_entrez_sig_adjp1,
#            "output/2026_01_24_AgeXEE_split_protein_comp_entrez_sig_adjp1.xlsx")
```
