---
title: "MSstatsTMT_2025_10_v3.1.6_plotting"
output: html_document
---

Load Packages

```{r}
# ## Install missing packages
# if (!require("BiocManager", quietly = TRUE))
#   install.packages("BiocManager")
# bio_pkgs <- c("biomaRt", "Biostrings")
# for (pkg_i in bio_pkgs) {
#   if (!require(pkg_i, quietly = T, character.only = T))
#     BiocManager::install(pkg_i)
# }
# if (!require("remotes", quietly = T)) install.packages("remotes")
# if (!require("MSnID", quietly = T, character.only = T))
#   remotes::install_github("PNNL-Comp-Mass-Spec/MSnID@pnnl-master")
# ## ------------------------
# library(biomaRt) # ID conversion
# library(Biostrings) # read FASTA files
# library(MSnID) # parse_FASTA_names

# BiocManager::install("pcaMethods")
```

```{r}
library(tidyverse)
library(ggh4x) # facet_nested - extensions of facet_grid for nesting
library(biomaRt) # ID conversion
library(Biostrings) # read FASTA files
library(MSnID) # parse_FASTA_names
library(pcaMethods) # for various pca()
```

TO DO - Import Data - Merge - Append Uniprot - Plot Volcanos - Comparisons

```{r}
# read in rda 
# can do manually 
# Environment > Load Workspace > Click on .rda file to load | or...
load("input/MSstatsTMT/psum/2025_09_12_rp_1pspg_Protein_Summarization_v3.1.5_t8.rda")

# assign Protein level data frame to psum_t8 
psum_t8 <- quant.msstats.t8$ProteinLevelData
```

Import annotation file

```{r}
anno_df <- read_csv("input/anno/Steph_TMT_all_EE&SH_F&M_final_annotation_file_v2_2025_08.csv")
```

```{r}
anno <- anno_df %>%
  filter(!(Condition == "Norm")) %>% # remove norm 
  mutate(Age = ifelse(PND <100, "YNG", "OLD")) %>% # add age col
  dplyr::select(!(c(Run, TechRepMixture, Fraction))) %>% # rm run info besides mixture
  unique() # get unique - so filter only 56 epxperimental sample
```

```{r}
# write_csv(anno, "output/2025_10_07_annotation_56_samples_only_v2.csv")
```

FASTA

-   [Conversion Using FASTA Headers](https://pnnl-comp-mass-spec.github.io/proteomics-data-analysis-tutorial/conversion-using-fasta-headers.html)

-   don't need to do fst_path \<- system.file step as not finding df from package

-   <https://www.uniprot.org/help/fasta-headers>

    -   <https://www.uniprot.org/help/gene_name>

```{r}
# now throwing error?!?!?! why it worked before!?!?!
# read in fasta used for PD search & parse into df
# fasta_1pspg_rat <- MSnID::parse_FASTA_names(path_to_FASTA = "input/db/UP000002494_10116_Rat_Ref_Proteome_22403_1pspg_2025_04.fasta")
```

```{r}
# write out the parsed-a fast-a df into a csv!
# write_csv(fasta_1pspg_rat, "output/general/UP000002494_10116_Rat_Ref_Proteome_22403_1pspg_2025_04_UPfastadf.csv" )
```

```{r}
#nvm maybe import instead 
my_fasta <- read_csv("input/db/UP000002494_10116_Rat_Ref_Proteome_22403_1pspg_2025_04_UPfastadf.csv")
```

-   

-   [https://r4ds.hadley.nz/joins.html](https://r4ds.hadley.nz/joins.html#introduction){.uri}

```{r}
psum_fasta <- psum_t8 %>%
  filter(!str_detect(Protein, "Cont_")) %>% # remove contaminants drops by 1500
  left_join(my_fasta, 
            join_by(Protein == uniprot_acc))
```

```{r}
length(unique(psum_fasta$gene)) #7700 - seem to be gene names + ensembl ones??? 
```

```{r}
unmatched_df <- psum_fasta %>%
  filter(is.na(gene))
```

```{r}
unique(unmatched_df$description)
length(unique(unmatched_df$description)) # 22 unmatched! okay seems better. 
```

\~ \~ \~ \~ \~ \~ \~ \~

All 3 dfs

```{r}
psum_fasta_anno_t8 <- left_join(psum_fasta, anno, 
                       join_by(Mixture, Channel, # can prob join by less
                               BioReplicate, Condition))
```

```{r}
# save combined psum df 
# write_csv(psum_fasta_anno_t8, "2025_10_10_mss_v3_t8_ee_vs_sh_psum_anno_fasta.csv")
# save(psum_fasta_anno_t8, file = "2025_10_10_mss_v3_t8_ee_vs_sh_psum_anno_fasta.rda")
```

\~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~

PCA plot

-   <https://www.bioconductor.org/packages/release/bioc/html/pcaMethods.html>

-   See pg 5 of docu

```{r}
# help("bpca") # to get detailed info on pcaMethods function
data(metaboliteData) # see example of pca data
data(metaboliteDataComplete) # has metabolite rows & ID/subject columns
```

\~ \~ \~ \~ \~ \~ \~ \~ \~

```{r}
prots <- psum_fasta_anno_t8 %>%
  mutate(rcbr = paste0(Run, '_', Channel, '_', BioReplicate),
         .after = Channel) # make comb var
```

```{r}
#quick check 
unique(prots$rcbr)
length(unique(prots$rcbr)) # 168 = 16 x 4 x 3

# similar to Sam's MSstats_PD_SS.Rmd
# https://tidyr.tidyverse.org/reference/pivot_wider.html - see station fish seen
prot_mat <- prots %>%
  dplyr::select(rcbr, Protein, Abundance) %>% # select cols for matrix
  pivot_wider(names_from = rcbr, # speads id rcbr to cols
              values_from = Abundance) %>% # & protname as rows & then abun = val
  as.data.frame() # otherwise tibble does not allow row names argh

# set proteins as the row names
rownames(prot_mat) <- prot_mat$Protein

# set proteins as the row names
prot_mat <- prot_mat[, -1] %>% # take minus Protins col (1st col) 
  as.matrix()

dim(prot_mat) #7725 x 168 - 1297900 elements! OR 168 x 7725
```

```{r fig.height=7, fig.width=11}
# or transpose?!?! see data(metaboliteData) - mat <- prep(t(metaboliteData))
# prot_mat_2 <- prep(t(prot_mat))
pc_t <- pca(t(prot_mat), method="nipals", nPcs=2) # transposed (so 168 scores)

# score & loadings plot
slplot(pc_t,
       pcs=c(1,2), 
       scoresLoadings=c(TRUE,TRUE))
```

```{r}
# example with ggplot iris pca annotated
# pc <- pca(iris)
# irdf <- merge(iris, scores(pc), by=0)
# ggplot(irdf, aes(PC1, PC2, colour=Species)) +
#   geom_point() +
#   stat_ellipse()
```

```{r}
# prep labeling with the rcbr col
prots_anno <- prots %>%
  dplyr::select(Mixture:rcbr, Name:Age) %>%
  unique()

rownames(prots_anno) <- prots_anno$rcbr # needed for by = 0 in merge to work
```

```{r}
# Do this way instead. for some reason the doing it directly results in 28224 rows
pca_scores <- scores(pc_t)

merged_pca <- merge(prots_anno, pca_scores, by = 0) # needs row names set for by = 0 
```

```{r fig.height=7, fig.width=7}
ggplot(merged_pca,
       aes(x = PC1, y = PC2,
           color = Mixture,
           fill = Housing)) +
  geom_point(shape = 21) +
  stat_ellipse() +
  theme_bw()
```

\~ \~ \~ \~ \~ \~ \~

```{r}
# pcaMethods intro pg 5 
# center the data matrix
prot_mat_d <- prep(prot_mat,
                   scale = "none",
                   center = TRUE)
```

```{r}
#listPcaMethods() # to see methods
# run specified pca by pca() wrapper function
# Nipals (Nonlinear Estimation by Iterative Partial Least Squares)
res_pcaNipals <- pca(prot_mat_d, method = "nipals", # nipals for missing vals
                  center = FALSE,
                  nPcs=5)
```

```{r fig.height=7, fig.width=11}
plotPcs(res_pcaNipals, pc=1:5, type="score")

```

```{r fig.height=7, fig.width=11}
slplot(res_pcaNipals,
       pcs=c(1,2), 
       scoresLoadings=c(TRUE,TRUE))
```

```{r}
plot(res_pcaNipals)
```

\~ \~ \~ \~ \~ \~ \~ \~ \~ \~

remove missing values

```{r}
# get complete cases only
# https://tidyr.tidyverse.org/reference/drop_na.html
prot_mat_cc <- prot_mat %>%
  as.data.frame() %>% # back to df
  drop_na() # tidyr drop_na to keep only complete rows
```

```{r}
pca_cc <- pca(t(prot_mat_cc))

# pc_t <- pca(t(prot_mat), method="nipals", nPcs=2) # transposed (so 168 scores)
```

```{r fig.height=7, fig.width=7}
# Do this way instead. for some reason the doing it directly results in 28224 rows
pca_scores_cc <- scores(pca_cc)

merged_pca_cc <- merge(prots_anno, pca_scores_cc, by = 0) # row names set for by = 0 

ggplot(merged_pca_cc,
       aes(x = PC1, y = PC2,
           color = Mixture,
           fill = Housing)) +
  geom_point(shape = 21) +
  stat_ellipse() +
  theme_bw()

```

\~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~

```{r}
# gria <- psum_t8_a %>%
#   filter(str_detect(Protein, "P1949")) 

#save gria file
# write_csv(gria, "output/2025_10_09_v3_t8_gria_proteins.csv")
```

```{r}
# gria1 <- gria %>%
#   filter(Protein == "P19490")

# write_csv(gria1, "output/2025_10_09_v3_t8_gria1.csv")

# ~ ~ ~ ~ ~ ~ ~ ~

# gria4 <- gria %>%
#   filter(Protein == "P19493")

# write_csv(gria4, "output/2025_10_09_v3_t8_gria4.csv")
```

```{r}
gria <- psum_fasta_anno_t8 %>%
  filter(str_detect(gene, "Gria"))
```

```{r fig.height=9, fig.width=7}
gria_plots <- gria %>%
  ggplot(aes(x = interaction(Cohort, Condition, lex.order = TRUE), 
             y = Abundance,
             fill = Condition)) +
  geom_boxplot(color = "grey10") +
  geom_point(shape = 1) +
  scale_fill_manual(values = c("#b6d7a8", "#ffe599")) +
  facet_wrap(vars(gene), ncol = 1,
             scales = 'free_y',
             space = 'free_y') +
  theme_bw() +
  theme(legend.position = "bottom")

gria_plots
```

```{r}
hmgcs1 <- psum_fasta_anno_t8 %>%
  filter(str_detect(Protein, "P17425")) %>%
   ggplot(aes(x = interaction(Mixture, Condition, lex.order = TRUE), 
             y = Abundance,
             fill = Condition)) +
  geom_boxplot(color = "grey10") +
  geom_point(shape = 1) +
  scale_fill_manual(values = c("#b6d7a8", "#ffe599")) +
  facet_wrap(vars(Protein), ncol = 1,
             scales = 'free_y',
             space = 'free_y') +
  theme_bw() +
  theme(legend.position = "bottom")

hmgcs1
```

```{r}
# hmgcs1_df <- psum_fasta_anno_t8 %>%
#   filter(str_detect(Protein, "P17425")) 

# write_csv(hmgcs1_df, "output/2025_10_09_v3_t8_hmgcs1.csv")
```

```{r}
Mapk3 <- psum_fasta_anno_t8 %>%
  filter(str_detect(Protein, "P21708")) %>%
   ggplot(aes(x = interaction(Mixture, Condition, lex.order = TRUE), 
             y = Abundance,
             fill = Condition)) +
  geom_boxplot(color = "grey10") +
  geom_point(shape = 1) +
  scale_fill_manual(values = c("#b6d7a8", "#ffe599")) +
  facet_wrap(vars(Protein), ncol = 1,
             scales = 'free_y',
             space = 'free_y') +
  theme_bw() +
  theme(legend.position = "bottom")

Mapk3
```

```{r}
Mapk3_df <- psum_fasta_anno_t8 %>%
  filter(str_detect(Protein, "P21708")) 

# write_csv(Mapk3_df, "output/2025_10_09_v3_t8_Mapk3.csv")
```

```{r}
Syt12 <- psum_fasta_anno_t8 %>%
  filter(str_detect(Protein, "P97610")) %>%
   ggplot(aes(x = interaction(Mixture, Condition, lex.order = TRUE), 
             y = Abundance,
             fill = Condition)) +
  geom_boxplot(color = "grey10") +
  geom_point(shape = 1) +
  scale_fill_manual(values = c("#b6d7a8", "#ffe599")) +
  facet_wrap(vars(Protein), ncol = 1,
             scales = 'free_y',
             space = 'free_y') +
  theme_bw() +
  theme(legend.position = "bottom")

Syt12
```
