---
title: "MSstatsTMT_2025_09_v3.1.4"
author: "Stephanie Huang"
date: "2025-09-11-12"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Checking PD 2.4's As Batch Function.

### Background

This is a part of the re-analysis of my EE-exp LCMS data (96 raw files) collected from 2021-2022 for my PhD write-up (2025-08 to 2025-09). In this re-analysis series, I tested different UniProt databases/FASTAs in PD2.4, including using the UniProt rat reference proteome (1 protein sequence per gene 1pspg - 22400s), Swiss-Prot (8215s - cannonical only), and Swiss-Prot with isoforms (9878s). As expected, differences prevail as a result of using different search databases. I compared this to an analysis done in Nov 2024 for a conference poster (ANS), which used identical search parameters except a slightly older Swiss-Prot with isoforms FASTA (9855s). Strangely, I could not replicate my results and got different significantly regulated proteins in MSstatsTMT (same parameters). Upon closer inspection, I noticed that the number of PSMs from the 2024 run was lower than 2025. For example...

-   2024 - Swiss-Prot with isoforms FASTA (9855s) - 788953 PSMs identified across 96 raw files.

-   2025 - Swiss-Prot with isoforms FASTA (9878s) - 768226 PSMs identified across 96 raw files.

-   [Side Note: Yes, aware of caveats with using isoforms & challenges with protein grouping. Will likely stick to Swiss-Prot canonical only or Reference Proteome (1protein sequence per gene) for further analyses.]{.smallcaps}

I was quite surprised because the new FASTA only had 22 more sequences, and the search method was the same. However, I remember that in the 2024 analysis was accidentally run with as a batch unchecked (forgot it was the opposite at the time) which produced one large .pdresults file (no run errors & all 96 raw files in PSM sheet). Whereas all the analyses in 2025 were run one by one (as File/as batch checked). Because of this, I decided to double-check as see whether.

Using the exact same .pdstudy (processingWF + consensusWF + FASTAs) I re-ran a subset of 8 raw files, either with as batch checked (by file/one by one) or as batch unchecked (by batch/1 result file)

### Current Analysis

This finally brings me to the current analyses presented in this R script. Here I run a quick analysis on the PSMs and MSMS Spectrum Info outputs from PD2.4 analysis of those same 8 raw files.

#### Key

-   **2024** - *e.g. PSMs_2024 or MSMS_2024*

    -   Data from the 8 raw files extracted from the large .pdresults file export produced by the PD2.4 analysis that submitted 96 files run as a batch (As batch unchecked). PD analysis submitted on 01 Nov 2024.

-   **2024_rrb** - *e.g. PSMs_2024_rrb or MSMS_2024_rrb*

    -   Here, those same 8 raw files were re–run as a batch of 8 (as batch unchecked) in PD2.4 with the exact same method. PD analysis submitted on 11 Sept 2025.

-   **2024_rro** - *e.g. PSMs_2024_rro or MSMS_2024_rro*

    -   Again, those same 8 raw files were re–run but one by one (as batch checked/by file) in PD2.4 with the exact same method. The individual .pdresults were exported & pulled in R, and appended together. PD analysis submitted on 11 Sept 2025.

-   **2025** - *e.g. PSMs_2025 or MSMS_2025*

    -   The exported results from the 2025 runs of those 8 files were pulled in and appended together. This is similar to 2024_rro in the sense that all analyses here were submitted one by one (as batched checked/by file). However, different from all the 3 2024 analyses above in the sense that it uses a slightly new search database with 22 more sequences (see info above), but otherwise uses the exact same analysis method.

## Setup

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r message=FALSE, warning=FALSE}
library(readr) # read_tsv(), write_csv etc
library(dplyr) # %>% + other wrangling
library(stringr) # strings e.g. str_detect()
library(tidyr) # pivot_wider
library(ggplot2) # plots 
library(paletteer) # color palette
library(purrr) # for map()
library(ggvenn) # use old one instead
library(cowplot) #plot_grid()
library(knitr) # kable()
```

-   [**7.4 Reading data from multiple files**](https://r4ds.hadley.nz/data-import.html#sec-readr-directory)

```{r}
# Individual .txt from 2025_09_sp_w_iso
# Using method from R4ds 2e (7.4) - simpler than before as no bind_rows is required.

mydir = 'input/2025_09_sp_w_iso_Steph_TMTpro_PSMs_MSstatsTMT_EEvsSH_for_PhD/txt_exports/'

PSMs_F1_8_files <- list.files(
  path = mydir, 
  pattern = ".*DEC2021_B1_TR1.*PSM.*\\.txt$", # DEC2021 1st then PSM 2nd & end .txt
  full.names = TRUE)

# Can also have pattern below if order varies
# pattern = ".*PSM.*DEC2021_B1_TR1_.*\\.txt$|.*DEC2021_B1_TR1_.*PSM.*\\.txt$",

PSMs_2025 <- read_tsv(PSMs_F1_8_files, id = "file_path")
```

-   [filter with str_detect()](https://r4ds.hadley.nz/regexps.html#sec-stringr-regex-funs) - <https://r4ds.hadley.nz/regexps.html#sec-stringr-regex-funs>

```{r}
# 2024 mega PSMs from PD
df_2024 <- read_tsv("input/Steph_01NOV2024_all_EE-SH_M-F_MSstatsTMT/NOV2024_PSMs__Steph_TMTpro-16plex_all.txt",
                    id = "file_path")
  
colnames(df_2024)

# 
PSMs_2024 <- df_2024 %>%
  filter(str_detect(`Spectrum File`,  "DEC2021_B1_TR1"))
```

```{r}
# 2024 rrb (DEC2021_B1_TR1_F1-8 batch .pdresults - rrb_2025_09_11)
PSMs_2024_rrb <- read_tsv('input/Steph_01NOV2024_all_EE-SH_M-F_MSstatsTMT_2025_09_rr/txt_exports/Steph_TMTpro-16plex_DEC2021_B1_TR1_HpHF_F1-8_rrb_2025_09_11_PSMs.txt',
                          id = "file_path")
```

```{r}
# 2024 rro (DEC2021_B1_TR1_F1-8 individual .pdresults - rr0_2025_09_11)
mydir_rro = 'input/Steph_01NOV2024_all_EE-SH_M-F_MSstatsTMT_2025_09_rr/txt_exports/'

PSMs_rro_files <- list.files(path = mydir_rro,
                             pattern = ".*rro.*PSM.*\\.txt$",
                             full.names = TRUE)


PSMs_2024_rro <- read_tsv(PSMs_rro_files,
                          id = "file_path")
  

```

#### Quick Checks

```{r}
# check it has the correct spectrum files. all the same!
unique(PSMs_2024$`Spectrum File`)
unique(PSMs_2024_rrb$`Spectrum File`)
unique(PSMs_2024_rro$`Spectrum File`)
unique(PSMs_2025$`Spectrum File`)

# no. of annotated sequences/PSMs differ!?!?
dim(PSMs_2024) #73250 x 55
dim(PSMs_2024_rrb) #76101 x 55
dim(PSMs_2024_rro) #77117 x 55
dim(PSMs_2025) #77233 x 55
```

```{r}
# summary(PSMs_2024)
# summary(PSMs_2024_rrb)
# summary(PSMs_2024_rro)
# summary(PSMs_2025)
```

```{r}
# No. of unique annotated Sequences ()
length(unique(PSMs_2024$`Annotated Sequence`)) #33688
length(unique(PSMs_2024_rrb$`Annotated Sequence`)) #34781
length(unique(PSMs_2024_rro$`Annotated Sequence`)) #36268
length(unique(PSMs_2025$`Annotated Sequence`)) #36343

# No of unique Protein Accessions (note does account for ; ;)
length(unique(PSMs_2024$`Protein Accessions`)) #4238
length(unique(PSMs_2024_rrb$`Protein Accessions`)) #4393
length(unique(PSMs_2024_rro$`Protein Accessions`)) #4565
length(unique(PSMs_2025$`Protein Accessions`)) #4554

# No. of unique Master Protein Accessions (note does account for ; ;)
length(unique(PSMs_2024$`Master Protein Accessions`)) #3808
length(unique(PSMs_2024_rrb$`Master Protein Accessions`)) #3844
length(unique(PSMs_2024_rro$`Master Protein Accessions`)) #4058
length(unique(PSMs_2025$`Master Protein Accessions`)) #4045
```

## Part 1 - Looking at Unique Annotated Sequences (PSMs)

Make combined df. Get unique sequences in each df. Then do barplot & then vennys

```{r}
#https://dplyr.tidyverse.org/reference/bind_rows.html
# bind_rows() .id alone doesn't work (just gives 1,2,3,4). need list & var name = df
cPSMs <- bind_rows(list(PSMs_2024 = PSMs_2024, 
                        PSMs_2024_rrb = PSMs_2024_rrb, 
                        PSMs_2024_rro = PSMs_2024_rro, 
                        PSMs_2025 = PSMs_2025),
                   .id = "df_id")
```

cPSMs_u_aseq

```{r}
# https://dplyr.tidyverse.org/reference/distinct.html
# get distinct/unique annotated sequences by df group
cPSMs_u_aseq <- cPSMs %>%
  group_by(df_id) %>% 
  distinct(`Annotated Sequence`) 

# dim(cPSMs_u_aseq) # 141080 x 2
# length(unique(cPSMs_u_aseq$`Annotated Sequence`)) # 37343
```

### Barplot

```{r fig.height=6, fig.width=9}
# https://r-charts.com/color-palettes/
# https://r-graph-gallery.com/package/paletteer.html
# plot no. of unique annotated seqeunces per df/group
cPSMs_u_aseq_plot <- cPSMs_u_aseq %>%
  group_by(df_id) %>%
  tally() %>%  # get tally of unique sequences by df for plot
  ggplot(aes(x = df_id, y = n, fill = df_id)) +
  geom_col(width = 0.6, color = "grey10", alpha = .7) + 
  ylim(0,50000) + 
  geom_text(aes(label = paste0(n)), # annotate count on graph
            vjust = -.2) +
  scale_fill_paletteer_d("ggthemes::excel_Organic") +
  labs(x = "PSM dataframe set (different PD analysis)",
       y = "no. of unique annotated sequences (n)",
       title = "Count of distinct annotated sequences in each PD PSM set (df_id)",
       subtitle = "Analysing as batch drops the no. of unique annotated sequences!",
       caption = "Each set has the SAME 8 LC-MS files! These are DEC2021_B1_TR1_F1-8.\n 
       PSMs_2024, PSMs_2024_rrb & PSMs_2024_rro all use the SAME FASTA & PD method! \n         The ONLY DIFFERENCE is that PSMs_2024 was analysed as batch (96 files submitted),
       PSMs_2024_rrb was analysed as batch (8 files submitted), while PSMs_2024_rro was submitted one by one (not as batch). \n 
       PSMs_2025 also submitted files one by one like PSMs_2024_rro & used the same PD method, but had a slightly newer SwissProt db (23 more seq).\n
       The mild increase between PSMs_2025 & PSMs_2024_rro of 75 seems reasonable for increasing the FASTA db by 23 sequences (although need to look at seq ID too in venny!). 
       However, the change between the other 3 based on running individually or by batch is bizzare. 
       The LC-MS raw files were input as files not fractions in PD & the PSMs table from PD shows PSMs by separate LCMS runs. So it should not be grouping it!
       Also the PD userguide does not indicate grouping PSMs (it groups at the peptide & protein grouping level!)
       Also the bizzare thing is that changing the no. of files submitted changes the no. of PSMs observed?!? ") +
  theme_bw() + 
  theme(legend.position = "bottom")


plot(cPSMs_u_aseq_plot)
```

### Vennys

```{r}
# make venny function 
# see old folder in R_WD > thesis_dev > L&H2 > L&H2_LCMS_protein_counts_for_PhD_2025_v2
venny <- function(data, colors = NULL) {
  v <- ggvenn(data,
              digits = 0,
              stroke_alpha = 0.7,
              fill_alpha = 0.45,
              stroke_size = 0.5,
              set_name_size = 4,
              text_size = 3.5,
              fill_color = colors)
  }
```

```{r}
# prep data
# map() function from purrr https://purrr.tidyverse.org/reference/map.html
#split_u_aseq <- split(cPSMs_u_aseq, f = cPSMs_u_aseq$df_id) # nvm 

lists_u_aseq <- cPSMs_u_aseq %>%
  split(.$df_id) %>% # split by df group
  map(~pull(., `Annotated Sequence`)) # makes mini func ~ to pull aseq & applies 2 all 

str(lists_u_aseq) # large list with 4 elements
```

```{r}
# v1 - all 4 
v1 <- venny(data = lists_u_aseq,
            colors = c('#83992A', '#3C9770', '#44709D', '#A23C33'))

# v2 - difference is just FASTA
c1 <- lists_u_aseq[c('PSMs_2024_rro', 'PSMs_2025')] # select elements/lists
v2 <- venny(data = c1,
            colors = c('#44709D', '#A23C33'))

# v3 - difference is processed in set of 96 or 1 by 1
c2 <- lists_u_aseq[c('PSMs_2024' ,'PSMs_2024_rro')] # select elements/lists
v3 <- venny(data = c2,
            colors = c('#83992A', '#44709D'))

# v4 - difference is processed in set of 8 or 1 by 1
c3 <- lists_u_aseq[c('PSMs_2024_rrb' ,'PSMs_2024_rro')] # select elements/lists
v4 <- venny(data = c3,
            colors = c('#3C9770', '#44709D'))

# v4 - difference is processed in set of 96 or 8
c4 <- lists_u_aseq[c('PSMs_2024','PSMs_2024_rrb')] # select elements/lists
v5 <- venny(data = c4,
            colors = c('#83992A', '#3C9770'))

# v6 - comparing all 2024 - only difference is no. of files submitted 
c5 <- lists_u_aseq[c('PSMs_2024', 'PSMs_2024_rrb' ,'PSMs_2024_rro')]
v6 <- venny(data = c5,
            colors = c('#83992A', '#3C9770', '#44709D'))
```

```{r fig.height=12, fig.width=10}
# cowplot all vennys
plot_grid(v1, v2, v3, v4, v5, v6,
          ncol = 2,
          labels = c('A','B', 'C', 'D', 'E', 'F'))
```

### Interim Summary (2025-09-12)

Overall, it seems that processing as batch (aka not clicking the as batch button on the analysis tab) drops the number of unique PSMs/annotated sequences compared to processing 1 by 1. And that the larger the number of files submitted as a batch are, the more the PSMs drop.

The PD 2.4 userguide is confusing & is the opposite of what they mean. The new PD 3.1 user guide has at least changed this button to ['by file'](https://docs.thermofisher.com/r/Proteome-Discoverer-3.1-User-Guide/en-US1410394507v1) now, and somewhat gives [more info on what it is doing in the search](https://docs.thermofisher.com/r/Proteome-Discoverer-3.1-User-Guide/en-US1297231499v1). Still why should processing as batch & increasing the number of files affect the PSMs identified? The separate spectrum file is retained in the PSM sheet. I thought batch affects the grouping level, e.g. (peptide groups to protein group sheets)?!?! Confused! -\> CONT next week -\> to email thermo.

\~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~

## Part 2 - 2025-09-16

### Check MS/MS Spectrum Info File too.

```{r}
# 2025 sp w iso 
MSMS_2025_files <- list.files(
  path = 'input/2025_09_sp_w_iso_Steph_TMTpro_PSMs_MSstatsTMT_EEvsSH_for_PhD/txt_exports/',
  pattern = '.*DEC2021_B1_TR1.*MSMSSpectrumInfo.*\\.txt$',
  full.names = TRUE
)

MSMS_2025 <- read_tsv(MSMS_2025_files, id = "file_path")

dim(MSMS_2025)
```

```{r}
# Nov 2024 sp w iso rb as 96
MSMS_2024_file <- read_tsv('input/Steph_01NOV2024_all_EE-SH_M-F_MSstatsTMT/NOV2024_MSMSSpectrumInfo__Steph_TMTpro-16plex_all.txt', id = "file_path")

# filter to just DEC_2021_TR1
MSMS_2024 <- MSMS_2024_file %>%
  filter(str_detect(`Spectrum File`, "DEC2021_B1_TR1"))

dim(MSMS_2024)
```

```{r}
# Nov 24 sp w iso rrb F1-8
MSMS_2024_rrb <- read_tsv('input/Steph_01NOV2024_all_EE-SH_M-F_MSstatsTMT_2025_09_rr/txt_exports/Steph_TMTpro-16plex_DEC2021_B1_TR1_HpHF_F1-8_rrb_2025_09_11_MSMSSpectrumInfo.txt',
                          id = "file_path")

dim(MSMS_2024_rrb)
```

```{r}
# Nov 24 sp w iso rro F1-8
MSMS_2024_rro_file <- list.files(path = 'input/Steph_01NOV2024_all_EE-SH_M-F_MSstatsTMT_2025_09_rr/txt_exports/',
                             pattern = ".*rro.*MSMSSpectrumInfo.*\\.txt$",
                             full.names = TRUE)

MSMS_2024_rro <- read_tsv(MSMS_2024_rro_file,
                          id = "file_path")

dim(MSMS_2024_rro)
```

All the same dimensions - 333979 x 20 (19)

Next make master file & determine if the precursor identifed are different (

```{r}
cMSMS <- bind_rows(list(MSMS_2024 = MSMS_2024,
                        MSMS_2024_rrb = MSMS_2024_rrb,
                        MSMS_2024_rro = MSMS_2024_rro,
                        MSMS_2025 = MSMS_2025),
                   .id = "df_id")
```

```{r}
colnames(cMSMS)

# No. of Precursors -#333979 same across all (as expected since they're the same files!)
cMSMS %>%
  group_by(df_id) %>%
  count(`Number of Precursors`)
  
# Differs despite the same files - links to issue shown in the analysis above.
# note this is total annotated sequences & not unique - hence larger number than above.
no_id_pc <- cMSMS %>%
  group_by(df_id) %>%
  # count(`Number of PSMs`) # same metric as No. of Identified Precursors.
  count(`Number of Identified Precursors`) %>%
  pivot_wider(names_from = `Number of Identified Precursors`,
              values_from = n) %>%
  rename(ID_precursor_0 = '0', # in quote otherwise does column order.
         ID_precursor_1 = '1') %>%
  mutate(no_precursors = ID_precursor_0 + ID_precursor_1)
  
no_id_pc
```

### Interim Summary (2025-09-16)

This bit doesn't add much, but essentially shows that MSMS spectra are not being lost; rather, something is happening when running as a batch, in which PSMs/Precursors are no longer identified in some of the MS/MS spectra.

\~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~

### (2025-09-17) Additional Calculations

```{r}
colnames(cPSMs)

# count of annotated sequences (nrow) in each raw file by analysis/df_id
anno_seq <- cPSMs %>%
  group_by(df_id) %>% 
  count(`Spectrum File`) %>%
  mutate(fraction = gsub("Steph_TMTpro-16plex_DEC2021_B1_TR1_HpHF_|\\.raw$",
                         "", # trim file name to just keep fraction info
                         `Spectrum File`))
```

```{r}
anno_seq_plot <- anno_seq %>%
  ggplot(aes(x = fraction,
             y = n,
             fill = df_id)) +
  geom_col(position = position_dodge(),
           width = 0.8, color = "grey10", alpha = .7) + 
  scale_fill_paletteer_d("ggthemes::excel_Organic") +
  labs(
    x = "Fraction No.",
    y = "No. of Annotated Sequences (PSMs)",
    title = "",
    subtitle = ""
  ) +
  theme_bw() +
  theme(legend.position = "bottom")
  
anno_seq_plot
```

\~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~ \~

```{r}
# P19493_all <- cPSMs %>%
#   filter(str_detect(`Protein Accessions`, "P19493")) %>%
#   group_by(df_id, `Spectrum File`)


P19493_mp <- cPSMs %>%
  filter(str_starts(`Master Protein Accessions`, "P19493"))
  
unique(P19493_mp$`Annotated Sequence`) # 13

P19493_mp %>%
  count(`Annotated Sequence`, sort = TRUE)
```

```{r}
# prep data

list_P19493 <- P19493_mp %>%
  split(.$df_id) %>%
  map(~pull(., `Annotated Sequence`))


str(list_P19493)
```

```{r}
v_P19493 <- venny(data = list_P19493,
                 colors = c('#83992A', '#3C9770', '#44709D', '#A23C33'))

v_P19493
```

```{r}
#dplyr set diff

# 2025 vs 2024 run by files has no differences despite slightly diff FASTA
setdiff(list_P19493$PSMs_2024_rro,list_P19493$PSMs_2025) 


# 8 files run as batch vs by file differ despite exact same FASTA & pWF & cWF!
setdiff(list_P19493$PSMs_2024_rro, list_P19493$PSMs_2024_rrb) 



```

```{r}
# present but assigned to different protein accessions
# ambiguity is inherent since part of same protein family plus use of isoform db.
# but not sure why assignment differs based on running as batch & as file
cont_setdiff <- cPSMs %>%
  filter(`Annotated Sequence` %in% c("[K].vGGNLDSk.[G]",
                                     "[K].gYGVATPk.[G]",
                                     "[R].kPcDTMk.[V]",
                                     "[R].kPcDTmk.[V]",
                                     "[K].iAVYEk.[M]"))
```

# Part 3 - Email to Thermo (2025-09-18)

```{r}

```

```{r}
kable(anno_seq)
```

```{r}
anno_seq_plot
```

### Barplot of unique annotated sequences

```{r fig.height=12, fig.width=9, message=FALSE, warning=FALSE}
cPSMs_u_aseq_plot
```

### Venn Diagrams showing similarities and differences of the unique annotated sequences across the analysis

#### All 4 analyses.

Note -

```{r fig.height=5, fig.width=9, message=FALSE, warning=FALSE}
v1
```

#### The 3 2024 analyses. 

This Venn diagram shows that the number of unique identified annotated sequences (PSMs) from the same 8 raw files differs depending on whether it was submitted as batch (2024) or by file (2024_rro) in PD2.4. And also based on the size of the batch submitted - 2024 - 96 files vs 2024_rrb - 8 files.

*Note the analysis method and database were exactly the same for all.*

```{r fig.height=5, fig.width=9, message=FALSE, warning=FALSE}
v6
```

### MS/MS Spectrum Info Tables - ID precursors

Shows that the MS/MS info is the same but the PSMs/precursors identified differ.

```{r}
kable(no_id_pc)
```

```{r}


```
